{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] ‘1.22.3’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘1.6.0’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘3.2.1’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-4\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘2.5.4’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘3.4.4’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘1.5’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘0.8.0.1’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "------------------------------------------------------------------------------\n",
      "You have loaded plyr after dplyr - this is likely to cause problems.\n",
      "If you need functions from both plyr and dplyr, please load plyr first, then dplyr:\n",
      "library(plyr); library(dplyr)\n",
      "------------------------------------------------------------------------------\n",
      "\n",
      "Attaching package: ‘plyr’\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    arrange, count, desc, failwith, id, mutate, rename, summarise,\n",
      "    summarize\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘1.8.4’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘0.5.2’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘1.4.0’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘1.22’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: S4Vectors\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "The following objects are masked from ‘package:biomformat’:\n",
      "\n",
      "    colnames, rownames\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:plyr’:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    first, rename\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "The following object is masked from ‘package:plyr’:\n",
      "\n",
      "    desc\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "The following object is masked from ‘package:phyloseq’:\n",
      "\n",
      "    distance\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "Loading required package: SummarizedExperiment\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "The following object is masked from ‘package:phyloseq’:\n",
      "\n",
      "    sampleNames\n",
      "\n",
      "Loading required package: DelayedArray\n",
      "Loading required package: matrixStats\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "The following objects are masked from ‘package:Biobase’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "The following object is masked from ‘package:plyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "Attaching package: ‘DelayedArray’\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    apply\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘1.18.1’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "[1] ‘0.4.7’"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Install and load R pakcages that are necessary for the analysis - Packages are collections of R functions, data, \n",
    "# and compiled code in a well-defined format. Remove the hash sign to download and install the packages.\n",
    "\n",
    "#source('http://bioconductor.org/biocLite.R')\n",
    "#biocLite('phyloseq')\n",
    "library(\"phyloseq\")\n",
    "packageVersion(\"phyloseq\")\n",
    "\n",
    "#biocLite(\"biomformat\")\n",
    "library(\"biomformat\")\n",
    "packageVersion(\"biomformat\")\n",
    "\n",
    "#install.packages(\"ggplot2\")\n",
    "library(\"ggplot2\")\n",
    "packageVersion(\"ggplot2\")\n",
    "\n",
    "#install.packages(\"vegan\")\n",
    "library(\"vegan\")\n",
    "packageVersion('vegan')\n",
    "\n",
    "#install.packages(\"grid\")\n",
    "library(\"grid\")\n",
    "packageVersion('grid')\n",
    "\n",
    "#install.packages(\"magrittr\")\n",
    "library(magrittr)\n",
    "packageVersion('magrittr')\n",
    "\n",
    "library(dplyr)\n",
    "packageVersion('dplyr')\n",
    "\n",
    "library(plyr)\n",
    "packageVersion('plyr')\n",
    "\n",
    "library(broom)\n",
    "packageVersion('broom')\n",
    "\n",
    "library('stringr')\n",
    "packageVersion('stringr')\n",
    "\n",
    "library('knitr')\n",
    "packageVersion('knitr')\n",
    "\n",
    "library('DESeq2')\n",
    "packageVersion('DESeq2')\n",
    "\n",
    "#install.packages(\"ggfortify\")\n",
    "library('ggfortify')\n",
    "packageVersion('ggfortify')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>582.T1</li>\n",
       "\t<li>586.T5</li>\n",
       "\t<li>571.T2</li>\n",
       "\t<li>578.cacum.T5</li>\n",
       "\t<li>576.T5</li>\n",
       "\t<li>581.T4</li>\n",
       "\t<li>584.T5</li>\n",
       "\t<li>585.T0</li>\n",
       "\t<li>601.T1</li>\n",
       "\t<li>600.T4</li>\n",
       "\t<li>578.T5</li>\n",
       "\t<li>596.T2</li>\n",
       "\t<li>585.T1</li>\n",
       "\t<li>578.T1</li>\n",
       "\t<li>601.T5</li>\n",
       "\t<li>572.T1</li>\n",
       "\t<li>602.T1</li>\n",
       "\t<li>577.T4</li>\n",
       "\t<li>592.T4</li>\n",
       "\t<li>572.T3</li>\n",
       "\t<li>578.T2</li>\n",
       "\t<li>577.T0</li>\n",
       "\t<li>597.T3</li>\n",
       "\t<li>596.T1</li>\n",
       "\t<li>589.T3</li>\n",
       "\t<li>588.T2</li>\n",
       "\t<li>599.T2</li>\n",
       "\t<li>602.T2</li>\n",
       "\t<li>594.T1</li>\n",
       "\t<li>597.T2</li>\n",
       "\t<li>584.T3</li>\n",
       "\t<li>575.T1</li>\n",
       "\t<li>570.T4</li>\n",
       "\t<li>577.cacum.T5</li>\n",
       "\t<li>590.T3</li>\n",
       "\t<li>589.T2</li>\n",
       "\t<li>573.T2</li>\n",
       "\t<li>588.T5</li>\n",
       "\t<li>575.T2</li>\n",
       "\t<li>594.T5</li>\n",
       "\t<li>574.T0</li>\n",
       "\t<li>598.T3</li>\n",
       "\t<li>581.cacum.T5</li>\n",
       "\t<li>598.T0</li>\n",
       "\t<li>584.T0</li>\n",
       "\t<li>575.T0</li>\n",
       "\t<li>601.T0</li>\n",
       "\t<li>577.T3</li>\n",
       "\t<li>598.T5</li>\n",
       "\t<li>580.T5</li>\n",
       "\t<li>571.T4</li>\n",
       "\t<li>576.T2</li>\n",
       "\t<li>581.T2</li>\n",
       "\t<li>580.T4</li>\n",
       "\t<li>573.T5</li>\n",
       "\t<li>600.T2</li>\n",
       "\t<li>599.T3</li>\n",
       "\t<li>584.T2</li>\n",
       "\t<li>599.T5</li>\n",
       "\t<li>602.T5</li>\n",
       "\t<li>601.T2</li>\n",
       "\t<li>583.T2</li>\n",
       "\t<li>598.T4</li>\n",
       "\t<li>589.T0</li>\n",
       "\t<li>595.T2</li>\n",
       "\t<li>571.T1</li>\n",
       "\t<li>593.T1</li>\n",
       "\t<li>589.T1</li>\n",
       "\t<li>599.T0</li>\n",
       "\t<li>571.cacum.T5</li>\n",
       "\t<li>580.T0</li>\n",
       "\t<li>572.T0</li>\n",
       "\t<li>586.T4</li>\n",
       "\t<li>592.T1</li>\n",
       "\t<li>590.T2</li>\n",
       "\t<li>583.cacum.T5</li>\n",
       "\t<li>570.T2</li>\n",
       "\t<li>574.T2</li>\n",
       "\t<li>578.T0</li>\n",
       "\t<li>586.T2</li>\n",
       "\t<li>572.cacum.T5</li>\n",
       "\t<li>588.T1</li>\n",
       "\t<li>590.T1</li>\n",
       "\t<li>593.T5</li>\n",
       "\t<li>577.T2</li>\n",
       "\t<li>581.T1</li>\n",
       "\t<li>588.T4</li>\n",
       "\t<li>574.T5</li>\n",
       "\t<li>603.T2</li>\n",
       "\t<li>595.T1</li>\n",
       "\t<li>585.cacum.T5</li>\n",
       "\t<li>582.T2</li>\n",
       "\t<li>599.T1</li>\n",
       "\t<li>588.T3</li>\n",
       "\t<li>572.T4</li>\n",
       "\t<li>582.T5</li>\n",
       "\t<li>573.T1</li>\n",
       "\t<li>593.T2</li>\n",
       "\t<li>581.T3</li>\n",
       "\t<li>584.cacum.T5</li>\n",
       "\t<li>582.cacum.T5</li>\n",
       "\t<li>592.T5</li>\n",
       "\t<li>576.T0</li>\n",
       "\t<li>602.T4</li>\n",
       "\t<li>583.T1</li>\n",
       "\t<li>574.T4</li>\n",
       "\t<li>592.T0</li>\n",
       "\t<li>573.T4</li>\n",
       "\t<li>578.T4</li>\n",
       "\t<li>590.T4</li>\n",
       "\t<li>574.cacum.T5</li>\n",
       "\t<li>591.T5</li>\n",
       "\t<li>600.T5</li>\n",
       "\t<li>595.T3</li>\n",
       "\t<li>590.T5</li>\n",
       "\t<li>585.T3</li>\n",
       "\t<li>570.T1</li>\n",
       "\t<li>575.T4</li>\n",
       "\t<li>580.T2</li>\n",
       "\t<li>596.T0</li>\n",
       "\t<li>594.T3</li>\n",
       "\t<li>600.T0</li>\n",
       "\t<li>595.T5</li>\n",
       "\t<li>589.T4</li>\n",
       "\t<li>589.T5</li>\n",
       "\t<li>596.T4</li>\n",
       "\t<li>582.T0</li>\n",
       "\t<li>593.T4</li>\n",
       "\t<li>575.T5</li>\n",
       "\t<li>583.T5</li>\n",
       "\t<li>570.cacum.T5</li>\n",
       "\t<li>570.T3</li>\n",
       "\t<li>592.T3</li>\n",
       "\t<li>572.T5</li>\n",
       "\t<li>575.T3</li>\n",
       "\t<li>603.T1</li>\n",
       "\t<li>595.T4</li>\n",
       "\t<li>602.T0</li>\n",
       "\t<li>591.T1</li>\n",
       "\t<li>594.T0</li>\n",
       "\t<li>571.T3</li>\n",
       "\t<li>581.T5</li>\n",
       "\t<li>580.cacum.T5</li>\n",
       "\t<li>580.T3</li>\n",
       "\t<li>576.T1</li>\n",
       "\t<li>570.T5</li>\n",
       "\t<li>574.T1</li>\n",
       "\t<li>576.T3</li>\n",
       "\t<li>588.T0</li>\n",
       "\t<li>584.T1</li>\n",
       "\t<li>583.T3</li>\n",
       "\t<li>602.T3</li>\n",
       "\t<li>571.T0</li>\n",
       "\t<li>593.T0</li>\n",
       "\t<li>603.T5</li>\n",
       "\t<li>573.T0</li>\n",
       "\t<li>573.cacum.T5</li>\n",
       "\t<li>603.T3</li>\n",
       "\t<li>599.T4</li>\n",
       "\t<li>570.T0</li>\n",
       "\t<li>576.T4</li>\n",
       "\t<li>597.T1</li>\n",
       "\t<li>573.T3</li>\n",
       "\t<li>594.T2</li>\n",
       "\t<li>603.T4</li>\n",
       "\t<li>591.T4</li>\n",
       "\t<li>582.T3</li>\n",
       "\t<li>581.T0</li>\n",
       "\t<li>586.T0</li>\n",
       "\t<li>597.T0</li>\n",
       "\t<li>585.T4</li>\n",
       "\t<li>586.T1</li>\n",
       "\t<li>590.T0</li>\n",
       "\t<li>596.T5</li>\n",
       "\t<li>586.cacum.T5</li>\n",
       "\t<li>583.T0</li>\n",
       "\t<li>584.T4</li>\n",
       "\t<li>601.T4</li>\n",
       "\t<li>574.T3</li>\n",
       "\t<li>585.T5</li>\n",
       "\t<li>600.T3</li>\n",
       "\t<li>598.T1</li>\n",
       "\t<li>571.T5</li>\n",
       "\t<li>578.T3</li>\n",
       "\t<li>596.T3</li>\n",
       "\t<li>586.T3</li>\n",
       "\t<li>585.T2</li>\n",
       "\t<li>576.cacum.T5</li>\n",
       "\t<li>598.T2</li>\n",
       "\t<li>597.T5</li>\n",
       "\t<li>601.T3</li>\n",
       "\t<li>597.T4</li>\n",
       "\t<li>583.T4</li>\n",
       "\t<li>582.T4</li>\n",
       "\t<li>577.T5</li>\n",
       "\t<li>591.T2</li>\n",
       "\t<li>591.T3</li>\n",
       "\t<li>595.T0</li>\n",
       "\t<li>580.T1</li>\n",
       "\t<li>600.T1</li>\n",
       "\t<li>603.T0</li>\n",
       "\t<li>577.T1</li>\n",
       "\t<li>591.T0</li>\n",
       "\t<li>572.T2</li>\n",
       "\t<li>593.T3</li>\n",
       "\t<li>594.T4</li>\n",
       "\t<li>592.T2</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'570.cacum.T5'</li>\n",
       "\t\t<li>'570.T0'</li>\n",
       "\t\t<li>'570.T1'</li>\n",
       "\t\t<li>'570.T2'</li>\n",
       "\t\t<li>'570.T3'</li>\n",
       "\t\t<li>'570.T4'</li>\n",
       "\t\t<li>'570.T5'</li>\n",
       "\t\t<li>'571.cacum.T5'</li>\n",
       "\t\t<li>'571.T0'</li>\n",
       "\t\t<li>'571.T1'</li>\n",
       "\t\t<li>'571.T2'</li>\n",
       "\t\t<li>'571.T3'</li>\n",
       "\t\t<li>'571.T4'</li>\n",
       "\t\t<li>'571.T5'</li>\n",
       "\t\t<li>'572.cacum.T5'</li>\n",
       "\t\t<li>'572.T0'</li>\n",
       "\t\t<li>'572.T1'</li>\n",
       "\t\t<li>'572.T2'</li>\n",
       "\t\t<li>'572.T3'</li>\n",
       "\t\t<li>'572.T4'</li>\n",
       "\t\t<li>'572.T5'</li>\n",
       "\t\t<li>'573.cacum.T5'</li>\n",
       "\t\t<li>'573.T0'</li>\n",
       "\t\t<li>'573.T1'</li>\n",
       "\t\t<li>'573.T2'</li>\n",
       "\t\t<li>'573.T3'</li>\n",
       "\t\t<li>'573.T4'</li>\n",
       "\t\t<li>'573.T5'</li>\n",
       "\t\t<li>'574.cacum.T5'</li>\n",
       "\t\t<li>'574.T0'</li>\n",
       "\t\t<li>'574.T1'</li>\n",
       "\t\t<li>'574.T2'</li>\n",
       "\t\t<li>'574.T3'</li>\n",
       "\t\t<li>'574.T4'</li>\n",
       "\t\t<li>'574.T5'</li>\n",
       "\t\t<li>'575.T0'</li>\n",
       "\t\t<li>'575.T1'</li>\n",
       "\t\t<li>'575.T2'</li>\n",
       "\t\t<li>'575.T3'</li>\n",
       "\t\t<li>'575.T4'</li>\n",
       "\t\t<li>'575.T5'</li>\n",
       "\t\t<li>'576.cacum.T5'</li>\n",
       "\t\t<li>'576.T0'</li>\n",
       "\t\t<li>'576.T1'</li>\n",
       "\t\t<li>'576.T2'</li>\n",
       "\t\t<li>'576.T3'</li>\n",
       "\t\t<li>'576.T4'</li>\n",
       "\t\t<li>'576.T5'</li>\n",
       "\t\t<li>'577.cacum.T5'</li>\n",
       "\t\t<li>'577.T0'</li>\n",
       "\t\t<li>'577.T1'</li>\n",
       "\t\t<li>'577.T2'</li>\n",
       "\t\t<li>'577.T3'</li>\n",
       "\t\t<li>'577.T4'</li>\n",
       "\t\t<li>'577.T5'</li>\n",
       "\t\t<li>'578.cacum.T5'</li>\n",
       "\t\t<li>'578.T0'</li>\n",
       "\t\t<li>'578.T1'</li>\n",
       "\t\t<li>'578.T2'</li>\n",
       "\t\t<li>'578.T3'</li>\n",
       "\t\t<li>'578.T4'</li>\n",
       "\t\t<li>'578.T5'</li>\n",
       "\t\t<li>'580.cacum.T5'</li>\n",
       "\t\t<li>'580.T0'</li>\n",
       "\t\t<li>'580.T1'</li>\n",
       "\t\t<li>'580.T2'</li>\n",
       "\t\t<li>'580.T3'</li>\n",
       "\t\t<li>'580.T4'</li>\n",
       "\t\t<li>'580.T5'</li>\n",
       "\t\t<li>'581.cacum.T5'</li>\n",
       "\t\t<li>'581.T0'</li>\n",
       "\t\t<li>'581.T1'</li>\n",
       "\t\t<li>'581.T2'</li>\n",
       "\t\t<li>'581.T3'</li>\n",
       "\t\t<li>'581.T4'</li>\n",
       "\t\t<li>'581.T5'</li>\n",
       "\t\t<li>'582.cacum.T5'</li>\n",
       "\t\t<li>'582.T0'</li>\n",
       "\t\t<li>'582.T1'</li>\n",
       "\t\t<li>'582.T2'</li>\n",
       "\t\t<li>'582.T3'</li>\n",
       "\t\t<li>'582.T4'</li>\n",
       "\t\t<li>'582.T5'</li>\n",
       "\t\t<li>'583.cacum.T5'</li>\n",
       "\t\t<li>'583.T0'</li>\n",
       "\t\t<li>'583.T1'</li>\n",
       "\t\t<li>'583.T2'</li>\n",
       "\t\t<li>'583.T3'</li>\n",
       "\t\t<li>'583.T4'</li>\n",
       "\t\t<li>'583.T5'</li>\n",
       "\t\t<li>'584.cacum.T5'</li>\n",
       "\t\t<li>'584.T0'</li>\n",
       "\t\t<li>'584.T1'</li>\n",
       "\t\t<li>'584.T2'</li>\n",
       "\t\t<li>'584.T3'</li>\n",
       "\t\t<li>'584.T4'</li>\n",
       "\t\t<li>'584.T5'</li>\n",
       "\t\t<li>'585.cacum.T5'</li>\n",
       "\t\t<li>'585.T0'</li>\n",
       "\t\t<li>'585.T1'</li>\n",
       "\t\t<li>'585.T2'</li>\n",
       "\t\t<li>'585.T3'</li>\n",
       "\t\t<li>'585.T4'</li>\n",
       "\t\t<li>'585.T5'</li>\n",
       "\t\t<li>'586.cacum.T5'</li>\n",
       "\t\t<li>'586.T0'</li>\n",
       "\t\t<li>'586.T1'</li>\n",
       "\t\t<li>'586.T2'</li>\n",
       "\t\t<li>'586.T3'</li>\n",
       "\t\t<li>'586.T4'</li>\n",
       "\t\t<li>'586.T5'</li>\n",
       "\t\t<li>'588.T0'</li>\n",
       "\t\t<li>'588.T1'</li>\n",
       "\t\t<li>'588.T2'</li>\n",
       "\t\t<li>'588.T3'</li>\n",
       "\t\t<li>'588.T4'</li>\n",
       "\t\t<li>'588.T5'</li>\n",
       "\t\t<li>'589.T0'</li>\n",
       "\t\t<li>'589.T1'</li>\n",
       "\t\t<li>'589.T2'</li>\n",
       "\t\t<li>'589.T3'</li>\n",
       "\t\t<li>'589.T4'</li>\n",
       "\t\t<li>'589.T5'</li>\n",
       "\t\t<li>'590.T0'</li>\n",
       "\t\t<li>'590.T1'</li>\n",
       "\t\t<li>'590.T2'</li>\n",
       "\t\t<li>'590.T3'</li>\n",
       "\t\t<li>'590.T4'</li>\n",
       "\t\t<li>'590.T5'</li>\n",
       "\t\t<li>'591.T0'</li>\n",
       "\t\t<li>'591.T1'</li>\n",
       "\t\t<li>'591.T2'</li>\n",
       "\t\t<li>'591.T3'</li>\n",
       "\t\t<li>'591.T4'</li>\n",
       "\t\t<li>'591.T5'</li>\n",
       "\t\t<li>'592.T0'</li>\n",
       "\t\t<li>'592.T1'</li>\n",
       "\t\t<li>'592.T2'</li>\n",
       "\t\t<li>'592.T3'</li>\n",
       "\t\t<li>'592.T4'</li>\n",
       "\t\t<li>'592.T5'</li>\n",
       "\t\t<li>'593.T0'</li>\n",
       "\t\t<li>'593.T1'</li>\n",
       "\t\t<li>'593.T2'</li>\n",
       "\t\t<li>'593.T3'</li>\n",
       "\t\t<li>'593.T4'</li>\n",
       "\t\t<li>'593.T5'</li>\n",
       "\t\t<li>'594.T0'</li>\n",
       "\t\t<li>'594.T1'</li>\n",
       "\t\t<li>'594.T2'</li>\n",
       "\t\t<li>'594.T3'</li>\n",
       "\t\t<li>'594.T4'</li>\n",
       "\t\t<li>'594.T5'</li>\n",
       "\t\t<li>'595.T0'</li>\n",
       "\t\t<li>'595.T1'</li>\n",
       "\t\t<li>'595.T2'</li>\n",
       "\t\t<li>'595.T3'</li>\n",
       "\t\t<li>'595.T4'</li>\n",
       "\t\t<li>'595.T5'</li>\n",
       "\t\t<li>'596.T0'</li>\n",
       "\t\t<li>'596.T1'</li>\n",
       "\t\t<li>'596.T2'</li>\n",
       "\t\t<li>'596.T3'</li>\n",
       "\t\t<li>'596.T4'</li>\n",
       "\t\t<li>'596.T5'</li>\n",
       "\t\t<li>'597.T0'</li>\n",
       "\t\t<li>'597.T1'</li>\n",
       "\t\t<li>'597.T2'</li>\n",
       "\t\t<li>'597.T3'</li>\n",
       "\t\t<li>'597.T4'</li>\n",
       "\t\t<li>'597.T5'</li>\n",
       "\t\t<li>'598.T0'</li>\n",
       "\t\t<li>'598.T1'</li>\n",
       "\t\t<li>'598.T2'</li>\n",
       "\t\t<li>'598.T3'</li>\n",
       "\t\t<li>'598.T4'</li>\n",
       "\t\t<li>'598.T5'</li>\n",
       "\t\t<li>'599.T0'</li>\n",
       "\t\t<li>'599.T1'</li>\n",
       "\t\t<li>'599.T2'</li>\n",
       "\t\t<li>'599.T3'</li>\n",
       "\t\t<li>'599.T4'</li>\n",
       "\t\t<li>'599.T5'</li>\n",
       "\t\t<li>'600.T0'</li>\n",
       "\t\t<li>'600.T1'</li>\n",
       "\t\t<li>'600.T2'</li>\n",
       "\t\t<li>'600.T3'</li>\n",
       "\t\t<li>'600.T4'</li>\n",
       "\t\t<li>'600.T5'</li>\n",
       "\t\t<li>'601.T0'</li>\n",
       "\t\t<li>'601.T1'</li>\n",
       "\t\t<li>'601.T2'</li>\n",
       "\t\t<li>'601.T3'</li>\n",
       "\t\t<li>'601.T4'</li>\n",
       "\t\t<li>'601.T5'</li>\n",
       "\t\t<li>'602.T0'</li>\n",
       "\t\t<li>'602.T1'</li>\n",
       "\t\t<li>'602.T2'</li>\n",
       "\t\t<li>'602.T3'</li>\n",
       "\t\t<li>'602.T4'</li>\n",
       "\t\t<li>'602.T5'</li>\n",
       "\t\t<li>'603.T0'</li>\n",
       "\t\t<li>'603.T1'</li>\n",
       "\t\t<li>'603.T2'</li>\n",
       "\t\t<li>'603.T3'</li>\n",
       "\t\t<li>'603.T4'</li>\n",
       "\t\t<li>'603.T5'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 582.T1\n",
       "\\item 586.T5\n",
       "\\item 571.T2\n",
       "\\item 578.cacum.T5\n",
       "\\item 576.T5\n",
       "\\item 581.T4\n",
       "\\item 584.T5\n",
       "\\item 585.T0\n",
       "\\item 601.T1\n",
       "\\item 600.T4\n",
       "\\item 578.T5\n",
       "\\item 596.T2\n",
       "\\item 585.T1\n",
       "\\item 578.T1\n",
       "\\item 601.T5\n",
       "\\item 572.T1\n",
       "\\item 602.T1\n",
       "\\item 577.T4\n",
       "\\item 592.T4\n",
       "\\item 572.T3\n",
       "\\item 578.T2\n",
       "\\item 577.T0\n",
       "\\item 597.T3\n",
       "\\item 596.T1\n",
       "\\item 589.T3\n",
       "\\item 588.T2\n",
       "\\item 599.T2\n",
       "\\item 602.T2\n",
       "\\item 594.T1\n",
       "\\item 597.T2\n",
       "\\item 584.T3\n",
       "\\item 575.T1\n",
       "\\item 570.T4\n",
       "\\item 577.cacum.T5\n",
       "\\item 590.T3\n",
       "\\item 589.T2\n",
       "\\item 573.T2\n",
       "\\item 588.T5\n",
       "\\item 575.T2\n",
       "\\item 594.T5\n",
       "\\item 574.T0\n",
       "\\item 598.T3\n",
       "\\item 581.cacum.T5\n",
       "\\item 598.T0\n",
       "\\item 584.T0\n",
       "\\item 575.T0\n",
       "\\item 601.T0\n",
       "\\item 577.T3\n",
       "\\item 598.T5\n",
       "\\item 580.T5\n",
       "\\item 571.T4\n",
       "\\item 576.T2\n",
       "\\item 581.T2\n",
       "\\item 580.T4\n",
       "\\item 573.T5\n",
       "\\item 600.T2\n",
       "\\item 599.T3\n",
       "\\item 584.T2\n",
       "\\item 599.T5\n",
       "\\item 602.T5\n",
       "\\item 601.T2\n",
       "\\item 583.T2\n",
       "\\item 598.T4\n",
       "\\item 589.T0\n",
       "\\item 595.T2\n",
       "\\item 571.T1\n",
       "\\item 593.T1\n",
       "\\item 589.T1\n",
       "\\item 599.T0\n",
       "\\item 571.cacum.T5\n",
       "\\item 580.T0\n",
       "\\item 572.T0\n",
       "\\item 586.T4\n",
       "\\item 592.T1\n",
       "\\item 590.T2\n",
       "\\item 583.cacum.T5\n",
       "\\item 570.T2\n",
       "\\item 574.T2\n",
       "\\item 578.T0\n",
       "\\item 586.T2\n",
       "\\item 572.cacum.T5\n",
       "\\item 588.T1\n",
       "\\item 590.T1\n",
       "\\item 593.T5\n",
       "\\item 577.T2\n",
       "\\item 581.T1\n",
       "\\item 588.T4\n",
       "\\item 574.T5\n",
       "\\item 603.T2\n",
       "\\item 595.T1\n",
       "\\item 585.cacum.T5\n",
       "\\item 582.T2\n",
       "\\item 599.T1\n",
       "\\item 588.T3\n",
       "\\item 572.T4\n",
       "\\item 582.T5\n",
       "\\item 573.T1\n",
       "\\item 593.T2\n",
       "\\item 581.T3\n",
       "\\item 584.cacum.T5\n",
       "\\item 582.cacum.T5\n",
       "\\item 592.T5\n",
       "\\item 576.T0\n",
       "\\item 602.T4\n",
       "\\item 583.T1\n",
       "\\item 574.T4\n",
       "\\item 592.T0\n",
       "\\item 573.T4\n",
       "\\item 578.T4\n",
       "\\item 590.T4\n",
       "\\item 574.cacum.T5\n",
       "\\item 591.T5\n",
       "\\item 600.T5\n",
       "\\item 595.T3\n",
       "\\item 590.T5\n",
       "\\item 585.T3\n",
       "\\item 570.T1\n",
       "\\item 575.T4\n",
       "\\item 580.T2\n",
       "\\item 596.T0\n",
       "\\item 594.T3\n",
       "\\item 600.T0\n",
       "\\item 595.T5\n",
       "\\item 589.T4\n",
       "\\item 589.T5\n",
       "\\item 596.T4\n",
       "\\item 582.T0\n",
       "\\item 593.T4\n",
       "\\item 575.T5\n",
       "\\item 583.T5\n",
       "\\item 570.cacum.T5\n",
       "\\item 570.T3\n",
       "\\item 592.T3\n",
       "\\item 572.T5\n",
       "\\item 575.T3\n",
       "\\item 603.T1\n",
       "\\item 595.T4\n",
       "\\item 602.T0\n",
       "\\item 591.T1\n",
       "\\item 594.T0\n",
       "\\item 571.T3\n",
       "\\item 581.T5\n",
       "\\item 580.cacum.T5\n",
       "\\item 580.T3\n",
       "\\item 576.T1\n",
       "\\item 570.T5\n",
       "\\item 574.T1\n",
       "\\item 576.T3\n",
       "\\item 588.T0\n",
       "\\item 584.T1\n",
       "\\item 583.T3\n",
       "\\item 602.T3\n",
       "\\item 571.T0\n",
       "\\item 593.T0\n",
       "\\item 603.T5\n",
       "\\item 573.T0\n",
       "\\item 573.cacum.T5\n",
       "\\item 603.T3\n",
       "\\item 599.T4\n",
       "\\item 570.T0\n",
       "\\item 576.T4\n",
       "\\item 597.T1\n",
       "\\item 573.T3\n",
       "\\item 594.T2\n",
       "\\item 603.T4\n",
       "\\item 591.T4\n",
       "\\item 582.T3\n",
       "\\item 581.T0\n",
       "\\item 586.T0\n",
       "\\item 597.T0\n",
       "\\item 585.T4\n",
       "\\item 586.T1\n",
       "\\item 590.T0\n",
       "\\item 596.T5\n",
       "\\item 586.cacum.T5\n",
       "\\item 583.T0\n",
       "\\item 584.T4\n",
       "\\item 601.T4\n",
       "\\item 574.T3\n",
       "\\item 585.T5\n",
       "\\item 600.T3\n",
       "\\item 598.T1\n",
       "\\item 571.T5\n",
       "\\item 578.T3\n",
       "\\item 596.T3\n",
       "\\item 586.T3\n",
       "\\item 585.T2\n",
       "\\item 576.cacum.T5\n",
       "\\item 598.T2\n",
       "\\item 597.T5\n",
       "\\item 601.T3\n",
       "\\item 597.T4\n",
       "\\item 583.T4\n",
       "\\item 582.T4\n",
       "\\item 577.T5\n",
       "\\item 591.T2\n",
       "\\item 591.T3\n",
       "\\item 595.T0\n",
       "\\item 580.T1\n",
       "\\item 600.T1\n",
       "\\item 603.T0\n",
       "\\item 577.T1\n",
       "\\item 591.T0\n",
       "\\item 572.T2\n",
       "\\item 593.T3\n",
       "\\item 594.T4\n",
       "\\item 592.T2\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item '570.cacum.T5'\n",
       "\\item '570.T0'\n",
       "\\item '570.T1'\n",
       "\\item '570.T2'\n",
       "\\item '570.T3'\n",
       "\\item '570.T4'\n",
       "\\item '570.T5'\n",
       "\\item '571.cacum.T5'\n",
       "\\item '571.T0'\n",
       "\\item '571.T1'\n",
       "\\item '571.T2'\n",
       "\\item '571.T3'\n",
       "\\item '571.T4'\n",
       "\\item '571.T5'\n",
       "\\item '572.cacum.T5'\n",
       "\\item '572.T0'\n",
       "\\item '572.T1'\n",
       "\\item '572.T2'\n",
       "\\item '572.T3'\n",
       "\\item '572.T4'\n",
       "\\item '572.T5'\n",
       "\\item '573.cacum.T5'\n",
       "\\item '573.T0'\n",
       "\\item '573.T1'\n",
       "\\item '573.T2'\n",
       "\\item '573.T3'\n",
       "\\item '573.T4'\n",
       "\\item '573.T5'\n",
       "\\item '574.cacum.T5'\n",
       "\\item '574.T0'\n",
       "\\item '574.T1'\n",
       "\\item '574.T2'\n",
       "\\item '574.T3'\n",
       "\\item '574.T4'\n",
       "\\item '574.T5'\n",
       "\\item '575.T0'\n",
       "\\item '575.T1'\n",
       "\\item '575.T2'\n",
       "\\item '575.T3'\n",
       "\\item '575.T4'\n",
       "\\item '575.T5'\n",
       "\\item '576.cacum.T5'\n",
       "\\item '576.T0'\n",
       "\\item '576.T1'\n",
       "\\item '576.T2'\n",
       "\\item '576.T3'\n",
       "\\item '576.T4'\n",
       "\\item '576.T5'\n",
       "\\item '577.cacum.T5'\n",
       "\\item '577.T0'\n",
       "\\item '577.T1'\n",
       "\\item '577.T2'\n",
       "\\item '577.T3'\n",
       "\\item '577.T4'\n",
       "\\item '577.T5'\n",
       "\\item '578.cacum.T5'\n",
       "\\item '578.T0'\n",
       "\\item '578.T1'\n",
       "\\item '578.T2'\n",
       "\\item '578.T3'\n",
       "\\item '578.T4'\n",
       "\\item '578.T5'\n",
       "\\item '580.cacum.T5'\n",
       "\\item '580.T0'\n",
       "\\item '580.T1'\n",
       "\\item '580.T2'\n",
       "\\item '580.T3'\n",
       "\\item '580.T4'\n",
       "\\item '580.T5'\n",
       "\\item '581.cacum.T5'\n",
       "\\item '581.T0'\n",
       "\\item '581.T1'\n",
       "\\item '581.T2'\n",
       "\\item '581.T3'\n",
       "\\item '581.T4'\n",
       "\\item '581.T5'\n",
       "\\item '582.cacum.T5'\n",
       "\\item '582.T0'\n",
       "\\item '582.T1'\n",
       "\\item '582.T2'\n",
       "\\item '582.T3'\n",
       "\\item '582.T4'\n",
       "\\item '582.T5'\n",
       "\\item '583.cacum.T5'\n",
       "\\item '583.T0'\n",
       "\\item '583.T1'\n",
       "\\item '583.T2'\n",
       "\\item '583.T3'\n",
       "\\item '583.T4'\n",
       "\\item '583.T5'\n",
       "\\item '584.cacum.T5'\n",
       "\\item '584.T0'\n",
       "\\item '584.T1'\n",
       "\\item '584.T2'\n",
       "\\item '584.T3'\n",
       "\\item '584.T4'\n",
       "\\item '584.T5'\n",
       "\\item '585.cacum.T5'\n",
       "\\item '585.T0'\n",
       "\\item '585.T1'\n",
       "\\item '585.T2'\n",
       "\\item '585.T3'\n",
       "\\item '585.T4'\n",
       "\\item '585.T5'\n",
       "\\item '586.cacum.T5'\n",
       "\\item '586.T0'\n",
       "\\item '586.T1'\n",
       "\\item '586.T2'\n",
       "\\item '586.T3'\n",
       "\\item '586.T4'\n",
       "\\item '586.T5'\n",
       "\\item '588.T0'\n",
       "\\item '588.T1'\n",
       "\\item '588.T2'\n",
       "\\item '588.T3'\n",
       "\\item '588.T4'\n",
       "\\item '588.T5'\n",
       "\\item '589.T0'\n",
       "\\item '589.T1'\n",
       "\\item '589.T2'\n",
       "\\item '589.T3'\n",
       "\\item '589.T4'\n",
       "\\item '589.T5'\n",
       "\\item '590.T0'\n",
       "\\item '590.T1'\n",
       "\\item '590.T2'\n",
       "\\item '590.T3'\n",
       "\\item '590.T4'\n",
       "\\item '590.T5'\n",
       "\\item '591.T0'\n",
       "\\item '591.T1'\n",
       "\\item '591.T2'\n",
       "\\item '591.T3'\n",
       "\\item '591.T4'\n",
       "\\item '591.T5'\n",
       "\\item '592.T0'\n",
       "\\item '592.T1'\n",
       "\\item '592.T2'\n",
       "\\item '592.T3'\n",
       "\\item '592.T4'\n",
       "\\item '592.T5'\n",
       "\\item '593.T0'\n",
       "\\item '593.T1'\n",
       "\\item '593.T2'\n",
       "\\item '593.T3'\n",
       "\\item '593.T4'\n",
       "\\item '593.T5'\n",
       "\\item '594.T0'\n",
       "\\item '594.T1'\n",
       "\\item '594.T2'\n",
       "\\item '594.T3'\n",
       "\\item '594.T4'\n",
       "\\item '594.T5'\n",
       "\\item '595.T0'\n",
       "\\item '595.T1'\n",
       "\\item '595.T2'\n",
       "\\item '595.T3'\n",
       "\\item '595.T4'\n",
       "\\item '595.T5'\n",
       "\\item '596.T0'\n",
       "\\item '596.T1'\n",
       "\\item '596.T2'\n",
       "\\item '596.T3'\n",
       "\\item '596.T4'\n",
       "\\item '596.T5'\n",
       "\\item '597.T0'\n",
       "\\item '597.T1'\n",
       "\\item '597.T2'\n",
       "\\item '597.T3'\n",
       "\\item '597.T4'\n",
       "\\item '597.T5'\n",
       "\\item '598.T0'\n",
       "\\item '598.T1'\n",
       "\\item '598.T2'\n",
       "\\item '598.T3'\n",
       "\\item '598.T4'\n",
       "\\item '598.T5'\n",
       "\\item '599.T0'\n",
       "\\item '599.T1'\n",
       "\\item '599.T2'\n",
       "\\item '599.T3'\n",
       "\\item '599.T4'\n",
       "\\item '599.T5'\n",
       "\\item '600.T0'\n",
       "\\item '600.T1'\n",
       "\\item '600.T2'\n",
       "\\item '600.T3'\n",
       "\\item '600.T4'\n",
       "\\item '600.T5'\n",
       "\\item '601.T0'\n",
       "\\item '601.T1'\n",
       "\\item '601.T2'\n",
       "\\item '601.T3'\n",
       "\\item '601.T4'\n",
       "\\item '601.T5'\n",
       "\\item '602.T0'\n",
       "\\item '602.T1'\n",
       "\\item '602.T2'\n",
       "\\item '602.T3'\n",
       "\\item '602.T4'\n",
       "\\item '602.T5'\n",
       "\\item '603.T0'\n",
       "\\item '603.T1'\n",
       "\\item '603.T2'\n",
       "\\item '603.T3'\n",
       "\\item '603.T4'\n",
       "\\item '603.T5'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 582.T1\n",
       "2. 586.T5\n",
       "3. 571.T2\n",
       "4. 578.cacum.T5\n",
       "5. 576.T5\n",
       "6. 581.T4\n",
       "7. 584.T5\n",
       "8. 585.T0\n",
       "9. 601.T1\n",
       "10. 600.T4\n",
       "11. 578.T5\n",
       "12. 596.T2\n",
       "13. 585.T1\n",
       "14. 578.T1\n",
       "15. 601.T5\n",
       "16. 572.T1\n",
       "17. 602.T1\n",
       "18. 577.T4\n",
       "19. 592.T4\n",
       "20. 572.T3\n",
       "21. 578.T2\n",
       "22. 577.T0\n",
       "23. 597.T3\n",
       "24. 596.T1\n",
       "25. 589.T3\n",
       "26. 588.T2\n",
       "27. 599.T2\n",
       "28. 602.T2\n",
       "29. 594.T1\n",
       "30. 597.T2\n",
       "31. 584.T3\n",
       "32. 575.T1\n",
       "33. 570.T4\n",
       "34. 577.cacum.T5\n",
       "35. 590.T3\n",
       "36. 589.T2\n",
       "37. 573.T2\n",
       "38. 588.T5\n",
       "39. 575.T2\n",
       "40. 594.T5\n",
       "41. 574.T0\n",
       "42. 598.T3\n",
       "43. 581.cacum.T5\n",
       "44. 598.T0\n",
       "45. 584.T0\n",
       "46. 575.T0\n",
       "47. 601.T0\n",
       "48. 577.T3\n",
       "49. 598.T5\n",
       "50. 580.T5\n",
       "51. 571.T4\n",
       "52. 576.T2\n",
       "53. 581.T2\n",
       "54. 580.T4\n",
       "55. 573.T5\n",
       "56. 600.T2\n",
       "57. 599.T3\n",
       "58. 584.T2\n",
       "59. 599.T5\n",
       "60. 602.T5\n",
       "61. 601.T2\n",
       "62. 583.T2\n",
       "63. 598.T4\n",
       "64. 589.T0\n",
       "65. 595.T2\n",
       "66. 571.T1\n",
       "67. 593.T1\n",
       "68. 589.T1\n",
       "69. 599.T0\n",
       "70. 571.cacum.T5\n",
       "71. 580.T0\n",
       "72. 572.T0\n",
       "73. 586.T4\n",
       "74. 592.T1\n",
       "75. 590.T2\n",
       "76. 583.cacum.T5\n",
       "77. 570.T2\n",
       "78. 574.T2\n",
       "79. 578.T0\n",
       "80. 586.T2\n",
       "81. 572.cacum.T5\n",
       "82. 588.T1\n",
       "83. 590.T1\n",
       "84. 593.T5\n",
       "85. 577.T2\n",
       "86. 581.T1\n",
       "87. 588.T4\n",
       "88. 574.T5\n",
       "89. 603.T2\n",
       "90. 595.T1\n",
       "91. 585.cacum.T5\n",
       "92. 582.T2\n",
       "93. 599.T1\n",
       "94. 588.T3\n",
       "95. 572.T4\n",
       "96. 582.T5\n",
       "97. 573.T1\n",
       "98. 593.T2\n",
       "99. 581.T3\n",
       "100. 584.cacum.T5\n",
       "101. 582.cacum.T5\n",
       "102. 592.T5\n",
       "103. 576.T0\n",
       "104. 602.T4\n",
       "105. 583.T1\n",
       "106. 574.T4\n",
       "107. 592.T0\n",
       "108. 573.T4\n",
       "109. 578.T4\n",
       "110. 590.T4\n",
       "111. 574.cacum.T5\n",
       "112. 591.T5\n",
       "113. 600.T5\n",
       "114. 595.T3\n",
       "115. 590.T5\n",
       "116. 585.T3\n",
       "117. 570.T1\n",
       "118. 575.T4\n",
       "119. 580.T2\n",
       "120. 596.T0\n",
       "121. 594.T3\n",
       "122. 600.T0\n",
       "123. 595.T5\n",
       "124. 589.T4\n",
       "125. 589.T5\n",
       "126. 596.T4\n",
       "127. 582.T0\n",
       "128. 593.T4\n",
       "129. 575.T5\n",
       "130. 583.T5\n",
       "131. 570.cacum.T5\n",
       "132. 570.T3\n",
       "133. 592.T3\n",
       "134. 572.T5\n",
       "135. 575.T3\n",
       "136. 603.T1\n",
       "137. 595.T4\n",
       "138. 602.T0\n",
       "139. 591.T1\n",
       "140. 594.T0\n",
       "141. 571.T3\n",
       "142. 581.T5\n",
       "143. 580.cacum.T5\n",
       "144. 580.T3\n",
       "145. 576.T1\n",
       "146. 570.T5\n",
       "147. 574.T1\n",
       "148. 576.T3\n",
       "149. 588.T0\n",
       "150. 584.T1\n",
       "151. 583.T3\n",
       "152. 602.T3\n",
       "153. 571.T0\n",
       "154. 593.T0\n",
       "155. 603.T5\n",
       "156. 573.T0\n",
       "157. 573.cacum.T5\n",
       "158. 603.T3\n",
       "159. 599.T4\n",
       "160. 570.T0\n",
       "161. 576.T4\n",
       "162. 597.T1\n",
       "163. 573.T3\n",
       "164. 594.T2\n",
       "165. 603.T4\n",
       "166. 591.T4\n",
       "167. 582.T3\n",
       "168. 581.T0\n",
       "169. 586.T0\n",
       "170. 597.T0\n",
       "171. 585.T4\n",
       "172. 586.T1\n",
       "173. 590.T0\n",
       "174. 596.T5\n",
       "175. 586.cacum.T5\n",
       "176. 583.T0\n",
       "177. 584.T4\n",
       "178. 601.T4\n",
       "179. 574.T3\n",
       "180. 585.T5\n",
       "181. 600.T3\n",
       "182. 598.T1\n",
       "183. 571.T5\n",
       "184. 578.T3\n",
       "185. 596.T3\n",
       "186. 586.T3\n",
       "187. 585.T2\n",
       "188. 576.cacum.T5\n",
       "189. 598.T2\n",
       "190. 597.T5\n",
       "191. 601.T3\n",
       "192. 597.T4\n",
       "193. 583.T4\n",
       "194. 582.T4\n",
       "195. 577.T5\n",
       "196. 591.T2\n",
       "197. 591.T3\n",
       "198. 595.T0\n",
       "199. 580.T1\n",
       "200. 600.T1\n",
       "201. 603.T0\n",
       "202. 577.T1\n",
       "203. 591.T0\n",
       "204. 572.T2\n",
       "205. 593.T3\n",
       "206. 594.T4\n",
       "207. 592.T2\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. '570.cacum.T5'\n",
       "2. '570.T0'\n",
       "3. '570.T1'\n",
       "4. '570.T2'\n",
       "5. '570.T3'\n",
       "6. '570.T4'\n",
       "7. '570.T5'\n",
       "8. '571.cacum.T5'\n",
       "9. '571.T0'\n",
       "10. '571.T1'\n",
       "11. '571.T2'\n",
       "12. '571.T3'\n",
       "13. '571.T4'\n",
       "14. '571.T5'\n",
       "15. '572.cacum.T5'\n",
       "16. '572.T0'\n",
       "17. '572.T1'\n",
       "18. '572.T2'\n",
       "19. '572.T3'\n",
       "20. '572.T4'\n",
       "21. '572.T5'\n",
       "22. '573.cacum.T5'\n",
       "23. '573.T0'\n",
       "24. '573.T1'\n",
       "25. '573.T2'\n",
       "26. '573.T3'\n",
       "27. '573.T4'\n",
       "28. '573.T5'\n",
       "29. '574.cacum.T5'\n",
       "30. '574.T0'\n",
       "31. '574.T1'\n",
       "32. '574.T2'\n",
       "33. '574.T3'\n",
       "34. '574.T4'\n",
       "35. '574.T5'\n",
       "36. '575.T0'\n",
       "37. '575.T1'\n",
       "38. '575.T2'\n",
       "39. '575.T3'\n",
       "40. '575.T4'\n",
       "41. '575.T5'\n",
       "42. '576.cacum.T5'\n",
       "43. '576.T0'\n",
       "44. '576.T1'\n",
       "45. '576.T2'\n",
       "46. '576.T3'\n",
       "47. '576.T4'\n",
       "48. '576.T5'\n",
       "49. '577.cacum.T5'\n",
       "50. '577.T0'\n",
       "51. '577.T1'\n",
       "52. '577.T2'\n",
       "53. '577.T3'\n",
       "54. '577.T4'\n",
       "55. '577.T5'\n",
       "56. '578.cacum.T5'\n",
       "57. '578.T0'\n",
       "58. '578.T1'\n",
       "59. '578.T2'\n",
       "60. '578.T3'\n",
       "61. '578.T4'\n",
       "62. '578.T5'\n",
       "63. '580.cacum.T5'\n",
       "64. '580.T0'\n",
       "65. '580.T1'\n",
       "66. '580.T2'\n",
       "67. '580.T3'\n",
       "68. '580.T4'\n",
       "69. '580.T5'\n",
       "70. '581.cacum.T5'\n",
       "71. '581.T0'\n",
       "72. '581.T1'\n",
       "73. '581.T2'\n",
       "74. '581.T3'\n",
       "75. '581.T4'\n",
       "76. '581.T5'\n",
       "77. '582.cacum.T5'\n",
       "78. '582.T0'\n",
       "79. '582.T1'\n",
       "80. '582.T2'\n",
       "81. '582.T3'\n",
       "82. '582.T4'\n",
       "83. '582.T5'\n",
       "84. '583.cacum.T5'\n",
       "85. '583.T0'\n",
       "86. '583.T1'\n",
       "87. '583.T2'\n",
       "88. '583.T3'\n",
       "89. '583.T4'\n",
       "90. '583.T5'\n",
       "91. '584.cacum.T5'\n",
       "92. '584.T0'\n",
       "93. '584.T1'\n",
       "94. '584.T2'\n",
       "95. '584.T3'\n",
       "96. '584.T4'\n",
       "97. '584.T5'\n",
       "98. '585.cacum.T5'\n",
       "99. '585.T0'\n",
       "100. '585.T1'\n",
       "101. '585.T2'\n",
       "102. '585.T3'\n",
       "103. '585.T4'\n",
       "104. '585.T5'\n",
       "105. '586.cacum.T5'\n",
       "106. '586.T0'\n",
       "107. '586.T1'\n",
       "108. '586.T2'\n",
       "109. '586.T3'\n",
       "110. '586.T4'\n",
       "111. '586.T5'\n",
       "112. '588.T0'\n",
       "113. '588.T1'\n",
       "114. '588.T2'\n",
       "115. '588.T3'\n",
       "116. '588.T4'\n",
       "117. '588.T5'\n",
       "118. '589.T0'\n",
       "119. '589.T1'\n",
       "120. '589.T2'\n",
       "121. '589.T3'\n",
       "122. '589.T4'\n",
       "123. '589.T5'\n",
       "124. '590.T0'\n",
       "125. '590.T1'\n",
       "126. '590.T2'\n",
       "127. '590.T3'\n",
       "128. '590.T4'\n",
       "129. '590.T5'\n",
       "130. '591.T0'\n",
       "131. '591.T1'\n",
       "132. '591.T2'\n",
       "133. '591.T3'\n",
       "134. '591.T4'\n",
       "135. '591.T5'\n",
       "136. '592.T0'\n",
       "137. '592.T1'\n",
       "138. '592.T2'\n",
       "139. '592.T3'\n",
       "140. '592.T4'\n",
       "141. '592.T5'\n",
       "142. '593.T0'\n",
       "143. '593.T1'\n",
       "144. '593.T2'\n",
       "145. '593.T3'\n",
       "146. '593.T4'\n",
       "147. '593.T5'\n",
       "148. '594.T0'\n",
       "149. '594.T1'\n",
       "150. '594.T2'\n",
       "151. '594.T3'\n",
       "152. '594.T4'\n",
       "153. '594.T5'\n",
       "154. '595.T0'\n",
       "155. '595.T1'\n",
       "156. '595.T2'\n",
       "157. '595.T3'\n",
       "158. '595.T4'\n",
       "159. '595.T5'\n",
       "160. '596.T0'\n",
       "161. '596.T1'\n",
       "162. '596.T2'\n",
       "163. '596.T3'\n",
       "164. '596.T4'\n",
       "165. '596.T5'\n",
       "166. '597.T0'\n",
       "167. '597.T1'\n",
       "168. '597.T2'\n",
       "169. '597.T3'\n",
       "170. '597.T4'\n",
       "171. '597.T5'\n",
       "172. '598.T0'\n",
       "173. '598.T1'\n",
       "174. '598.T2'\n",
       "175. '598.T3'\n",
       "176. '598.T4'\n",
       "177. '598.T5'\n",
       "178. '599.T0'\n",
       "179. '599.T1'\n",
       "180. '599.T2'\n",
       "181. '599.T3'\n",
       "182. '599.T4'\n",
       "183. '599.T5'\n",
       "184. '600.T0'\n",
       "185. '600.T1'\n",
       "186. '600.T2'\n",
       "187. '600.T3'\n",
       "188. '600.T4'\n",
       "189. '600.T5'\n",
       "190. '601.T0'\n",
       "191. '601.T1'\n",
       "192. '601.T2'\n",
       "193. '601.T3'\n",
       "194. '601.T4'\n",
       "195. '601.T5'\n",
       "196. '602.T0'\n",
       "197. '602.T1'\n",
       "198. '602.T2'\n",
       "199. '602.T3'\n",
       "200. '602.T4'\n",
       "201. '602.T5'\n",
       "202. '603.T0'\n",
       "203. '603.T1'\n",
       "204. '603.T2'\n",
       "205. '603.T3'\n",
       "206. '603.T4'\n",
       "207. '603.T5'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] 582.T1       586.T5       571.T2       578.cacum.T5 576.T5      \n",
       "  [6] 581.T4       584.T5       585.T0       601.T1       600.T4      \n",
       " [11] 578.T5       596.T2       585.T1       578.T1       601.T5      \n",
       " [16] 572.T1       602.T1       577.T4       592.T4       572.T3      \n",
       " [21] 578.T2       577.T0       597.T3       596.T1       589.T3      \n",
       " [26] 588.T2       599.T2       602.T2       594.T1       597.T2      \n",
       " [31] 584.T3       575.T1       570.T4       577.cacum.T5 590.T3      \n",
       " [36] 589.T2       573.T2       588.T5       575.T2       594.T5      \n",
       " [41] 574.T0       598.T3       581.cacum.T5 598.T0       584.T0      \n",
       " [46] 575.T0       601.T0       577.T3       598.T5       580.T5      \n",
       " [51] 571.T4       576.T2       581.T2       580.T4       573.T5      \n",
       " [56] 600.T2       599.T3       584.T2       599.T5       602.T5      \n",
       " [61] 601.T2       583.T2       598.T4       589.T0       595.T2      \n",
       " [66] 571.T1       593.T1       589.T1       599.T0       571.cacum.T5\n",
       " [71] 580.T0       572.T0       586.T4       592.T1       590.T2      \n",
       " [76] 583.cacum.T5 570.T2       574.T2       578.T0       586.T2      \n",
       " [81] 572.cacum.T5 588.T1       590.T1       593.T5       577.T2      \n",
       " [86] 581.T1       588.T4       574.T5       603.T2       595.T1      \n",
       " [91] 585.cacum.T5 582.T2       599.T1       588.T3       572.T4      \n",
       " [96] 582.T5       573.T1       593.T2       581.T3       584.cacum.T5\n",
       "[101] 582.cacum.T5 592.T5       576.T0       602.T4       583.T1      \n",
       "[106] 574.T4       592.T0       573.T4       578.T4       590.T4      \n",
       "[111] 574.cacum.T5 591.T5       600.T5       595.T3       590.T5      \n",
       "[116] 585.T3       570.T1       575.T4       580.T2       596.T0      \n",
       "[121] 594.T3       600.T0       595.T5       589.T4       589.T5      \n",
       "[126] 596.T4       582.T0       593.T4       575.T5       583.T5      \n",
       "[131] 570.cacum.T5 570.T3       592.T3       572.T5       575.T3      \n",
       "[136] 603.T1       595.T4       602.T0       591.T1       594.T0      \n",
       "[141] 571.T3       581.T5       580.cacum.T5 580.T3       576.T1      \n",
       "[146] 570.T5       574.T1       576.T3       588.T0       584.T1      \n",
       "[151] 583.T3       602.T3       571.T0       593.T0       603.T5      \n",
       "[156] 573.T0       573.cacum.T5 603.T3       599.T4       570.T0      \n",
       "[161] 576.T4       597.T1       573.T3       594.T2       603.T4      \n",
       "[166] 591.T4       582.T3       581.T0       586.T0       597.T0      \n",
       "[171] 585.T4       586.T1       590.T0       596.T5       586.cacum.T5\n",
       "[176] 583.T0       584.T4       601.T4       574.T3       585.T5      \n",
       "[181] 600.T3       598.T1       571.T5       578.T3       596.T3      \n",
       "[186] 586.T3       585.T2       576.cacum.T5 598.T2       597.T5      \n",
       "[191] 601.T3       597.T4       583.T4       582.T4       577.T5      \n",
       "[196] 591.T2       591.T3       595.T0       580.T1       600.T1      \n",
       "[201] 603.T0       577.T1       591.T0       572.T2       593.T3      \n",
       "[206] 594.T4       592.T2      \n",
       "207 Levels: 570.cacum.T5 570.T0 570.T1 570.T2 570.T3 570.T4 ... 603.T5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Make sure the data has the sample names on rows if not just transform the data with t()\n",
    "abund_table<-read.csv(\"Known_Cutoff_no_BA.csv\",row.names=1,check.names=FALSE)\n",
    "meta_table<-read.csv(\"Mapping_file_w_og.csv\",row.names=1,check.names=FALSE)\n",
    "#Transpose the data to have sample names on rows\n",
    "abund_table<-t(abund_table)\n",
    "\n",
    "meta_table<-data.frame(meta_table)\n",
    "#Filter out samples not present in meta_table\n",
    "abund_table<-abund_table[rownames(abund_table) %in% rownames(meta_table),]\n",
    "\n",
    "#Now load the taxonomy\n",
    "OTU_taxonomy<-read.csv(\"taxonomy_cutoff_no_BA.csv\",row.names=1,check.names=FALSE)\n",
    "#Convert to phyloseq\n",
    "library(phyloseq)\n",
    "#Convert the data to phyloseq format\n",
    "OTU = otu_table(as.matrix(abund_table), taxa_are_rows = FALSE)\n",
    "TAX = tax_table(as.matrix(OTU_taxonomy))\n",
    "SAM = sample_data(meta_table)\n",
    "physeq<-merge_phyloseq(phyloseq(OTU, TAX),SAM)\n",
    "# Removed Blank\n",
    "physeq = subset_samples(physeq, ID_OG_MAP != 'Blank_injection_GE1_01_33205.mzXML')\n",
    "sample_data(physeq)$ID_OG_MAP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. \n",
       "  18196 1179795 1683833 1744231 2087042 7759084 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Number of Samples in our Biom Table\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "207"
      ],
      "text/latex": [
       "207"
      ],
      "text/markdown": [
       "207"
      ],
      "text/plain": [
       "[1] 207"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Number of Sequence variants we have.\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "131"
      ],
      "text/latex": [
       "131"
      ],
      "text/markdown": [
       "131"
      ],
      "text/plain": [
       "[1] 131"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Sequencing depth.\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. \n",
       "  18196 1179795 1683833 1744231 2087042 7759084 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<dl class=dl-horizontal>\n",
       "\t<dt>570.T0_RA1_01_33057</dt>\n",
       "\t\t<dd>18196</dd>\n",
       "\t<dt>583.CT5_BB3_01_33413</dt>\n",
       "\t\t<dd>311937</dd>\n",
       "\t<dt>595.T1_RE8_01_33117</dt>\n",
       "\t\t<dd>528633</dd>\n",
       "\t<dt>572.CT5_BA6_01_33403</dt>\n",
       "\t\t<dd>531703</dd>\n",
       "\t<dt>583.T3_BB1_01_33180</dt>\n",
       "\t\t<dd>577765</dd>\n",
       "\t<dt>580.T5_BG2_01_33372</dt>\n",
       "\t\t<dd>662636</dd>\n",
       "\t<dt>578.T4_BD5_01_33210</dt>\n",
       "\t\t<dd>668860</dd>\n",
       "\t<dt>581.T3_BA11_01_33177</dt>\n",
       "\t\t<dd>695223</dd>\n",
       "\t<dt>584.CT5_BB4_01_33414</dt>\n",
       "\t\t<dd>706019</dd>\n",
       "\t<dt>585.CT5_BB5_01_33415</dt>\n",
       "\t\t<dd>711040</dd>\n",
       "\t<dt>580C.T5_RA12_01_33409</dt>\n",
       "\t\t<dd>726698</dd>\n",
       "\t<dt>580.T3_BA10_01_33176</dt>\n",
       "\t\t<dd>733067</dd>\n",
       "\t<dt>583.T5_BG5_01_33375</dt>\n",
       "\t\t<dd>747635</dd>\n",
       "\t<dt>577.T5_BF12_01_33369</dt>\n",
       "\t\t<dd>761863</dd>\n",
       "\t<dt>570.CT5_BA4_01_33401</dt>\n",
       "\t\t<dd>787510</dd>\n",
       "\t<dt>576.T1_RD3_01_33099</dt>\n",
       "\t\t<dd>799268</dd>\n",
       "\t<dt>578.T3_BA9_01_33175</dt>\n",
       "\t\t<dd>816534</dd>\n",
       "\t<dt>571.CT5_BA5_01_33402</dt>\n",
       "\t\t<dd>823569</dd>\n",
       "\t<dt>573.T4_BC12_01_33204</dt>\n",
       "\t\t<dd>857115</dd>\n",
       "\t<dt>585.T3_BB3_01_33182</dt>\n",
       "\t\t<dd>887824</dd>\n",
       "\t<dt>582.T5_BG4_01_33374</dt>\n",
       "\t\t<dd>892233</dd>\n",
       "\t<dt>582.CT5_BB2_01_33412</dt>\n",
       "\t\t<dd>910028</dd>\n",
       "\t<dt>581.T2_RG3_01_33141</dt>\n",
       "\t\t<dd>919118</dd>\n",
       "\t<dt>575.T2_RF10_01_33135</dt>\n",
       "\t\t<dd>933917</dd>\n",
       "\t<dt>593.T1_RE6_01_33115</dt>\n",
       "\t\t<dd>944624</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[570.T0\\textbackslash{}\\_RA1\\textbackslash{}\\_01\\textbackslash{}\\_33057] 18196\n",
       "\\item[583.CT5\\textbackslash{}\\_BB3\\textbackslash{}\\_01\\textbackslash{}\\_33413] 311937\n",
       "\\item[595.T1\\textbackslash{}\\_RE8\\textbackslash{}\\_01\\textbackslash{}\\_33117] 528633\n",
       "\\item[572.CT5\\textbackslash{}\\_BA6\\textbackslash{}\\_01\\textbackslash{}\\_33403] 531703\n",
       "\\item[583.T3\\textbackslash{}\\_BB1\\textbackslash{}\\_01\\textbackslash{}\\_33180] 577765\n",
       "\\item[580.T5\\textbackslash{}\\_BG2\\textbackslash{}\\_01\\textbackslash{}\\_33372] 662636\n",
       "\\item[578.T4\\textbackslash{}\\_BD5\\textbackslash{}\\_01\\textbackslash{}\\_33210] 668860\n",
       "\\item[581.T3\\textbackslash{}\\_BA11\\textbackslash{}\\_01\\textbackslash{}\\_33177] 695223\n",
       "\\item[584.CT5\\textbackslash{}\\_BB4\\textbackslash{}\\_01\\textbackslash{}\\_33414] 706019\n",
       "\\item[585.CT5\\textbackslash{}\\_BB5\\textbackslash{}\\_01\\textbackslash{}\\_33415] 711040\n",
       "\\item[580C.T5\\textbackslash{}\\_RA12\\textbackslash{}\\_01\\textbackslash{}\\_33409] 726698\n",
       "\\item[580.T3\\textbackslash{}\\_BA10\\textbackslash{}\\_01\\textbackslash{}\\_33176] 733067\n",
       "\\item[583.T5\\textbackslash{}\\_BG5\\textbackslash{}\\_01\\textbackslash{}\\_33375] 747635\n",
       "\\item[577.T5\\textbackslash{}\\_BF12\\textbackslash{}\\_01\\textbackslash{}\\_33369] 761863\n",
       "\\item[570.CT5\\textbackslash{}\\_BA4\\textbackslash{}\\_01\\textbackslash{}\\_33401] 787510\n",
       "\\item[576.T1\\textbackslash{}\\_RD3\\textbackslash{}\\_01\\textbackslash{}\\_33099] 799268\n",
       "\\item[578.T3\\textbackslash{}\\_BA9\\textbackslash{}\\_01\\textbackslash{}\\_33175] 816534\n",
       "\\item[571.CT5\\textbackslash{}\\_BA5\\textbackslash{}\\_01\\textbackslash{}\\_33402] 823569\n",
       "\\item[573.T4\\textbackslash{}\\_BC12\\textbackslash{}\\_01\\textbackslash{}\\_33204] 857115\n",
       "\\item[585.T3\\textbackslash{}\\_BB3\\textbackslash{}\\_01\\textbackslash{}\\_33182] 887824\n",
       "\\item[582.T5\\textbackslash{}\\_BG4\\textbackslash{}\\_01\\textbackslash{}\\_33374] 892233\n",
       "\\item[582.CT5\\textbackslash{}\\_BB2\\textbackslash{}\\_01\\textbackslash{}\\_33412] 910028\n",
       "\\item[581.T2\\textbackslash{}\\_RG3\\textbackslash{}\\_01\\textbackslash{}\\_33141] 919118\n",
       "\\item[575.T2\\textbackslash{}\\_RF10\\textbackslash{}\\_01\\textbackslash{}\\_33135] 933917\n",
       "\\item[593.T1\\textbackslash{}\\_RE6\\textbackslash{}\\_01\\textbackslash{}\\_33115] 944624\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "570.T0_RA1_01_33057\n",
       ":   18196583.CT5_BB3_01_33413\n",
       ":   311937595.T1_RE8_01_33117\n",
       ":   528633572.CT5_BA6_01_33403\n",
       ":   531703583.T3_BB1_01_33180\n",
       ":   577765580.T5_BG2_01_33372\n",
       ":   662636578.T4_BD5_01_33210\n",
       ":   668860581.T3_BA11_01_33177\n",
       ":   695223584.CT5_BB4_01_33414\n",
       ":   706019585.CT5_BB5_01_33415\n",
       ":   711040580C.T5_RA12_01_33409\n",
       ":   726698580.T3_BA10_01_33176\n",
       ":   733067583.T5_BG5_01_33375\n",
       ":   747635577.T5_BF12_01_33369\n",
       ":   761863570.CT5_BA4_01_33401\n",
       ":   787510576.T1_RD3_01_33099\n",
       ":   799268578.T3_BA9_01_33175\n",
       ":   816534571.CT5_BA5_01_33402\n",
       ":   823569573.T4_BC12_01_33204\n",
       ":   857115585.T3_BB3_01_33182\n",
       ":   887824582.T5_BG4_01_33374\n",
       ":   892233582.CT5_BB2_01_33412\n",
       ":   910028581.T2_RG3_01_33141\n",
       ":   919118575.T2_RF10_01_33135\n",
       ":   933917593.T1_RE6_01_33115\n",
       ":   944624\n",
       "\n"
      ],
      "text/plain": [
       "  570.T0_RA1_01_33057  583.CT5_BB3_01_33413   595.T1_RE8_01_33117 \n",
       "                18196                311937                528633 \n",
       " 572.CT5_BA6_01_33403   583.T3_BB1_01_33180   580.T5_BG2_01_33372 \n",
       "               531703                577765                662636 \n",
       "  578.T4_BD5_01_33210  581.T3_BA11_01_33177  584.CT5_BB4_01_33414 \n",
       "               668860                695223                706019 \n",
       " 585.CT5_BB5_01_33415 580C.T5_RA12_01_33409  580.T3_BA10_01_33176 \n",
       "               711040                726698                733067 \n",
       "  583.T5_BG5_01_33375  577.T5_BF12_01_33369  570.CT5_BA4_01_33401 \n",
       "               747635                761863                787510 \n",
       "  576.T1_RD3_01_33099   578.T3_BA9_01_33175  571.CT5_BA5_01_33402 \n",
       "               799268                816534                823569 \n",
       " 573.T4_BC12_01_33204   585.T3_BB3_01_33182   582.T5_BG4_01_33374 \n",
       "               857115                887824                892233 \n",
       " 582.CT5_BB2_01_33412   581.T2_RG3_01_33141  575.T2_RF10_01_33135 \n",
       "               910028                919118                933917 \n",
       "  593.T1_RE6_01_33115 \n",
       "               944624 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAC+lBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaIiIiKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWW\nlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eo\nqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6\nurq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vM\nzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e\n3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w\n8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///899WbvAAAA\nCXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3de3wV5Z348W8ISQBvIBc1qJiIoFXbuioI\nFVGCl4qSmgS5im0Vr21VQGx10Vovpa1rbXdr17bSrfa37u/3W1d6AXVdb12tXbbVuoIiVxGx\nUgFBLglJ5vXamTnnJJOQMznOec6TOd/5vP9I5iRPvvPEzWfPJVMiDoC8SU9vANCAkAADCAkw\ngJAAAwgJMICQAAMICTCAkAADCAkwgJAAAwgJMICQAAMICTCAkAADCAkwgJAAAwgJMICQAAMI\nCTCAkAADCAkwgJAAAwgJMICQAAMICTCAkAADCAkwgJAAAwgJMICQAAMICTCAkAADCAkwgJAA\nAwgJMICQAAMICTCAkAADCAkwgJAMeVFEXkwdVojc4ThL3Q/s23/dhi02dvPz0w444PgfdPrg\nSndHb3f7pf4Gvc0XZGNq8Z/LkNxC2nNXvz9Z2MxPxfPNTh/NJaT0Bgnpk+I/lyE5hfRhtYiN\nkM4WOXDW5Oc6fTSHkDIbJKRPiv9chuwX0t6NGzd2XrRO7IQ03N9BZzmElNkgIX1S/OcyZL+Q\numIrpCNFfrT/RwmpgPjPZUjWh3abbvjUAaWHnvlAk+PM9J+6yN+5H21+5LzK8sopmUdfr9YP\n7HvqYucAkXWO8wORma+dXl75kLPvgbH9S/seN8e7a7tN5La/fu2oipHfa3V+dkqfITM3BE8f\nnHd+6jTHtn/2D7WH9h31y7aQXp95RNlhtU96hw+LTGq8Z2Sfo77ivcbQtkE/pEdH9zv0klf9\nAYFvAl0iJEOyhbRyUOqHU2qbAiG9Pz790euava/41/LUjbaQJh7m3lzWMi29asDbfkjTjvRv\nfXW2/+6ID9vP3mHefiE92tv/wMXpkB4tSy24yfFDqjnTv3X0mk4hXe0f913udPwm0CVCMsQL\nqbzCJ8GQLhA5/Z77Z5eI/NT5/ufdj02s/7XT4v3cDz7d7UYWuF/7zoHu19ZO7e/9qK7zQ/L0\nb3pMpOTCa89zv/YaPySRc+f4P9KDvjzaffuttpN3nLdoZj+RUTPnZT67uq9I2eTph0oqpD+6\n1Y68YoJ742E/JHfcRSe4b0e1tm/Q27wcN2ei+3aS0/GbQJcIyZAXJag9JLeRFe6nvzXhxmXt\nT0F+4b6/rcXZfpFIL/ezt4iU/s5x1h7VHtJp6z74jfO9T5ctdFdfK3JmKqTvOc4z3n3RZqdl\nlHsP03byTvOcoR2eI93ojnfvLDccnQrpC24c7sZ+6JbS6oc0ZqvTOt99/2TH50jnu/c+V7k9\nO52+CXSFkAzJFpL7M105b9mO1KLMz6l7d3CGd/uvh4h8w3H+RmSqd/NH7SE9k/qCpkbH2flV\nkc/4IR3kztvhfvLr7mfcD45rO3mneZ1COklklvf+B35IjX1Elrq3mt37qVV+SK+4t/YeITK3\nY0jeA9V/7eKbQFcIyZBsz5Fu98MqHfPdbU77z+kAkXv8pe5DqXMd5yCR73i3XmsPaW9q1AeP\nXn+694zmZD8kNyenxb31T+77b4iMbjt5p3mdQnIf6N3vvX/FD2lFoPd/80Iqa/E+OVnkoo4h\nbXff/7v7flenbwJdISRDsoXUfHP6qf3AZ9t+TltK/OcnrivcJx9Oq6RvrmkL6RD/s7uu8F4l\nKD0ic4801vug+5HHHf9mW0id5nUKyRv/iHewwg/p94GQ/skL6TB/lfsg7uyOIXl9Peu+39np\nm0BXCMmQ7Fc2bHlwsv8yQmVz28/p4Mw9iPss/nz/LuM+79YLbSFV+p91s+h/81M7/kHkFL+c\nz3kfdNf8yukYUud5ne6R3Gc4D3jv/8MP6c/u2+d3puzzQir3dzlNZPL+v0dKh9Thm0BXCMmQ\nrPdIq9zn5y2vXeneesv/Of2ju2J84DnN7Y5zisgXvJt/2xbS0d7N3e7dwBPu+2+LnBoeUqd5\nnUI6Lf0U7Jt+SLvce7kfu7e2/fP/NKVetXvWvdVa5b/gl9lgp5A6fBPoCiEZkiWkfy9PvXDw\nB/fWu84m9+1Tzj7nwdSrbB9dJNI79SuiEvfB17MHtoU0zJvzjvgh7f5s5qFd1pA6zesU0kL3\n4eFvHeeP/VOv2p0nUr3Fz2rADj+kk951nLtTQWU22DGkjt8EukJIhmQJqekEkQOv+97NR/gv\nYe8pdZ/+HHeXs/dv3E8OHuWFs9D9gs0HuQefOrmXdAxp36EiQxZ9d6T70erwkDrN6xTS5gEi\nvc46x/v9lhfSC+4zqqFfmdbLf/XP/z3SweedJKnpmQ12DKnjN4GuEJIh2R7avXl0+on9sDXu\np87xjq5272zOSH/0xlbvK57wLz3o5T20W98ekvP3qTXuhLLdoSF1ntcxJOfJvv7nbhmYurLh\nh71Sa7+wzw+p/2T/VpV/yVF6g50e2nX8JtAFQjIk64sNH90z+pDSg0+703s12dlYf3Df4//B\nPdj38/OPKK+cmv4K55VJ/Q885/nX3S/ZHAjJ+edRfSqOv3WH+2P8WHhIneZ1Csl5o+HQfp/7\n/046JGf5rKMrhox7xHvdwA1pYPN3jis/6obUBUfpDXZ+saHDN4EuEFIMvLvZvx/xnoOU7LV7\nai8ku2dUipBi4CYpO+Y/3Pd3iBxn+dSEZAghxcDj7gOo/jOvGOO+u9PyqQnJEEKKg0vTz+Vl\n9G7LZyYkQwgpDlofmXRkRe9BZ/+w0faZCckQQgIMICTAAEICDCAkwABCAgwgJMAAQgIMICTA\nAEICDCAkwABCAgwgJMAAQgIMICTAAEICDCAkwABCAgwgJMAAQgIMICTAAEICDCAkwABCAgwg\nJMAAQgIMICTAAEICDCAkwABCAgwgJMAAQgIMICTAAEICDCAkwABCAgwgJMAAQgIMyCukretW\nrN5maidAEYse0tKGQeIZWPeMwf0ARSlqSE0NItU1ddPraqpFZjcZ3RNQdKKGdKdMXZ8+XD9N\nFpnaDlCcooZUNaql7bh1zEgzmwGKVdSQKuYHbizoY2IrQPGKGtKIce3HrWOrTGwFKF5RQ7pX\nLlubPtw4SxYa2g1QpKKG1DhTZFhN/YyGicNFavca3RNQdKL/HmnZJYf7v0caUr/E4H6AopTX\nlQ3bN6xYy5UNQF4hte7LHO2kJiRc5JDevKii5DMPt/rH47n0FQkXNYHV/aX6xFKZ/LF3g5CQ\ndFETmCGL3Xulc2XsToeQgKgJDL7Qe9tytdQ0dhXSpjXt3slrg0AxiBpS75v8d61fkmmt+4e0\nWgJK9u3/5YAuUUMaOiH1ft+FMreLe6R32u+Q/q805rE/oChEDWmO3Je6/HvXaJkzOmzKfxIS\n9Isa0qZKGfKwf7R1jPvwLWQlISEBIr/etunKgT9NHe25qTweIb33tGHvWdo4il8+L1w3Zw42\nPx6yyl5IV5QPMKr8CksbR/Er/G+A7IX0xS/Gex4UIyR786AYIdmbB8WihvSdjkJWEhISIGpI\nA6WDkJWEhASIGtKW8XLu4+1CVhISEiDyc6Q9Y0qezWkhISEBor/Y8F7/kTldjUpISIA8XrV7\n8MSncllGSEgAXv62Nw+KEZK9eVCMkOzNg2KEZG8eFCMke/OgGCHZmwfFCMnePChGSPbmQTFC\nsjcPihGSvXlQjJDszYNihGRvHhQjJHvzoBgh2ZsHxQjJ3jwoRkj25kExQrI3D4oRkr15UIyQ\n7M2DYoRkbx4UIyR786AYIdmbB8UIyd48KEZI9uZBMUKyNw+KEZK9eVCMkOzNg2KEZG8eFCMk\ne/OgGCHZmwfFCMnePChGSPbmQTFCsjcPihGSvXlQjJDszYNihGRvHhQjJHvzoBgh2ZsHxQjJ\n3jwoRkj25kExQrI3D4oRkr15UIyQ7M2DYoRkbx4UIyR786AYIdmbB8UIyd48KEZI9uZBMUKy\nNw+KEZK9eVCMkOzNg2KEZG8eFCMke/OgGCHZmwfFCMnePChGSPbmQTFCsjcPihGSvXlQjJDs\nzYNihGRvHhQjJHvzoBgh2ZsHxQjJ3jwoRkj25kExQrI3D4rlFdLWdStWb+tuESEhAaKHtLRh\nkHgG1j0Tuo6QkABRQ2pqEKmuqZteV1MtMrspZCUhIQGihnSnTF2fPlw/TRaFrCQkJEDUkKpG\ntbQdt44ZGbKSkJAAUUOqmB+4saBPyEpCQgJEDWnEuPbj1rFVISsJCQkQNaR75bK16cONs2Rh\nyEpCQgJEDalxpsiwmvoZDROHi9TuDVlJSEiA6L9HWnbJ4f7vkYbULwldR0hIgLyubNi+YcVa\nrmwAuETI5jwoxiVC9uZBMS4RsjcPinGJkL15UIxLhOzNg2KFuURo8wUT25xOSNCvMJcIfXzn\nLW1mERL04xIhe/OgGJcI2ZsHxbhEyN48KMYlQvbmQTH+OS5786AYIdmbB8UIyd48KBY1pO90\nFLKSkJAAUUMaKB2ErCQkJEDUkLaMl3MfbxeykpCQAJGfI+0ZU/JsTgsJCQkQ/cWG9/qP3JfL\nOkJCAuTxqt2DJz6VyzJCQgLw8re9eVCMkOzNg2KEZG8eFCMke/OgGCHZmwfFCMnePChGSPbm\nQTFCsjcPihGSvXlQjJDszYNihGRvHhQjJHvzoBgh2ZsHxQjJ3jwoRkj25kExQrI3D4oRkr15\nUIyQ7M2DYoRkbx4UIyR786AYIdmbB8UIyd48KEZI9uZBMUKyNw+KEZK9eVCMkOzNg2KEZG8e\nFCMke/OgGCHZmwfFCMnePChGSPbmQTFCsjcPihGSvXlQjJDszYNihGRvHhQjJHvzoBgh2ZsH\nxQjJ3jwoRkj25kExQrI3D4oRkr15UIyQ7M2DYoRkbx4UIyR786AYIdmbB8UIyd48KEZI9uZB\nMUKyNw+KEZK9eVCMkOzNg2KEZG8eFCMke/OgGCHZmwfFCMnePChGSPbmQTFCsjcPihGSvXlQ\njJDszYNihGRvHhQjJHvzoBgh2ZsHxQjJ3jwoRkj25kExQrI3D4oRkr15UIyQ7M2DYoRkbx4U\nIyR786AYIdmbB8UIyd48KEZI9uZBsbxC2rpuxept3S0iJCRA9JCWNgwSz8C6Z0LXERISIGpI\nTQ0i1TV10+tqqkVmN4WsJCQkQNSQ7pSp69OH66fJopCVhIQEiBpS1aiWtuPWMSNDVhISEiBq\nSBXzAzcW9AlZSUhIgKghjRjXftw6tipkJSEhAaKGdK9ctjZ9uHGWLAxZSUhIgKghNc4UGVZT\nP6Nh4nCR2r0hKwkJCRD990jLLjnc/z3SkPoloesICQmQ15UN2zesWNvllQ2Ni3/cZgEhQb+8\nr7Vb9/LG/T/4zgnVbSol7IGfSYSEHhM5pF3fv/xv33NWnOY+uBu3KmwhD+2QAFFD2naSW9CR\nq4/odf4142VIF3dKbQgJCRA1pOvlq8t/OWhonxfc48dKrghZSUhIgKghDT3HfbNY5vk3LqoM\nWUlISICoIfW+2X3zqjzs37i5LGQlISEBooZ05Hj3zT/KXP/G+UNDVhISEiBqSDfINS882L+6\n4in3+CGZE7KSkJAAUUPacbqIHPCH82T01JN41Q6JF/n3SE2Lr1+4ytlWXyIlNW+HLSQkJEDe\nVzbsWPHX8AWEhATgn+OyNw+KEZK9eVCMkOzNg2KEZG8eFCMke/OgGCHZmwfFCMnePChGSPbm\nQTFCsjcPihGSvXlQjJDszYNihGRvHhQjJHvzoBgh2ZsHxQjJ3jwoRkj25kExQrI3D4oRkr15\nUIyQ7M2DYoRkbx4UIyR786AYIdmbB8UIyd48KEZI9uZBMUKyNw+KEZK9eVCMkOzNg2KEZG8e\nFCMke/OgGCHZmwfFCMnePChGSPbmQTFCsjcPihGSvXlQjJDszYNihGRvHhQjJHvzoBgh2ZsH\nxQjJ3jwoRkj25kExQrI3D4oFQ5qxrLkAZyAkJEAwpBI5fO6fjJ+BkJAAwZDW3/MpkZMXvWv2\nDISEBOj0HOmP8yql18Sf7zR4BkJCAuz3YkPL8zcfJf0ue97YGQgJCbBfSKvuOaOXnNBHJvzF\n0BkICQnQMaRN950mUjn/dWf7N+QCQ2cgJCRAMKSHzuklfaenXwOv7mfoDISEBAiGJCVn/XRH\n5saZtYbOQEhIgGBI31pfiDMQEhKgw3OkxkdfdN/O+nmryTMQEhIgGNL2M+VWx/m4l3x+t8Ez\nEBISIBjSAvmad1XDlrnyDYNnICQkQDCkY89NH5wzzOAZCAkJEAypz/z0wbwyg2cgJCRAMKSq\n8emDc48yeAZCQgIEQ5oni/zX6x6Qrxg8AyEhAYIhfVgtwy698tIqOXaLwTMQEhKgw++RPrjy\nEBE54MsmOyIkJEGnq7/3rVn+5j6zZyAkJAD/+Im9eVCsQ0j/fe15Z4/3GTwDISEBgiH9updk\nGDwDISEBgsmMGvzbbftSDJ6BkJAAwZAqvluIMxASEiAY0oCHCnEGQkICBENq+HwhzkBISIBg\nSG8P+dJ/vrfFZ/AMhIQECIZ02AG8alfIeVAsmExtO4NnICQkAFc22JsHxTqFtPPlpc52o//2\nCSEhCTqEtHlab/fp0V0jXjR5BkJCAgRDen+YjDlFnL8r7ftng2cgJCRAMKRrSpc4d7gfeLZs\nSm5fvHXditXbultESEiAYEiVlzp+SM70I3P4yqUNg/xXygfWPRO6jpCQAMGQym5OhzS3otuv\na2oQqa6pm15XUy0yuylkJSEhAYIhHXNWOqTR1d1+3Z0yNfMvha+fJotCVhISEiAY0ny5t9UN\nad+tclO3X1c1qqXtuHXMyJCVhIQECIa07QQ5ZoTMPEaq/9rt11XMD9xY0CdkJSEhATr8Hmnr\nVf1EpGzG+91/3Yhx7cetY6tCVhISEqDTlQ2Nrz63fFcuX3evXLY2fbhxliwMWUlISICo19o1\nzhQZVlM/o2HicJHavSErCQkJEP3q72WXHO7/HmlI/ZL9P7lra5tlhAT9OvwN2YzBQ3P74u0b\nVqzt8sqG1aUSQEhQr8Ordr53nxp76vacv37H62u6+m3sa8vb/IyQoF9Xz5F2Hn5191+4avod\n7j3PJPeu55BrPwxbyHMkJECXLzZcPqTbr3vjYLneeXugHHvplOFS/UHISkJCAnQZ0oX9uv26\nL/Re4jh1cleL47TcL9eErCQkJEAwpD2+j9+7r9dZ3X7dYdPdNwedkrpxYWXISkJCAnT5ql3Z\nc91+Xb957puDL03dmMclQki4YEjn+y6YvGBl9183avhux6kdtsc7bj75pJCVhIQEiHplw6Ny\n1ipn5UH1Wxxn4yXy/ZCVhIQEiPzPcd1W0uu0ORdL2fEjS6ShOWQhISEBslwilMNlQq/MHuw/\noSo9Y3Hov99FSEiAYEhD+4tUDOmd+z9b/Jc//e6lN8IuWPUQEhIgmMu6o896udlpXl5z6H95\n1woZOgMhIQGCIU3+dOpHfs8JFxk8AyEhAYIhHXh3+uCm7q9syB0hIQGCIQ38Uvqgtvtr7XJH\nSEiAYEhTeqf+J3o/Dr127pMiJCRAMKS3DpUzvvL1q0+UEd3/K0K5IyQkQIcXud+c5L32XfHl\n0P990SdFSEiATr8t2vbysld2mz0DISEB+ENj9uZBMf7QmL15UIw/NGZvHhTL6w+N5YSQkADR\n/9BYrggJCRD1D43ljpCQAFH/0FjuCAkJEPUPjeWOkJAAUf/QWO4ICQkQ9Q+N5Y6QkABR/9BY\n7ggJCRAM6dKFhTgDISEBgiH1PbcQZyAkJECHl7/HFOIMhIQECIb07KGXP/du6q+NGTwDISEB\ngiENGyC5/5t2OSMkJEAwmUntDJ6BkJAAJu97ukZISIBMSL96qVBnICQkQCYkGe+9/dOz5s9A\nSEiAjiGNL8AjPUJCAhCSvXlQjJDszYNihGRvHhQjJHvzoBgh2ZsHxdpCOuoW11FyS4rBMxAS\nEqAtpI4MnoGQkACZZH7RkcEzEBISgGvt7M2DYoRkbx4UIyR786AYIdmbB8UIyd48KEZI9uZB\nMUKyNw+KEZK9eVCMkOzNg2KEZG8eFCMke/OgGCHZmwfFCMnePChGSPbmQTFCsjcPihGSvXlQ\njJDszYNihGRvHhQjJHvzoBgh2ZsHxQjJ3jwoRkj25kExQrI3D4oRkr15UIyQ7M2DYoRkbx4U\nIyR786AYIdmbB8UIyd48KEZI9uZBMUKyNw+KEZK9eVCMkOzNg2KEZG8eFCMke/OgGCHZmwfF\nCMnePCiWV0hb161Yva27RcUb0pSz/8WsjWb3hxiJHtLShkHiGVj3TOi64g3puF4DjCq/wuz+\nECNRQ2pqEKmuqZteV1MtMrspZGXxhjR8uNl5PFRULGpId8rU9enD9dNkUchKQsogJMWihlQ1\nqqXtuHXMyJCVhJRBSIpFDalifuDGgj4hKwkpg5AUixrSiHHtx61jq0JWElIGISkWNaR75bK1\n6cONs2RhyEpCyiAkxaKG1DhTZFhN/YyGicNFaveGrCSkDEJSLPrvkZZdcrj/e6Qh9UtC1xFS\nBiEplteVDds3rFjb5ZUNW2ZMaTNBwu6vTCIk9Jj8r7VbuWn/j22/4ao2tdwjpRGSYvmHJPXh\nn+ehXQYhKRY1pJfbyDnum5CVhJRBSIpFDUk6CllJSBmEpFjUkOaVynm3+GSk+yZkJSFlEJJi\nkZ8jvXJi76/v8SfwHClHhKRY9BcbGm8vG/G8Q0i5IyTF8nnV7rVTS67+iJByRkiK5fXyd/O3\n+wx9gpByRUiK5fl7pLfOFELKFSEplu8vZFsfGH97+ApCyiAkxfjnuLIjJOSMkLIjJOSMkLIj\nJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIj\nJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIj\nJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSMkLIjJOSsJ0Pau8as+svM7pyQkLOeDOl6Mex4\nszsnJOSsJ0P6Yr3Ze6Rhhn/wCQk569GQYv6chpCQM0KyN4+QFCMke/MISTFCsjePkBQjJHvz\nCEkxQrI3j5AUIyR78whJMUKyN4+QFCMke/MISTFCsjePkBQjJHvzCEkxQrI3j5AUIyR78whJ\nMUKyN4+QFCMke/MISTFCsjePkBQjJHvzCEkxQrI3j5AUIyR78whJMUKyN4+QFCMke/MISTFC\nsjePkBQjJHvzCEkxQrI3j5AUIyR78whJMUKyN4+QFCMke/MISTFCsjePkBQjJHvzCEkxQrI3\nj5AUIyR78whJMUKyN4+QFCMke/MISTFCsjePkBQjJHvzCEkxQrI3j5AUIyR78whJMUKyN4+Q\nFCMke/MISTFCsjePkBQjJHvzCEkxQrI3j5AUIyR78whJMUKyN4+QFCMke/MISbG8Qtq6bsXq\nbd0tIqQMQlIsekhLGwaJZ2DdM6HrCCmDkBSLGlJTg0h1Td30uppqkdlNISsJKYOQFIsa0p0y\ndX36cP00WRSykpAyCEmxqCFVjWppO24dMzJkJSFlEJJiUUOqmB+4saBPyEpCyiAkxaKGNGJc\n+3Hr2KqQlYSUQUiKRQ3pXrlsbfpw4yxZGLKSkDIISbGoITXOFBlWUz+jYeJwkdq9ISsJKYOQ\nFIv+e6Rllxzu/x5pSP2S0HWElEFIiuV1ZcP2DSvWdnllQ/OSf2lzFyGlEZJieYTUui9ztLNz\nTesOG9DmIMn2wI+QoEbkkN68qKLkMw+3+sfjw6bw0C6DkBSLGtLq/lJ9YqlM/ti7QUg5ISTF\nooY0Qxa790rnytidDiHliJAUixrS4Au9ty1XS00jIeWIkBSLGlLvm/x3rV+Saa2ElBtCUixq\nSEMnpN7vu1DmElJuCEmxqCHNkftSl3/vGi1zRhNSLghJsaghbaqUIQ/7R1vHiBBSLghJsci/\nR9p05cCfpo723FROSLkgJMXyuUSoOXOw+fGQVYSUQUiK8c9x2ZtHSIoRkr15hKQYIdmbR0iK\nEZK9eYSkGCHZm0dIihGSvXmEpBgh2ZtHSIoRkr15hKQYIdmbR0iKEZK9eYSkGCHZm0dIihGS\nvXmEpBgh2ZtHSIoRkr15hKQYIdmbR0iKEZK9eYSkGCHZm0dIihGSvXmEpBgh2ZtHSIoRkr15\nhKQYIdmbR0iKEZK9eYSkGCHZm0dIihGSvXmEpBgh2ZtHSIoRkr15hKQYIdmbR0iKEZK9eYSk\nGCHZm0dIihGSvXmEpBgh2ZtHSIoRkr15hKQYIdmb99nyAWZ90+z+kAdCsjiv8mmjzuMeLj4I\nqXjn8VAxRgipeOcRUowQUvHOI6QYIaTinUdIMUJIxTuPkGKEkIp3HiHFCCEV7zxCihFCKt55\nhBQjhFS88wgpRgipeOcRUowQUvHOI6QYIaTinUdIMUJIxTuPkGKEkIp3HiHFCCEV7zxCihFC\nKt55hBQjhFS88wgpRgipeOcRUowQUvHOI6QYIaTinUdIMUJIxTuPkGKEkIp3HiHFCCEV7zxC\nihFCKt55hBQjhFS88wgpRgipeOcRUowQUvHOI6QYIaTinUdIMUJIxTuPkGKEkIp3HiHFCCEV\n7zxCihFCKt55hBQjhFS88wgpRgipeOcRUowQUvHOI6QYIaTinUdIMUJIxTuPkGKEkIp3HiHF\nCCEV7zxCihFCKt55hBQjhFS88wgpRgipeOedN+Iqs540u79EIaQinnfgFKOO4R4uOkJiXgYP\nFfNASMzLIKQ8EBLzMggpD3mFtHXditXbultESMUyL+4h/eRUw35icHPRQ1raMEg8A+ueCV1H\nSMUyL+4hffHUbxt1qsnvN2pITQ0i1TV10+tqqkVmN4WsJKRimRf7kAzvz+i8qCHdKVPXpw/X\nT5NFISsJqVjmEVIeooZUNaql7bh1zMiQlYRULPMIKQ9RQ6qYH7ixoE+nz64dPKDNQZLtgd8V\n5QOM6tWLefkorTA7r29fs/MqSs3OK78i4g9/V6KGNGJc+3Hr2KpOn2159uk2Tz2SbcZ7T5v1\n2GPMY94n8F7EH7rbKEwAAAj+SURBVP6uRA3pXrlsbfpw4yxZaGg3QJGKGlLjTJFhNfUzGiYO\nF6nda3RPQNGJ/nukZZcc7v8eaUj9EoP7AYpSXlc2bN+wYm23VzYACVD4a+2ABCAkwABCAgwg\nJMAAQgIMICTAAEICDCAkwABCAgwgJMAAQgIMICTAAEICDCAkwABCAgwgJMAAQgIM6MmQzhCg\nB51h8Ie5J0OacfHyWLuY/eUl9vubYfCHuSdDStq/7Gka+8tPLP6lVRMS9R+6ANhffgjJEvaX\nn0Ttj5CyY3/5SdT+CCk79pefRO2PkLJjf/lJ1P4IKTv2l59E7Y+QsmN/+UnU/ggpO/aXn0Tt\nj5CyY3/5SdT+ejKkq67qwZPngP3lJ1H768mQtm7twZPngP3lJ1H7439GARhASIABhAQYQEiA\nAYQEGEBIgAGEBBhASIABhAQYQEiAAYQEGEBIgAGEBBhASIABhAQYYCGkpkUj+xx7977cVnS/\n2Lj3rjqy7LBZa0NWBDf16Kh+h3/hDTs7azdPXg75bA/vr9tT9uz+dtx4VNmRX9kWssLE/iyE\ndJmMmzdKpua2ovvFpm0aKhNu/Lwcuir7ksCmbpPh86b3PjhkcSG8UhoaUs/ur/tT9uj+mk6X\ncfPPlpN3ZV9iYn+FD+kpucxxWqbKk7ms6H6xcVfKfe7bxXJx1hWBTf2h5KzdjrNEplvbnqfx\nJAkLqWf31/0pe3Z/i8X7X5R/Vb6fdYWR/RU+pMmyxn27Xhr2+8yp8/dbkX1xwQwc0uK9q65o\nzWF/s3v5/7/qujvs7c91e+/zugopHvvLfsp47O96ecF9u1z2/5dOjO6v8CENGOa/qx7kvtk+\nv6p86HUfpD9TWr/fiuBiO5p/8jP//Ykl+3LY3+DTrG2s3WtlX78lFVIc9xc8ZRz3t1AedN/+\nUm4r7P4KHtKHUuO/r5GtzrZPybibp5RWvZ/6VPobCawILrbrzdLPON3v7y8yZ2XtIQfXr7G5\nteZTj9uTCimO+wueMo77c97s1/+h1xYPqtxU2P0VPKRVktpunaxzrpJvuUdPZB6Bpr+RwIrg\nYqtazhX3jqnb/f1JLj746CmflcFhr/GZ9u2S55xUSHHcX/CUcdyf+7ynWkQqVzuF3V/BQ3pD\nUn9gcLqsbOw3zH86Mqb3TufySZMmlRzuvvnH4IrAYaH31UHr1TKp1el+f78TuWSP43xTLrK3\nt7f6XO2kQorl/gKnjOX+nG3nVcx9eG5F5ZuF3V/BQ1qbfuGgTt55VUbc4TlFfu8MzfxB3BuC\nKwKHhd5XUPOXZcxOx+l+fy9JmfcAu/nY0h229tZ65tCP0iHFcn+BU8Zyf069POW+faHXqML+\n9yt4SNvkXP99jXz0XNufk17qfyh91xpYETgs9L4Cdk2SCW5HTvf7e0OO9w8vlddtbe4H8oST\nDimW+wucMpb721Ey2n9/vqwv6P4K/6rdkOH+u+pKZ7nM7PCZzKsm7SuCh9ZsHS3TGr2D7ve3\nt3SkfzhFrD1dHt/2f/2Vsdxf4JSx3N+K9G/3r5GXC7q/wodUJ++6bze495+7yo7xL8O4/+7t\n/mcy30j7iuChLXvGytzUr5By2N8ZJd5hy3GHNNna3u31npFyTv2mWO4vcMpY7u+vJSf578fL\nOwXdX+FDekJmtzqtM+Rpx5kht7ofeL50ROoHN/ONBFYEDm25SeZkDrvf32Kpa3acB+Rae/vz\npV61i+X+AqeM5f5q5CH37WNyZmH3Z+Fau1r53C1jZZZ79H6VjJ47s7zvi1lXBA/t2Fwuo8f7\nduawv9Z6OXn+RVL9ob0N+lIhxXJ/gVPGcn+rhsj5Cy4qGfRmYfdnIaS9C4+pGHGv/yxky43H\nlA9t+HPIisChHY+3PQfZlsv+mu8/saLy6g/2n1NY6SsbYrm/wCljub+NV1b2PuJLGwq8P/73\nSIABhAQYQEiAAYQEGEBIgAGEBBhASIABhAQYQEiAAYQEGEBIgAGEBBhASIABhAQYQEiAAYQE\nGEBIgAGEBBhASIABhAQYQEiAAYQEGEBIgAGEBBhASIABhAQYQEiAAYQEGEBIgAGEBBhASIAB\nhAQYQEiAAYRUDLZJfdbP/f5Zx/vjjVusbQZdIaRiEBLS0l7e3xompJ5GSMUgJKRfCCHFASEV\nA0KKPUKKuVVTBx106f/4IW2fX1U+9DrvL27XDvzjhH6Dpr7tODO9v8e+0g1p5ZzB/cY+7X6u\n5e5P9+s/4bc9vO3EIaR4e2tQr9prjxnqhbTtUzLu5imlVe+7IVUcdPmL/+eoQaucp2bLjB9s\nc0M6+rDZF5aU/sFxbpTxC64ZUPKbnt55whBSvE2W/+feFZ3hhXSVfMv9wBMy3XskV+ceriyv\nbX9o97mdjvNDuc7Z3fsc9/brJRf07L4Th5Bi7aPeZ3rvnndDauw3rMU7HtN7p9vNcu/w4t5b\n20J60n27VS5wdpceu9k9XNPYY3tOJkKKtZdkrveusbTeeVVG3OE5RX7v1Jb4ndwqz7eF9I73\ngVI3uyuk7Oxvv96De04mQoq138gd/vuD653nJGOpUzvQ/+hdsqTjq3aln3Ocfd//tLvmM6/0\n2J6TiZBi7SW5ynu3t6TeWS4z2z5cW+E/yvuavLR/SK53H54kg3ZZ32yiEVKs7Sg/udXxeqp3\ndpUds8/70P13b3e7+S/v8LN9PnYe6RTSmm/82juaLP/dU3tOJkKKt+nyI8fZc473qt0MudX9\nwPOlI1rdbibsdpyfyeWO85h81wmGtKnklL3u47tTSv/SsxtPGkKKt/eOlkk3nnBkuRvS+1Uy\neu7M8r4vet0MHHljQ0n1+47zshw8cUXwod1NcuzX5p6UepEC1hBSzG368mH9zn/rAO/Khi03\nHlM+tOHPjhfSqxf2rbzWv9O54ZADfxUMqfnBU/v3O+2h1p7cdAIRUjHi0rrYIaRiREixQ0jF\niJBih5CKESHFDiEBBhASYAAhAQYQEmAAIQEGEBJgACEBBhASYAAhAQYQEmAAIQEGEBJgACEB\nBhASYAAhAQYQEmAAIQEGEBJgACEBBhASYAAhAQYQEmAAIQEGEBJgACEBBhASYAAhAQb8L26P\nCEjJ4P91AAAAAElFTkSuQmCC",
      "text/plain": [
       "Plot with title “Histogram of depths”"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "depths <- sample_sums(physeq)\n",
    "summary(depths)\n",
    "hist(depths)\n",
    "\n",
    "# Start to explore the data a bit \n",
    "#number of samples\n",
    "print ('Number of Samples in our Biom Table')\n",
    "nsamples(physeq)\n",
    "# number of sequence variants\n",
    "print ('Number of Sequence variants we have.')\n",
    "ntaxa(physeq)\n",
    "#summary statistics of sampling depth\n",
    "print ('Sequencing depth.')\n",
    "depths <- sample_sums(physeq)\n",
    "summary(depths)\n",
    "hist(depths)\n",
    "head(sort(sample_sums(physeq)),25)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metabolite P/LET T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-L</li>\n",
       "\t<li>Co-P</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Placebo\n",
       "\\item Co-L\n",
       "\\item Co-P\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Placebo\n",
       "3. Co-L\n",
       "4. Co-P\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Placebo   Co-L      Co-P     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>1</li>\n",
       "\t<li>5</li>\n",
       "\t<li>2</li>\n",
       "\t<li>cecum.T5</li>\n",
       "\t<li>4</li>\n",
       "\t<li>0</li>\n",
       "\t<li>3</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'0'</li>\n",
       "\t\t<li>'1'</li>\n",
       "\t\t<li>'2'</li>\n",
       "\t\t<li>'3'</li>\n",
       "\t\t<li>'4'</li>\n",
       "\t\t<li>'5'</li>\n",
       "\t\t<li>'cecum.T5'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1\n",
       "\\item 5\n",
       "\\item 2\n",
       "\\item cecum.T5\n",
       "\\item 4\n",
       "\\item 0\n",
       "\\item 3\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item '0'\n",
       "\\item '1'\n",
       "\\item '2'\n",
       "\\item '3'\n",
       "\\item '4'\n",
       "\\item '5'\n",
       "\\item 'cecum.T5'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1\n",
       "2. 5\n",
       "3. 2\n",
       "4. cecum.T5\n",
       "5. 4\n",
       "6. 0\n",
       "7. 3\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. '0'\n",
       "2. '1'\n",
       "3. '2'\n",
       "4. '3'\n",
       "5. '4'\n",
       "6. '5'\n",
       "7. 'cecum.T5'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1        5        2        cecum.T5 4        0        3       \n",
       "Levels: 0 1 2 3 4 5 cecum.T5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-P</li>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Co-L</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Placebo\n",
       "\\item Co-P\n",
       "\\item Letrozole\n",
       "\\item Co-L\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Placebo\n",
       "2. Co-P\n",
       "3. Letrozole\n",
       "4. Co-L\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Placebo   Co-P      Letrozole Co-L     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'2'\n",
       "</details>"
      ],
      "text/latex": [
       "2\n",
       "\\emph{Levels}: '2'"
      ],
      "text/markdown": [
       "2\n",
       "**Levels**: '2'"
      ],
      "text/plain": [
       "[1] 2\n",
       "Levels: 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Letrozole</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Placebo\n",
       "\\item Letrozole\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Placebo\n",
       "2. Letrozole\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Letrozole'\n",
       "2. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Placebo   Letrozole\n",
       "Levels: Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'2'\n",
       "</details>"
      ],
      "text/latex": [
       "2\n",
       "\\emph{Levels}: '2'"
      ],
      "text/markdown": [
       "2\n",
       "**Levels**: '2'"
      ],
      "text/plain": [
       "[1] 2\n",
       "Levels: 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(sample_data(physeq)$Category)\n",
    "unique(sample_data(physeq)$Week)\n",
    "\n",
    "#Remove cecum and time 0\n",
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '0' & Week != '1'& Week != '3'& Week != '4'& Week != '5')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)\n",
    "\n",
    "# Remove Co-P and placebo\n",
    "physeq.pt.pl = subset_samples(physeq.pt, Category != 'Co-L' & Category != 'Co-P')\n",
    "unique(sample_data(physeq.pt.pl)$Category)\n",
    "unique(sample_data(physeq.pt.pl)$Week)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 47 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n"
     ]
    }
   ],
   "source": [
    "diagdds = phyloseq_to_deseq2(physeq.pt.pl, ~ Category)\n",
    "diagdds$Category = relevel(diagdds$Category, ref = \"Letrozole\")\n",
    "# calculate geometric means prior to estimate size factors\n",
    "gm_mean = function(x, na.rm=TRUE){\n",
    "  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))\n",
    "}\n",
    "geoMeans = apply(counts(diagdds), 1, gm_mean)\n",
    "diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)\n",
    "diagdds = DESeq(diagdds, fitType=\"local\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res = results(diagdds)\n",
    "res = res[order(res$padj, na.last=NA), ]\n",
    "sigtab = res\n",
    "lower = -15\n",
    "upper = 15\n",
    "sigtab= subset(res, log2FoldChange < lower | log2FoldChange > upper)\n",
    "sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(physeq.pt.pl)[rownames(sigtab), ], \"matrix\"))\n",
    "sigtab = sigtab[order(-abs(sigtab$log2FoldChange)),]\n",
    "library(\"ggplot2\")\n",
    "pdf('Known_no_BA_T2_PvsL_DESEQ2.pdf', width=11, height=10)\n",
    "theme_set(theme_bw())\n",
    "sigtabgen = subset(sigtab, !is.na(Genus))\n",
    "# Phylum order\n",
    "x = tapply(sigtabgen$log2FoldChange, sigtabgen$Phylum, function(x) max(x))\n",
    "sigtabgen$Phylum = factor(as.character(sigtabgen$Phylum), levels=names(x))\n",
    "# Genus order\n",
    "x = tapply(sigtabgen$log2FoldChange, sigtabgen$Genus, function(x) max(x))\n",
    "sigtabgen$Genus = factor(as.character(sigtabgen$Genus), levels=names(x))\n",
    "ggplot(sigtabgen, aes(y=Genus, x=log2FoldChange)) + \n",
    "  geom_vline(xintercept = 0.0, color = \"gray\", size = 0.5) +\n",
    "  geom_point(size=4) + scale_x_continuous(minor_breaks=NULL) + coord_fixed(ratio=3) +\n",
    "  theme(text= element_text(size=20),axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5),legend.position = 'none')\n",
    "           \n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metabolite Co-L/LET T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-L</li>\n",
       "\t<li>Co-P</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Placebo\n",
       "\\item Co-L\n",
       "\\item Co-P\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Placebo\n",
       "3. Co-L\n",
       "4. Co-P\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Placebo   Co-L      Co-P     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>1</li>\n",
       "\t<li>5</li>\n",
       "\t<li>2</li>\n",
       "\t<li>cecum.T5</li>\n",
       "\t<li>4</li>\n",
       "\t<li>0</li>\n",
       "\t<li>3</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'0'</li>\n",
       "\t\t<li>'1'</li>\n",
       "\t\t<li>'2'</li>\n",
       "\t\t<li>'3'</li>\n",
       "\t\t<li>'4'</li>\n",
       "\t\t<li>'5'</li>\n",
       "\t\t<li>'cecum.T5'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1\n",
       "\\item 5\n",
       "\\item 2\n",
       "\\item cecum.T5\n",
       "\\item 4\n",
       "\\item 0\n",
       "\\item 3\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item '0'\n",
       "\\item '1'\n",
       "\\item '2'\n",
       "\\item '3'\n",
       "\\item '4'\n",
       "\\item '5'\n",
       "\\item 'cecum.T5'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1\n",
       "2. 5\n",
       "3. 2\n",
       "4. cecum.T5\n",
       "5. 4\n",
       "6. 0\n",
       "7. 3\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. '0'\n",
       "2. '1'\n",
       "3. '2'\n",
       "4. '3'\n",
       "5. '4'\n",
       "6. '5'\n",
       "7. 'cecum.T5'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1        5        2        cecum.T5 4        0        3       \n",
       "Levels: 0 1 2 3 4 5 cecum.T5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-P</li>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Co-L</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Placebo\n",
       "\\item Co-P\n",
       "\\item Letrozole\n",
       "\\item Co-L\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Placebo\n",
       "2. Co-P\n",
       "3. Letrozole\n",
       "4. Co-L\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Placebo   Co-P      Letrozole Co-L     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'2'\n",
       "</details>"
      ],
      "text/latex": [
       "2\n",
       "\\emph{Levels}: '2'"
      ],
      "text/markdown": [
       "2\n",
       "**Levels**: '2'"
      ],
      "text/plain": [
       "[1] 2\n",
       "Levels: 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Co-L</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Co-L\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Letrozole'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Co-L\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Letrozole'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Co-L     \n",
       "Levels: Co-L Letrozole"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'2'\n",
       "</details>"
      ],
      "text/latex": [
       "2\n",
       "\\emph{Levels}: '2'"
      ],
      "text/markdown": [
       "2\n",
       "**Levels**: '2'"
      ],
      "text/plain": [
       "[1] 2\n",
       "Levels: 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(sample_data(physeq)$Category)\n",
    "unique(sample_data(physeq)$Week)\n",
    "\n",
    "#Remove cecum and time 0\n",
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '0' & Week != '1'& Week != '3'& Week != '4'& Week != '5')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)\n",
    "\n",
    "# Remove Co-P and placebo\n",
    "physeq.pt.pl = subset_samples(physeq.pt, Category != 'Placebo' & Category != 'Co-P')\n",
    "unique(sample_data(physeq.pt.pl)$Category)\n",
    "unique(sample_data(physeq.pt.pl)$Week)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "converting counts to integer mode\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "final dispersion estimates\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "fitting model and testing\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "-- replacing outliers and refitting for 42 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "fitting model and testing\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "diagdds = phyloseq_to_deseq2(physeq.pt.pl, ~ Category)\n",
    "diagdds$Category = relevel(diagdds$Category, ref = \"Letrozole\")\n",
    "# calculate geometric means prior to estimate size factors\n",
    "gm_mean = function(x, na.rm=TRUE){\n",
    "  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))\n",
    "}\n",
    "geoMeans = apply(counts(diagdds), 1, gm_mean)\n",
    "diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)\n",
    "diagdds = DESeq(diagdds, fitType=\"local\")\n",
    "res = results(diagdds)\n",
    "res = res[order(res$padj, na.last=NA), ]\n",
    "sigtab = res\n",
    "lower = -15\n",
    "upper = 15\n",
    "sigtab= subset(res, log2FoldChange < lower | log2FoldChange > upper)\n",
    "sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(physeq.pt.pl)[rownames(sigtab), ], \"matrix\"))\n",
    "sigtab = sigtab[order(-abs(sigtab$log2FoldChange)),]\n",
    "library(\"ggplot2\")\n",
    "pdf('Known_no_BA_T2_CoLvsL_DESEQ2.pdf', width=11, height=10)\n",
    "theme_set(theme_bw())\n",
    "sigtabgen = subset(sigtab, !is.na(Genus))\n",
    "# Phylum order\n",
    "x = tapply(sigtabgen$log2FoldChange, sigtabgen$Phylum, function(x) max(x))\n",
    "sigtabgen$Phylum = factor(as.character(sigtabgen$Phylum), levels=names(x))\n",
    "# Genus order\n",
    "x = tapply(sigtabgen$log2FoldChange, sigtabgen$Genus, function(x) max(x))\n",
    "sigtabgen$Genus = factor(as.character(sigtabgen$Genus), levels=names(x))\n",
    "ggplot(sigtabgen, aes(y=Genus, x=log2FoldChange)) + \n",
    "  geom_vline(xintercept = 0.0, color = \"gray\", size = 0.5) +\n",
    "  geom_point(size=4) + scale_x_continuous(minor_breaks=NULL) + coord_fixed(ratio=3) +\n",
    "  theme(text= element_text(size=20),axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5),legend.position = 'none')\n",
    "           \n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metabolite P/LET T5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-L</li>\n",
       "\t<li>Co-P</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Placebo\n",
       "\\item Co-L\n",
       "\\item Co-P\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Placebo\n",
       "3. Co-L\n",
       "4. Co-P\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Placebo   Co-L      Co-P     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>1</li>\n",
       "\t<li>5</li>\n",
       "\t<li>2</li>\n",
       "\t<li>cecum.T5</li>\n",
       "\t<li>4</li>\n",
       "\t<li>0</li>\n",
       "\t<li>3</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'0'</li>\n",
       "\t\t<li>'1'</li>\n",
       "\t\t<li>'2'</li>\n",
       "\t\t<li>'3'</li>\n",
       "\t\t<li>'4'</li>\n",
       "\t\t<li>'5'</li>\n",
       "\t\t<li>'cecum.T5'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1\n",
       "\\item 5\n",
       "\\item 2\n",
       "\\item cecum.T5\n",
       "\\item 4\n",
       "\\item 0\n",
       "\\item 3\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item '0'\n",
       "\\item '1'\n",
       "\\item '2'\n",
       "\\item '3'\n",
       "\\item '4'\n",
       "\\item '5'\n",
       "\\item 'cecum.T5'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1\n",
       "2. 5\n",
       "3. 2\n",
       "4. cecum.T5\n",
       "5. 4\n",
       "6. 0\n",
       "7. 3\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. '0'\n",
       "2. '1'\n",
       "3. '2'\n",
       "4. '3'\n",
       "5. '4'\n",
       "6. '5'\n",
       "7. 'cecum.T5'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1        5        2        cecum.T5 4        0        3       \n",
       "Levels: 0 1 2 3 4 5 cecum.T5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-L</li>\n",
       "\t<li>Co-P</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Placebo\n",
       "\\item Co-L\n",
       "\\item Co-P\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Placebo\n",
       "3. Co-L\n",
       "4. Co-P\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Placebo   Co-L      Co-P     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'5'\n",
       "</details>"
      ],
      "text/latex": [
       "5\n",
       "\\emph{Levels}: '5'"
      ],
      "text/markdown": [
       "5\n",
       "**Levels**: '5'"
      ],
      "text/plain": [
       "[1] 5\n",
       "Levels: 5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Placebo</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Placebo\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Placebo\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Letrozole'\n",
       "2. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Placebo  \n",
       "Levels: Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'5'\n",
       "</details>"
      ],
      "text/latex": [
       "5\n",
       "\\emph{Levels}: '5'"
      ],
      "text/markdown": [
       "5\n",
       "**Levels**: '5'"
      ],
      "text/plain": [
       "[1] 5\n",
       "Levels: 5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(sample_data(physeq)$Category)\n",
    "unique(sample_data(physeq)$Week)\n",
    "\n",
    "#Remove cecum and time 0\n",
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '0' & Week != '1'& Week != '3'& Week != '4'& Week != '2')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)\n",
    "\n",
    "# Remove Co-P and placebo\n",
    "physeq.pt.pl = subset_samples(physeq.pt, Category != 'Co-L' & Category != 'Co-P')\n",
    "unique(sample_data(physeq.pt.pl)$Category)\n",
    "unique(sample_data(physeq.pt.pl)$Week)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 44 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on 'Methyl γ -linolenate' in 'mbcsToSbcs': dot substituted for <b3>”"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "diagdds = phyloseq_to_deseq2(physeq.pt.pl, ~ Category)\n",
    "diagdds$Category = relevel(diagdds$Category, ref = \"Letrozole\")\n",
    "# calculate geometric means prior to estimate size factors\n",
    "gm_mean = function(x, na.rm=TRUE){\n",
    "  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))\n",
    "}\n",
    "geoMeans = apply(counts(diagdds), 1, gm_mean)\n",
    "diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)\n",
    "diagdds = DESeq(diagdds, fitType=\"local\")\n",
    "res = results(diagdds)\n",
    "res = res[order(res$padj, na.last=NA), ]\n",
    "sigtab = res\n",
    "lower = -15\n",
    "upper = 15\n",
    "sigtab= subset(res, log2FoldChange < lower | log2FoldChange > upper)\n",
    "sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(physeq.pt.pl)[rownames(sigtab), ], \"matrix\"))\n",
    "sigtab = sigtab[order(-abs(sigtab$log2FoldChange)),]\n",
    "library(\"ggplot2\")\n",
    "pdf('Known_no_BA_T5_PvsL_DESEQ2.pdf', width=11, height=10)\n",
    "theme_set(theme_bw())\n",
    "sigtabgen = subset(sigtab, !is.na(Genus))\n",
    "# Phylum order\n",
    "x = tapply(sigtabgen$log2FoldChange, sigtabgen$Phylum, function(x) max(x))\n",
    "sigtabgen$Phylum = factor(as.character(sigtabgen$Phylum), levels=names(x))\n",
    "# Genus order\n",
    "x = tapply(sigtabgen$log2FoldChange, sigtabgen$Genus, function(x) max(x))\n",
    "sigtabgen$Genus = factor(as.character(sigtabgen$Genus), levels=names(x))\n",
    "ggplot(sigtabgen, aes(y=Genus, x=log2FoldChange)) + \n",
    "  geom_vline(xintercept = 0.0, color = \"gray\", size = 0.5) +\n",
    "  geom_point(size=4) + scale_x_continuous(minor_breaks=NULL) + coord_fixed(ratio=3) +\n",
    "  theme(text= element_text(size=20),axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5),legend.position = 'none')\n",
    "           \n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metabolite CoL/LET T5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-L</li>\n",
       "\t<li>Co-P</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Placebo\n",
       "\\item Co-L\n",
       "\\item Co-P\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Placebo\n",
       "3. Co-L\n",
       "4. Co-P\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Placebo   Co-L      Co-P     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>1</li>\n",
       "\t<li>5</li>\n",
       "\t<li>2</li>\n",
       "\t<li>cecum.T5</li>\n",
       "\t<li>4</li>\n",
       "\t<li>0</li>\n",
       "\t<li>3</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'0'</li>\n",
       "\t\t<li>'1'</li>\n",
       "\t\t<li>'2'</li>\n",
       "\t\t<li>'3'</li>\n",
       "\t\t<li>'4'</li>\n",
       "\t\t<li>'5'</li>\n",
       "\t\t<li>'cecum.T5'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1\n",
       "\\item 5\n",
       "\\item 2\n",
       "\\item cecum.T5\n",
       "\\item 4\n",
       "\\item 0\n",
       "\\item 3\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item '0'\n",
       "\\item '1'\n",
       "\\item '2'\n",
       "\\item '3'\n",
       "\\item '4'\n",
       "\\item '5'\n",
       "\\item 'cecum.T5'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1\n",
       "2. 5\n",
       "3. 2\n",
       "4. cecum.T5\n",
       "5. 4\n",
       "6. 0\n",
       "7. 3\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. '0'\n",
       "2. '1'\n",
       "3. '2'\n",
       "4. '3'\n",
       "5. '4'\n",
       "6. '5'\n",
       "7. 'cecum.T5'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1        5        2        cecum.T5 4        0        3       \n",
       "Levels: 0 1 2 3 4 5 cecum.T5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-L</li>\n",
       "\t<li>Co-P</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Placebo\n",
       "\\item Co-L\n",
       "\\item Co-P\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Placebo\n",
       "3. Co-L\n",
       "4. Co-P\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Placebo   Co-L      Co-P     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'5'\n",
       "</details>"
      ],
      "text/latex": [
       "5\n",
       "\\emph{Levels}: '5'"
      ],
      "text/markdown": [
       "5\n",
       "**Levels**: '5'"
      ],
      "text/plain": [
       "[1] 5\n",
       "Levels: 5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Co-L</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Co-L\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Letrozole'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Co-L\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Letrozole'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Co-L     \n",
       "Levels: Co-L Letrozole"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'5'\n",
       "</details>"
      ],
      "text/latex": [
       "5\n",
       "\\emph{Levels}: '5'"
      ],
      "text/markdown": [
       "5\n",
       "**Levels**: '5'"
      ],
      "text/plain": [
       "[1] 5\n",
       "Levels: 5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(sample_data(physeq)$Category)\n",
    "unique(sample_data(physeq)$Week)\n",
    "\n",
    "#Remove cecum and time 0\n",
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '0' & Week != '1'& Week != '3'& Week != '4'& Week != '2')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)\n",
    "\n",
    "# Remove Co-P and placebo\n",
    "physeq.pt.pl = subset_samples(physeq.pt, Category != 'Placebo' & Category != 'Co-P')\n",
    "unique(sample_data(physeq.pt.pl)$Category)\n",
    "unique(sample_data(physeq.pt.pl)$Week)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "converting counts to integer mode\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "final dispersion estimates\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "fitting model and testing\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "-- replacing outliers and refitting for 45 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "fitting model and testing\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <ce>”Warning message in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :\n",
      "“conversion failure on '3β -Hydroxy-5-cholestenoic acid' in 'mbcsToSbcs': dot substituted for <b2>”"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "diagdds = phyloseq_to_deseq2(physeq.pt.pl, ~ Category)\n",
    "diagdds$Category = relevel(diagdds$Category, ref = \"Letrozole\")\n",
    "# calculate geometric means prior to estimate size factors\n",
    "gm_mean = function(x, na.rm=TRUE){\n",
    "  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))\n",
    "}\n",
    "geoMeans = apply(counts(diagdds), 1, gm_mean)\n",
    "diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)\n",
    "diagdds = DESeq(diagdds, fitType=\"local\")\n",
    "res = results(diagdds)\n",
    "res = res[order(res$padj, na.last=NA), ]\n",
    "sigtab = res\n",
    "lower = -15\n",
    "upper = 15\n",
    "sigtab= subset(res, log2FoldChange < lower | log2FoldChange > upper)\n",
    "sigtab = cbind(as(sigtab, \"data.frame\"), as(tax_table(physeq.pt.pl)[rownames(sigtab), ], \"matrix\"))\n",
    "sigtab = sigtab[order(-abs(sigtab$log2FoldChange)),]\n",
    "library(\"ggplot2\")\n",
    "pdf('Known_no_BA_T5_CoLvsL_DESEQ2.pdf', width=11, height=10)\n",
    "theme_set(theme_bw())\n",
    "sigtabgen = subset(sigtab, !is.na(Genus))\n",
    "# Phylum order\n",
    "x = tapply(sigtabgen$log2FoldChange, sigtabgen$Phylum, function(x) max(x))\n",
    "sigtabgen$Phylum = factor(as.character(sigtabgen$Phylum), levels=names(x))\n",
    "# Genus order\n",
    "x = tapply(sigtabgen$log2FoldChange, sigtabgen$Genus, function(x) max(x))\n",
    "sigtabgen$Genus = factor(as.character(sigtabgen$Genus), levels=names(x))\n",
    "ggplot(sigtabgen, aes(y=Genus, x=log2FoldChange)) + \n",
    "  geom_vline(xintercept = 0.0, color = \"gray\", size = 0.5) +\n",
    "  geom_point(size=4) + scale_x_continuous(minor_breaks=NULL) + coord_fixed(ratio=1.5) +\n",
    "  theme(text= element_text(size=20),axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5),legend.position = 'none')\n",
    "           \n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.4.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
