# Descirption
CAPs analysis of beta-diversity using bray-curtis dissimilarity. Three data sets were used: 1) Gut Metabolites, 2) Gut 16S sequencing, and 3) WGS metagenome.

## File mapping

### Seperate Folder
  - 16s_map.csv = 16S mapping file. Includes sample ID associated with timepoint and treatment.
  - CAP.ipynb = Jupyter notebook that includes analysis for creating CAPs analysis for Metabolite (week 0 - week 5), 16S (week 2 and 5), WGS (week 2 and 5)
  - Mapping_file_w_og.csv = WGS and metabolite mapping file. Includes meta data for each sample
  - T2_filtered_greater_00001.csv = Week 2 WGS centrifuge microbial taxonomy count data
  - T5_filtered_greater_00001.csv = Week 5 WGS centrifuge microbial taxonomy count data
  - Unknown_Cutoff.csv = Week 0 to week 5 Metabolite count data
  - otu_table_alltime_w_category.tsv = Week 0 to week 5 16S QIIME taxonomy count data
  - taxonomy.csv = Metabolite taxonomy mapping file
  - taxonomy_16s.csv = 16S microbial taxonomy mapping file
  - taxonomy_centrifuge.csv = WGS microbial taxonomy file 
