{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Number of Samples in our Biom Table\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "207"
      ],
      "text/latex": [
       "207"
      ],
      "text/markdown": [
       "207"
      ],
      "text/plain": [
       "[1] 207"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Number of Sequence variants we have.\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "2602"
      ],
      "text/latex": [
       "2602"
      ],
      "text/markdown": [
       "2602"
      ],
      "text/plain": [
       "[1] 2602"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Sequencing depth.\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. \n",
       " 4532877  7949730  9897549 11268334 13440524 30800723 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<dl class=dl-horizontal>\n",
       "\t<dt>595.T1_RE8_01_33117</dt>\n",
       "\t\t<dd>4532877</dd>\n",
       "\t<dt>576.T5_BF11_01_33368</dt>\n",
       "\t\t<dd>4536638</dd>\n",
       "\t<dt>585.T3_BB3_01_33182</dt>\n",
       "\t\t<dd>4585675</dd>\n",
       "\t<dt>583.T5_BG5_01_33375</dt>\n",
       "\t\t<dd>4740326</dd>\n",
       "\t<dt>585.T0_RB3_01_33073</dt>\n",
       "\t\t<dd>4995143</dd>\n",
       "\t<dt>588.T5_BG9_01_33379</dt>\n",
       "\t\t<dd>5002816</dd>\n",
       "\t<dt>577.T5_BF12_01_33369</dt>\n",
       "\t\t<dd>5182670</dd>\n",
       "\t<dt>583.CT5_BB3_01_33413</dt>\n",
       "\t\t<dd>5231305</dd>\n",
       "\t<dt>584.T0_RB2_01_33072</dt>\n",
       "\t\t<dd>5484288</dd>\n",
       "\t<dt>585.T1_RD11_01_33107</dt>\n",
       "\t\t<dd>5670427</dd>\n",
       "\t<dt>596.T4_BE9_01_33353</dt>\n",
       "\t\t<dd>5745523</dd>\n",
       "\t<dt>573.CT5_BA7_01_33404</dt>\n",
       "\t\t<dd>6009823</dd>\n",
       "\t<dt>582.T2_RG4_01_33142</dt>\n",
       "\t\t<dd>6038130</dd>\n",
       "\t<dt>582.T3_BA12_01_33178</dt>\n",
       "\t\t<dd>6194230</dd>\n",
       "\t<dt>582.T5_BG4_01_33374</dt>\n",
       "\t\t<dd>6373841</dd>\n",
       "\t<dt>583.T3_BB1_01_33180</dt>\n",
       "\t\t<dd>6492383</dd>\n",
       "\t<dt>578.T4_BD5_01_33210</dt>\n",
       "\t\t<dd>6493480</dd>\n",
       "\t<dt>577.T4_BD4_01_33209</dt>\n",
       "\t\t<dd>6552949</dd>\n",
       "\t<dt>581.T2_RG3_01_33141</dt>\n",
       "\t\t<dd>6561712</dd>\n",
       "\t<dt>575.T5_BF10_01_33367</dt>\n",
       "\t\t<dd>6721939</dd>\n",
       "\t<dt>584.T2_RG6_01_33144</dt>\n",
       "\t\t<dd>6771429</dd>\n",
       "\t<dt>584.T1_RD10_01_33106</dt>\n",
       "\t\t<dd>6778044</dd>\n",
       "\t<dt>576.T2_RF11_01_33136</dt>\n",
       "\t\t<dd>6827855</dd>\n",
       "\t<dt>573.T4_BC12_01_33204</dt>\n",
       "\t\t<dd>6830087</dd>\n",
       "\t<dt>585.CT5_BB5_01_33415</dt>\n",
       "\t\t<dd>6877359</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[595.T1\\textbackslash{}\\_RE8\\textbackslash{}\\_01\\textbackslash{}\\_33117] 4532877\n",
       "\\item[576.T5\\textbackslash{}\\_BF11\\textbackslash{}\\_01\\textbackslash{}\\_33368] 4536638\n",
       "\\item[585.T3\\textbackslash{}\\_BB3\\textbackslash{}\\_01\\textbackslash{}\\_33182] 4585675\n",
       "\\item[583.T5\\textbackslash{}\\_BG5\\textbackslash{}\\_01\\textbackslash{}\\_33375] 4740326\n",
       "\\item[585.T0\\textbackslash{}\\_RB3\\textbackslash{}\\_01\\textbackslash{}\\_33073] 4995143\n",
       "\\item[588.T5\\textbackslash{}\\_BG9\\textbackslash{}\\_01\\textbackslash{}\\_33379] 5002816\n",
       "\\item[577.T5\\textbackslash{}\\_BF12\\textbackslash{}\\_01\\textbackslash{}\\_33369] 5182670\n",
       "\\item[583.CT5\\textbackslash{}\\_BB3\\textbackslash{}\\_01\\textbackslash{}\\_33413] 5231305\n",
       "\\item[584.T0\\textbackslash{}\\_RB2\\textbackslash{}\\_01\\textbackslash{}\\_33072] 5484288\n",
       "\\item[585.T1\\textbackslash{}\\_RD11\\textbackslash{}\\_01\\textbackslash{}\\_33107] 5670427\n",
       "\\item[596.T4\\textbackslash{}\\_BE9\\textbackslash{}\\_01\\textbackslash{}\\_33353] 5745523\n",
       "\\item[573.CT5\\textbackslash{}\\_BA7\\textbackslash{}\\_01\\textbackslash{}\\_33404] 6009823\n",
       "\\item[582.T2\\textbackslash{}\\_RG4\\textbackslash{}\\_01\\textbackslash{}\\_33142] 6038130\n",
       "\\item[582.T3\\textbackslash{}\\_BA12\\textbackslash{}\\_01\\textbackslash{}\\_33178] 6194230\n",
       "\\item[582.T5\\textbackslash{}\\_BG4\\textbackslash{}\\_01\\textbackslash{}\\_33374] 6373841\n",
       "\\item[583.T3\\textbackslash{}\\_BB1\\textbackslash{}\\_01\\textbackslash{}\\_33180] 6492383\n",
       "\\item[578.T4\\textbackslash{}\\_BD5\\textbackslash{}\\_01\\textbackslash{}\\_33210] 6493480\n",
       "\\item[577.T4\\textbackslash{}\\_BD4\\textbackslash{}\\_01\\textbackslash{}\\_33209] 6552949\n",
       "\\item[581.T2\\textbackslash{}\\_RG3\\textbackslash{}\\_01\\textbackslash{}\\_33141] 6561712\n",
       "\\item[575.T5\\textbackslash{}\\_BF10\\textbackslash{}\\_01\\textbackslash{}\\_33367] 6721939\n",
       "\\item[584.T2\\textbackslash{}\\_RG6\\textbackslash{}\\_01\\textbackslash{}\\_33144] 6771429\n",
       "\\item[584.T1\\textbackslash{}\\_RD10\\textbackslash{}\\_01\\textbackslash{}\\_33106] 6778044\n",
       "\\item[576.T2\\textbackslash{}\\_RF11\\textbackslash{}\\_01\\textbackslash{}\\_33136] 6827855\n",
       "\\item[573.T4\\textbackslash{}\\_BC12\\textbackslash{}\\_01\\textbackslash{}\\_33204] 6830087\n",
       "\\item[585.CT5\\textbackslash{}\\_BB5\\textbackslash{}\\_01\\textbackslash{}\\_33415] 6877359\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "595.T1_RE8_01_33117\n",
       ":   4532877576.T5_BF11_01_33368\n",
       ":   4536638585.T3_BB3_01_33182\n",
       ":   4585675583.T5_BG5_01_33375\n",
       ":   4740326585.T0_RB3_01_33073\n",
       ":   4995143588.T5_BG9_01_33379\n",
       ":   5002816577.T5_BF12_01_33369\n",
       ":   5182670583.CT5_BB3_01_33413\n",
       ":   5231305584.T0_RB2_01_33072\n",
       ":   5484288585.T1_RD11_01_33107\n",
       ":   5670427596.T4_BE9_01_33353\n",
       ":   5745523573.CT5_BA7_01_33404\n",
       ":   6009823582.T2_RG4_01_33142\n",
       ":   6038130582.T3_BA12_01_33178\n",
       ":   6194230582.T5_BG4_01_33374\n",
       ":   6373841583.T3_BB1_01_33180\n",
       ":   6492383578.T4_BD5_01_33210\n",
       ":   6493480577.T4_BD4_01_33209\n",
       ":   6552949581.T2_RG3_01_33141\n",
       ":   6561712575.T5_BF10_01_33367\n",
       ":   6721939584.T2_RG6_01_33144\n",
       ":   6771429584.T1_RD10_01_33106\n",
       ":   6778044576.T2_RF11_01_33136\n",
       ":   6827855573.T4_BC12_01_33204\n",
       ":   6830087585.CT5_BB5_01_33415\n",
       ":   6877359\n",
       "\n"
      ],
      "text/plain": [
       " 595.T1_RE8_01_33117 576.T5_BF11_01_33368  585.T3_BB3_01_33182 \n",
       "             4532877              4536638              4585675 \n",
       " 583.T5_BG5_01_33375  585.T0_RB3_01_33073  588.T5_BG9_01_33379 \n",
       "             4740326              4995143              5002816 \n",
       "577.T5_BF12_01_33369 583.CT5_BB3_01_33413  584.T0_RB2_01_33072 \n",
       "             5182670              5231305              5484288 \n",
       "585.T1_RD11_01_33107  596.T4_BE9_01_33353 573.CT5_BA7_01_33404 \n",
       "             5670427              5745523              6009823 \n",
       " 582.T2_RG4_01_33142 582.T3_BA12_01_33178  582.T5_BG4_01_33374 \n",
       "             6038130              6194230              6373841 \n",
       " 583.T3_BB1_01_33180  578.T4_BD5_01_33210  577.T4_BD4_01_33209 \n",
       "             6492383              6493480              6552949 \n",
       " 581.T2_RG3_01_33141 575.T5_BF10_01_33367  584.T2_RG6_01_33144 \n",
       "             6561712              6721939              6771429 \n",
       "584.T1_RD10_01_33106 576.T2_RF11_01_33136 573.T4_BC12_01_33204 \n",
       "             6778044              6827855              6830087 \n",
       "585.CT5_BB5_01_33415 \n",
       "             6877359 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAC/VBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSV\nlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqan\np6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5\nubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrL\ny8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd\n3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v\n7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///9LWOIm\nAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3de3wV9Z3w8W8SkgAqckdBxUQFr7Wu\nCoIi1aC2okSTKFdxq/VGL1642Nq11np5Sm3XWp/WrtuqW+1jn92nPtpWEOuitau1a9XWihYV\nsKhYUUAod5J57ZzJbc58Tw7znfwGMubz/iNnzsnv/OZ34vl4bkMiHoBOk929AODjgJAABwgJ\ncICQAAcICXCAkAAHCAlwgJAABwgJcICQAAcICXCAkAAHCAlwgJAABwgJcICQAAcICXCAkAAH\nCAlwgJAABwgJcICQAAcICXCAkAAHCAlwgJAABwgJcICQAAcICXCAkAAHCAlwgJAABwgJcICQ\nAAcICXCAkAAHCAlwgJAABwgJcICQHHlaRJ5u3qwUucHzFvgXbNfj3lq9K1Zz33F77HHo9yIX\nvuqv6PWdXjVYYG7xqSzsY4sflyPxQtp8U+8Xd8FifiQ5X49cGieklgUSkhU/LkdihfRhtciu\nCOlTInvOmPRk5NIYIbUukJCs+HE5okLasnLlyuig5bJrQjo4WEFUjJBaF0hIVvy4HFEhFbKr\nQtpP5Af6UkJKET8uRzp8avfOlYfvUdb/pDu2ed704KWL/LN/6Y77Tx9aMfS81mdfL9UP6HXs\nvd4eIss973si0/94fMXQu73td4ztW9brkEtyD21fFfnqB1/av3Lkt5u8Hx/Tc/D0t8K7D893\nRvNuDmr/7u9r+/ca9dO2kF6evm/5kNrHcpv3iEzcesvInvt/IfceQ9sCg5AeGN27/7kvBROE\nbgQKIiRHOgrp1YHNd06p3RYK6b3xLZfO2pG7xs8rms+0hTRhiH92YeOUllH9Xg9CmrJfcO6L\nM4OTfT9s33vefCqkB3oEF5zdEtID5c0DrvaCkGpOCs4d8GYkpMuC7V7Pe/k3AgURkiO5kCoq\nAxIO6dMix99y+8wSkR953/2Mf9mE+l96jbn7/aDj/W5knn/dv+7pX7d2ct/cXXV5EFJO320/\nEyk584rT/eteHoQkctolwV164EWj/a/faNt5/nzzp/cWGTV9dut33+glUj5pan9pDukFv9qR\nF5/qn7knCMmf7qzD/K+jmtoXmFu8HHLJBP/rRC//RqAgQnLkaQlrD8lvZIn/7W+cetXC9pcg\nP/FPv9rorTtLpNT/7rUiZb/1vGX7t4d03PL3f+V9+xPl1/ujrxA5qTmkb3veE7nHolVe4yj/\nEaZt55H5vGF5r5Gu8qf3HyzfOqA5pHP8OPyF3emX0hSENGaN1zTHP30s/zXSGf6jz6V+z17k\nRqAQQnKko5D8+/TQ2QvXNw9qvZ/6Dwcn5M5/sLfIVzzvH0Qm587+oD2kJ5qvsG2r5234osjR\nQUh7+fOt97/5Zf87/oXj2nYemS8S0pEiM3Kn3wtC2tpTZIF/bof/OLU0COk5/9yWfUWuyQ8p\n90T15wVuBAohJEc6eo30tSCssjG3rfXa76f9RG4JhvpPpU7zvL1EvpU798f2kLY0T/X+A58/\nPveK5qggJD8nr9E/92/+6VdERrftPDJfJCT/id7tudPngpCWhHr//7mQyhtz35wkclZ+SOv8\n01/7pxsjNwKFEJIjHYW0Y27LS/sBi9vup40lwesT38X+iw+vSVrOvtkW0t7BdzdenHuXoGzf\n1keksbkL/Use8oKzbSFF5ouElJv+/tzGkiCk34VC+rdcSEOCUf6TuE/lh5Tra7F/uiFyI1AI\nITnS8ZENq++aFLyNMHRH2/10UOsjiP8q/ozgIeM7uXO/aQtpaPBdP4u+cxet/77IMUE5J+Yu\n9Mf8wssPKTpf5BHJf4VzR+70P4OQ/uR/fWpDs+25kCqCVU4RmaQ/R2oJKe9GoBBCcqTDR6Sl\n/uvzxj9+zj/3l+B++oI/YnzoNc3XPO8YkXNyZ/+pLaQDcmc3+Q8DD/un3xQ5tnhIkfkiIR3X\n8hLs60FIG/1HuR/659Y++Odtze/aLfbPNVUFb/i1LjASUt6NQCGE5EgHIf26ovmNg9/75972\n3vG/LvK2e3c1v8v20VkiPZo/Iirxn3wt3rMtpOG5ef4qQUibPtn61K7DkCLzRUK63n96+Kjn\nvdC3+V2700WqVwdZ9VsfhHTk2553c3NQrQvMDyn/RqAQQnKkg5C2HSay56xvz903eAt7c5n/\n8ueQm7wt/+B/c9CoXDi597dX7eVvHH5UqeSHtL2/yOD5t430L60uHlJkvkhIq/qJlJ58Su7z\nrVxIv/FfUQ37wpTS4N2/4HOkPqcfKc2zty4wP6T8G4FCCMmRjp7avXZAywv74W/63zolt3WZ\n/2BzQsulVzXlrvFwcOhBae6p3Yr2kLz/3TzGn6F8U9GQovPlh+Q91iv43rUDmo9suLO0eew5\n24OQ+k4KzlUFhxy1LDDy1C7/RqAAQnKkwzcbPrpl9N5lfY67Mfdusreyvk+vQ7/vb2y/74x9\nK4ZObrmG99zEvnue8tTL/lVWhULyHhzVs/LQ69b7d+OfFQ8pMl8kJO+Vhv69T/x/XktI3vMz\nDqgcPO7+3PsGfkgDdnzrkIr9r2w+4KhlgdE3G/JuBAogpC7g7VXB40juNUjJll2761xIu3aP\nH1OE1AVcLeUH/qd/eoPIIbt414TkCCF1AQ/5T6D6Tr94jH9y4y7eNSE5Qkhdwfktr+Vl9KZd\nvGdCcoSQuoKm+yfuV9lj4Kfu3Lqr90xIjhAS4AAhAQ4QEuAAIQEOEBLgACEBDhAS4AAhAQ4Q\nEuAAIQEOEBLgACEBDhAS4AAhAQ4QEuAAIQEOEBLgACEBDhAS4AAhAQ4QEuAAIQEOEBLgACEB\nDhAS4AAhAQ4QEuAAIQEOEBLgACEBDhAS4AAhAQ4QEuAAIQEOEBLgACEBDhAS4AAhAQ4QEuAA\nIQEOEBLgACEBDnQqpDXLl7yx1tVKgAxLHtKChoGSM6DuCYfrATIpaUjbGkSqa+qm1tVUi8zc\n5nRNQOYkDelGmbyiZXPFFJnvajlANiUNqWpUY9t205iRbhYDZFXSkCrnhM7M6+liKUB2JQ1p\nxLj27aaxVU7WAmRW0pBulQuWtWyunCHXu1oOkE1JQ9o6XWR4Tf20hgkHi9RucbomIHOSf460\n8Nx9gs+RBtc/4nA9QCZ16siGdW8tWcaRDQCHCAFOcIgQ4ACHCAEOcIgQ4EA6hwg1Ln68zaL7\nk66tE5qefDw963bDDUIXl84hQssG9Wuzl+yGJ35/lL790tLjf+3624OuLv1DhP5LtibcRyf8\nQT5Kbe6TbkptamRW+ocIERK6gfQPESIkdAPpHyJESOgG0j9EiJDQDaT/67gICd0AIZkREjRC\nMiMkaElD+la+IiMJCd1A0pAGSJ4iIwkJ3UDSkFaPl9MealdkJCGhG0j8GmnzmJLFsQYSErqB\n5G82vNt35PY44wgJ3UAn3rW764hFcYYREroB3v42IyRohGRGSNAIyYyQoBGSGSFBIyQzQoJG\nSGaEBI2QzAgJGiGZERI0QjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZkR\nEjRCMiMkaIRkRkjQCMmMkKARkhkhQSMkM0KCRkhmhASNkMwICRohmRESNEIyIyRohGRGSNAI\nyYyQoBGSGSFBIyQzQoJGSGaEBI2QzAgJGiGZERI0QjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNC\ngkZIZoQEjZDMCAkaIZkREjRCMiMkaIRkRkjQCMmMkKARkhkhQSMkM0KCRkhmhASNkMwICRoh\nmRESNEIyIyRohGRGSNAIyYyQoBGSGSFBIyQzQoJGSGaEBI2QzAgJGiGZERI0QjIjJGiEZEZI\n0AjJjJCgEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZkREjRCMiMkaIRkRkjQCMmMkKARkhkhQSMk\nM0KCRkhmhASNkMwICRohmRESNEIyIyRohGRGSNAIyYyQoBGSGSFBIyQzQoJGSGaEBI2QzAgJ\nGiGZERI0QjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZkREjRCMiMkaJ0K\nac3yJW+s3dkgQkI3kDykBQ0DJWdA3RNFxxESuoGkIW1rEKmuqZtaV1MtMnNbkZGEhG4gaUg3\nyuQVLZsrpsj8IiMJCd1A0pCqRjW2bTeNGVlkJCGhG0gaUuWc0Jl5PYuMJCR0A0lDGjGufbtp\nbFWRkYSEbiBpSLfKBctaNlfOkOuLjCQkdANJQ9o6XWR4Tf20hgkHi9RuKTKSkNANJP8caeG5\n+wSfIw2uf6ToOEJCN9CpIxvWvbVkWcEjG9bMurRNLSHh46/Tx9otf3alvpCQ0M0kDmnjdy/8\np3e9Jcf5T+7GLS02kKd26AaShrT2SL+g/d7Yt/SMy8fL4AIPSm0ICd1A0pA+L198/qcDh/X8\njb/9s5KLi4wkJHQDSUMador/5V6ZHZw5a2iRkYSEbiBpSD3m+l9eknuCM3PLi4wkJHQDSUPa\nb7z/5V/kmuDMGcOKjCQkdANJQ7pSLv/NXX2rKxf523fLJUVGEhK6gaQhrT9eRPb4/ekyevKR\nvGuHbi/x50jb7v389Uu9tfUlUlLzerGBhIRuoNNHNqxf8kHxAYSEboBfx2VGSNAIyYyQoBGS\nGSFBIyQzQoJGSGaEBI2QzAgJGiGZERI0QjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNCgkZIZoQE\njZDMCAkaIZkREjRCMiMkaIRkRkjQCMmMkKARkhkhQSMkM0KCRkhmhASNkMwICRohmRESNEIy\nIyRohGRGSNAIyYyQoBGSGSFBIyQzQoJGSGaEBI2QzAgJGiGZERI0QjIjJGiEZEZI0AjJjJCg\nEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZkREjRCMiMkaIRkRkjQCMmMkKARkhkhQSMkM0KCRkhm\nhASNkMwICRohmRESNEIyIyRohGRGSNAIyYyQoBGSGSFBIyQzQoJGSGaEBI2QzAgJGiGZERI0\nQjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZmdeN2a1GxKbdVIFyGZ9ZP0\nHJLaqpEuQjLrc+KbabljSGqrRroIyazPhNSmfpCQsoqQzAgJGiGZERI0QjIjJGiEZEZI0AjJ\njJCgEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZkREjRCMiMkaIRkRkjQCMmMkKARkhkhQSMkM0KC\nRkhmhASNkMwICRohmRESNEIyIyRohGRGSNAIyYyQoBGSGSFBIyQzQoJGSGaEBI2QzAgJGiGZ\nERI0QjIjJGidCmnN8iVvrN3ZIEKKj5AyK3lICxoGBn+JZEDdE0XHEVJ8hJRZSUPa1iBSXVM3\nta6mWmTmtiIjCSk+QsqspCHdKJNXtGyumCLzi4wkpPgIKbOShlQ1qrFtu2nMyCIjCSk+Qsqs\npCFVzgmdmdezyEhCio+QMitpSCPGtW83ja0qMpKQ4iOkzEoa0q1ywbKWzZUz5PoiIwkpPkLK\nrKQhbZ0uMrymflrDhINFarcUGUlI8RFSZiX/HGnhufsEnyMNrn+k6DhCio+QMqtTRzase2vJ\nsoJHNiwb1K/NXlLs8SolhIRdq7PH2q1/+c0Cn8Y2Ln68ze08IsVGSJmVOKSlU2/wvDcmlons\nfcWHxQby1C4+QsqspCG90kc+770+QA46/7yDpfr9IiMJKT5CyqykIZ3T4xHPq5ObGv3ncbfL\n5UVGElJ8hJRZSUMaMtX/stcxzWfOHFpkJCHFR0iZlTSk3rP9L33Obz4zm0OE3CCkzEoa0qiD\nN3le7fDNue0dRx1ZZCQhxUdImZU0pAfk5KXeq3vVr/a8lefKd4uMJKT4CCmzEr/9/dWS0uMu\nOVvKDx1ZIg07igwkpPgIKbOSfyD73MxBwSFCZSfc21RsHCHFR0iZFQ5p2sJijywF/O3F3z7z\nys4OACKk+Agps8Ihlcg+17zofA+EFB8hZVY4pBW3HC5y1Py33e6BkOIjpMyKvEZ6YfZQKZ1w\n3waHeyCk+Agps9SbDY1Pzd1fel/wlLM9EFJ8hJRZKqSlt5xQKof1lFP/5mgPhBQfIWVWfkjv\nfOc4kaFzXvbWfUU+7WgPhBQfIWVWOKS7TymVXlNb3gOv7u1oD4QUHyFlVjgkKTn5R+tbz5xU\n62gPhBQfIWVWOKRvrOhwWCcQUnyElFl5r5G2PvC0/3XGfUUP+bEipPgIKbPCIa07Sa7zvL+X\nymc2OdwDIcVHSJkVDmmefCl3VMPqa+QrDvdASPERUmaFQzrotJaNU4Y73AMhxUdImRUOqWfr\nX5iYXe5wD4QUHyFlVjikqvEtG6ft73APhBQfIWVWOKTZMj94v+4O+YLDPRBSfISUWeGQPqyW\n4ed/7vwqOWi1wz0QUnyElFl5nyO9/7m9RWSPi1x2REgGhJRZkaO/t7/5/Gvb3e6BkOIjpMzq\n7F+j2DlCio+QMisvpD9ccfqnxgcc7oGQ4iOkzAqH9MtSaeVwD4QUHyFlVjiZUYMeXbu9mcM9\nEFJ8hJRZ4ZAqb0tjD4QUHyFlVjikfnensQdCio+QMiscUsNn0tgDIcVHSJkVDun1wZ/9r3dX\nBxzugZDiI6TMCoc0ZA/etYuBkKCFk6lt53APhBQfIWUWRzaYERK0SEgbnl3grXP6u08IyYCQ\nMisvpFVTevgvj24a8bTLPRBSfISUWeGQ3hsuY44R75/Lev3J4R4IKT5CyqxwSJeXPeLd4F+w\nuPw8h3sgpPgIKbPCIQ093wtC8qbu53APhBQfIWVWOKTyuS0hXVPpcA+EFB8hZVY4pANPbglp\ndLXDPRBSfISUWeGQ5sitTX5I26+Tqx3ugZDiI6TMCoe09jA5cIRMP1CqP3C4B0KKj5AyK+9z\npDWX9haR8mnvudwDIcVHSJkVObJh60tPPr/R7R4IKT5CyiyOtTMjJGgc/W1GSNDy/oZsq0HD\nHO6BkOIjpMzKe9cu8Paisceuc7gHQoqPkDKr0GukDftc5nAPhBQfIWVWwTcbLhzscA+EFB8h\nZVbBkM7s7XAPhBQfIWVWOKTNgb+/+53Skx3ugZDiI6TMKviuXfmTDvdASPERUmaFQzoj8OlJ\n8151uQdCio+QMosjG8wICRohmREStA4OEXJ4mBAhxUdImRUOaVhfkcrBPRz/2mJCio+QMiuc\ny/IDTn52h7fj+Zr+/507VsjRHggpPkLKrHBIkz7RfJfffNhZDvdASPERUmaFQ9rz5paNqzmy\noQhCghYOacBnWzZqOdauCEKCFg7pvB6PBKc/lMsd7oGQ4iOkzAqH9Jf+csIXvnzZETKC3yJU\nBCFBy3uT+7WJufe+Ky/60OUeCCk+QsqsyKdFa59d+Nwmt3sgpPgIKbP4Q2NmhASNPzRmRkjQ\n+ENjZoQEjT80ZkZI0PhDY2aEBI0/NGZGSND4Q2NmhASNPzRmRkjQ+ENjZoQEjT80ZkZI0PhD\nY2aEBC0c0vnXp7EHQoqPkDIrHFKv09LYAyHFR0iZlff295g09kBI8RFSZoVDWtz/wiffbv5r\nYw73QEjxEVJmhUMa3k8c/067HEKKj5AyK5zMxHYO90BI8RFSZvG7v80ICVprSL94Jq09EFJ8\nhJRZrSHJ+NzXFxe73wMhxUdImZUf0vgUnukRUnyElFmEZEZI0AjJjJCgEZIZIUEjJDNCgkZI\nZoQErS2k/a/17S/XNnO4B0KKj5Ayqy2kfA73QEjxEVJmtSbzk3wO90BI8RFSZnXqsWfN8iVv\n7PRfXBBSfISUWclDWtAwMHgWOKDuiaLjCCk+QsqspCFtaxCprqmbWldTLTJzW5GRhBQfIWVW\n0pBulMkrWjZXTJH5RUYSUnyElFlJQ6oa1di23TRmZJGRhBQfIWVW0pAq54TOzOtZZCQhxUdI\nmZU0pBHj2rebxlYVGUlI8RFSZiUN6Va5YFnL5soZUuw3SxJSfISUWUlD2jpdZHhN/bSGCQeL\n1G4pMpKQ4iOkzEr+OdLCc/cJPkcaXP9I0XGEFB8hZVanjmxY99aSZQWPbGhc/Hib2wkpNkLK\nrM4fnvrqO/qyZYP6tdlLij3xSwkhYdfqfEhSX/z7PLWLj5AyK2lIz7aRU/wvRUYSUnyElFlJ\nQ4r/75cIKT5CyqykIc0uk9Ob/zGtjCz+L2oJKT5CyqzEr5GeO6LHlzcHM/AayRlCyqzkbzZs\n/Vr5iKc8QnKJkDKrM+/a/fHYkss+IiSHCCmzOvX2945v9hz2MCG5Q0iZ1cnPkf5ykhCSO4SU\nWZ39QLbpjvFfKz6CkOIjpMziL/aZERI0QjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNCgkZIZoQE\njZDMCAkaIZkREjRCMiMkaIRkRkjQCMmMkKARkhkhQSMkM0KCRkhmhASNkMwICRohmRESNEIy\nIyRohGRGSNAIyYyQoBGSGSFBIyQzQoJGSGaEBI2QzAgJGiGZERI0QjIjJGiEZEZI0AjJjJCg\nEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZkREjRCMiMkaIRkRkjQCMmMkKARkhkhQSMkM0KCRkhm\nhASNkMwICRohmRESNEIyIyRohGRGSNAIyYyQoBGSGSFBIyQzQoJGSGaEBI2QzAgJGiGZERI0\nQjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZkREjRCMiMkaIRkRkjQCMmM\nkKARkhkhQSMkM0KCRkhmhASNkMwICRohmRESNEIyIyRohGRGSNAIyYyQoBGSGSFBIyQzQoJG\nSGaEBI2QzAgJGiGZERI0QjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZkR\nEjRCMiMkaIRkRkjQCMmMkKARkhkhQSMkM0KCRkhmhASNkMwICRohmRESNEIyIyRohGRGSNAI\nySzNkAavSc3a1FYNj5ASSDGkL0uKHk1t2ehkSGuWL3ljp/+jI6T4rix9MzUDH0xt2ehMSAsa\nBgb/oxtQ90TRcYQU35WlqU3tDSGkNCUNaVuDSHVN3dS6mmqRmduKjCSk+Agps5KGdKNMXtGy\nuWKKzC8ykpDiI6TMShpS1ajGtu2mMSOLjCSk+Agps5KGVDkndGZezyIjCSk+QsqspCGNGNe+\n3TS2qshIQoqPkDIraUi3ygXLWjZXzpDri4wkpPgIKbOShrR1usjwmvppDRMOFqndUmQkIcVH\nSJmV/HOkhefuE3yONLj+kaLjCCk+QsqsTh3ZsO6tJcsKHtmwZtalbWoJKTZCyqxOhNS0vXVr\nQ7QmQkqGkDIrcUivnVVZcvQ9TcH2+GKz8NQuPkLKrKQhvdFXqo8ok0l/z50hJEcIKbOShjRN\n7vUflU6TsRs8QnKGkDIraUiDzsx9bbxMarYSkjOElFlJQ+pxdXDS9FmZ0kRIrhBSZiUNadip\nzafbz5RrCMkVQsqspCFdIt9pPvx742i5ZDQhuUFImZU0pHeGyuB7gq01Y0QIyQ1CyqzEnyO9\n87kBP2re2nx1BSG5QUiZ1ZlDhHa0bqx6qMgoQoqPkDKLX8dlRkjQCMmMkKARkhkhQSMkM0KC\nRkhmhASNkMwICRohmRESNEIyIyRohGRGSNAIyYyQoBGSGSFBIyQzQoJGSGaEBI2QzAgJGiGZ\nERI0QjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNCgkZIZoQEjZDMCAkaIZkREjRCMiMkaIRkRkjQ\nCMmMkKARkhkhQSMkM0KCRkhmhASNkMwICRohmRESNEIyIyRohGRGSNAIyYyQoBGSGSFBIyQz\nQoJGSGaEBI2QzAgJGiGZERI0QjIjJGiEZEZI0AjJjJCgEZIZIUEjJDNCgkZIZoQEjZDMCAka\nIZkREjRCMiMkaIRkRkjQCMmMkKARkhkhQSMks4yG1GfStan59/SWnRWEZJbRkEoPOS8tR56Y\n3rKzgpDMshrSlalNfTMhEZIdIUUREiElQEhRhERICRBSFCERUgKEFEVIhJQAIUUREiElQEhR\nhERICRBSFCERUgKEFEVIhJQAIUUREiElQEhRhERICRBSFCERUgKEFEVIhJQAIUUREiElQEhR\nhERICRBSFCERUgKEFEVIhJQAIUUREiElQEhRhERICRBSFCERUgKEFEVIuzekX16amjpZl9oN\nIqQoQtq9If3jiNRCOlXeTu0GEVIUIe3mkP4xtZ0+QEhRhJQqQjIjpChCIqQECCmKkAgpAUKK\nIiRCSoCQogiJkBIgpChCIqQECCmKkAgpAUKKIiRCSoCQogiJkBIgpChCIqQECCmKkAgpAUKK\nIiRCSoCQogiJkBIgpChCIqQECCmKkAgpAUKKmnvQD1Nz2ffTm/tlhz8DQjIjpKjTyo5Nyyek\nKrW5B3zW4c+AkMwIKWpCn9SmfkfuT21up/c/QjIjpChCIqQECCmKkAgpAUKKIiRCSoCQogip\nkyGtWb7kjbU7G0RI8RFSVDcIaUHDQMkZUPdE0XGEFB8hRX3sQ9rWIFJdUze1rqZaZOa2IiMJ\nKT5CivrYh3SjTF7RsrliiswvMpKQ4iOkqI99SFWjGtu2m8aMLDKSkOIjpKiPfUiVc0Jn5vWM\nfHfZoH5t9pKOnvhdXNEvLXvI3qnNXVKe2tSVktrU/aQytanLS1Kbem/ZI7W5Ky5OeOcvJGlI\nI8a1bzeNrYp8t3Hx420Wdfi/lHcfT82iH6Q3930PpTb1o3enNvXjdz+a2tQP3Zfa1I//YFF6\nc7+b8M5fSNKQbpULlrVsrpwh17taDpBNSUPaOl1keE39tIYJB4vUbnG6JiBzkn+OtPDcfYLP\nkQbXP+JwPUAmderIhnVvLVm20yMbgG4g/WPtgG6AkAAHCAlwgJAABwgJcICQAAcICXCAkAAH\nCAlwgJAABwgJcICQAAcICXCAkAAHCAlwgJAABwgJcGB3hnSCALvRCQ7vzLszpGlnP5+W++Wp\n1OY+elZqU9/aP7Wpn+9/a2pTzzo6tamfkvtTm/vsaQ7vzLszpBR/0+of5KPU5j7x5tSmfnBI\nalN7Qx5MbeoU/6r5R+c+hdYAAAowSURBVPKH1ObuEr9p1QVCiiKkKEKKgZCiCCmKkGIgpChC\niiKkGAgpipCiCCkGQooipChCioGQoggpipBiIKQoQooipBgIKYqQoggpBkKKIqQoQorh0ktT\nm/rl0o2pzX1KsT893Tk/3z+1qb39f57a1PNPSW3qjaUvpza30/vf7gxpzZr05n4zvalXpdfo\n9hU7H5PUiu2pTb1xVWpTp/kf0un9j39GAThASIADhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4\nQEiAA4QEOEBIgAOEBDhASIADhAQ4kGJI2+aP7HnQzduLXFDsKkUHX9X8S9AnJ5vae2BU733O\neaXDoasHfKvYzeh46g1tv5792Xhzq9uReO53L92vfMiMZaksW83tbtnvXVbV64jbkt1HdrZs\nNbe7ZReQYkgXyLjZo8LLVhcUu0rRwaeW1+TcXOiKO5/6q3Lw7Kk9+iztYOTm8fKtwlfc2dSb\nxgdGS2VH/0AvMre6HUnnfmeYnHrVZ6T/Un3FTi9bz+1s2asPLDlnzvFSW+CKnV62ntvZsgtJ\nL6RFcoHnNU6Wxzq8oNhVig8edNxO9lVs6t+XnLzJ8x6RqYVHvjtGwnd209TNZsldHQyNzK1u\nR+K5Pyff8b/eK2ensGw1t7tlz5J/9bymc2VRCstWc7tbdiHphTRJcv9KeIU0dHhBm2PnqBEd\nD/atkot2sq9iU88sDf7fOuuGgsu+q2/JhPCd3TR1YKGcVXBmPbe6HYnnHjC4MXdSXdnkftlq\nbnfLnnJUbs4H5NbWCxwuW83tbtmFpBdSv+HBSfXAQhesm1NVMWzW+y3fKatXI9S1wxbJdzve\n106nLv4/piMO+fVD4Tu7aeqc7YdWvhFz7vDt6NTcO/71x807KGl9SeBu2XpuZ8tuMUd+2rrp\n8qcdndv1svOlFtKHUhOc1sgafcHaw2Xc3PPKqt5r/lbLDQmNUNfOc5vcdHrffvWvFdrXTqf+\nm1zyau3efeo7+K0aj23zwnd209TB6Z0yr6OfSWTu8O3o9Nw5r5UdncayI3O7Xfb6F+aVjt6a\nzrLz5nb+086XWkhLpXl1dbJcX3CpfMPferj1ZUrLDQmNUNfOc6HIiVMPk71fLLCvnU79opzd\n54DzPimDlnkdCN/ZTVPnTnYMr3yvo4kjc4dvh4u5G0+TH6e07PDcbpf9BZGhr7aecbzsvLld\n/7QjUgvpFWn+w4JT5VV1wdbew4On3WN6bPAunDhxYsk+/pd/CY9Q185T2/uX/uvIG6X1/5Gm\nqX8rcu5mz/t6x89/w3d209S5k3+XS4r9WPJCar8dLuZuukwmtr6Mcb3s8Nxul73gP76+R+/F\n6Sw7b27HP+2o1EJa1vKSrU7+qi54SUbckHOM/M4b1vqG/ZXhEera+YL/pk1Hyp8TTP2MlOee\nG+84qGx9B0sP39lNU+dOzij+q0HzQmq/HQ7m3nGRjNmQ0rLz5na7bN9vSkY0prLsvLmdLztf\naiGtldOC05rWXx4cuuDJto+7FgQXtTy0hkaoaxdykfwiwdSvyKHB5vnS0e/wDN/ZTVP7Xz+q\nqO54wV40pLbb0fm5N06UU9vv626XnT+302UHjpcVaSw7f273y86T3rt2gw8OTqqH6guel+l5\nQ1vfNQldRV07ZPsrzY9EF8ivE0y9pWxksHmedPRLPPPu7KZVe97PZG4Hs+q5Q7ej03OvGS1T\ntobOu1x2ZG53y974o/8bnJ4pra923S1bze3yp11AeiHVydv+17faPxJov2Bj+YHBW6m337wu\n+E7rDQldRV07ZE1J8Cuyd4zo8UGSqU8oyW02HrL3tg5WnheSaWrPu1IeLfpTCc8duh2dnXvz\nWLmmKXyBw2VH53a37M0VQ3PPu7YNK2993uFu2Wpuhz/tQtIL6WGZ2eQ1TZPH/cWvXpt/wTS5\nzr/gqbIRzf+BWm9IaERoU6uRO/2vN8nMRFPfK3U7PO8OuaKjlbfc2RNM7XknyodFfyp5c4du\nRyfnvrr9hbHzZau53S37fJnvv2qZm3uAcL5sNbe7ZReS4rF2tXLitWNlhr/1YvP7a+0XvFcl\no6+ZXtHr6Q6vEt5UlvSTM2ePk8M+SDR1U70cNecsqe7w59RyZ0+yam9wwU+QO5g7dDs6N/eq\nChndfHTYBufL1nM7W7a3cqicfuUoOfzDFH7aam53yy4kxZC2XH9g5Yhbc0+uW35I7Rd4q686\nsGJYw586vkp4U1s2c0j58NnrEk694/YjKode9n50RJv8kGxTlx7T4bQF5m6/HZ2b+6G2V89r\nnS+7wNyulu157168T3nVvNyTL/c/bTW3u2UXwL9HAhwgJMABQgIcICTAAUICHCAkwAFCAhwg\nJMABQgIcICTAAUICHCAkwAFCAhwgJMABQgIcICTAAUICHCAkwAFCAhwgJMABQgIcICTAAUIC\nHCAkwAFCAhwgJMABQgIcICTAAUICHCAkwAFCAhwgJMABQgIcIKQsWCv1HX7vd4u93N9sXL3L\nFoNCCCkLioS0oPRuj5B2P0LKgiIh/UQIqSsgpCwgpC6PkLq4pZMH7nX+n4OQ1s2pqhg2K/fX\n2GsHvHBq74GTX/e86bk/N/6qH9KrlwzqPfZx/3uNN3+id99TH93Ny+52CKlr+8vA0torDhyW\nC2nt4TJu7nllVe/5IVXudeHT/2f/gUu9RTNl2vfW+iEdMGTmmSVlv/e8q2T8vMv7lfxqd6+8\nmyGkrm2S/If/UHRCLqRL5Rv+BQ/L1NwzuTp/89WK2vandidu8Lw7ZZa3qccp/vmXSz69e9fd\n7RBSl/ZRj5NyJ0/5IW3tPbwxtz2mxwa/m+dzm2f3WNMW0mP+1zXyaW9T2UGr/M03t+62NXdP\nhNSlPSPX5E62ltV7L8mIG3KOkd95tSVBJ9fJU20h/TV3QZmf3cVS/qlvvrwb19w9EVKX9iu5\nITjtU+89Ka0WeLUDgktvkkfy37UrO9Hztn/3E/6Yo5/bbWvungipS3tGLs2dbCmp956X6W0X\n11YGz/K+JM/okHxv3zNRBm7c5Yvt1gipS1tfcVSTl+up3ttYfuD23EW337zO7+a/c5uf7Pl3\n7/5ISG9+5Ze5rUnyh9215u6JkLq2qfIDz9t8Su5du2lynX/BU2UjmvxuTt3keT+WCz3vZ3Kb\nFw7pnZJjtvjP744p+9vuXXh3Q0hd27sHyMSrDtuvwg/pvSoZfc30il5P57oZMPKqhpLq9zzv\nWekzYUn4qd3VctCXrjmy+U0K7DKE1MW9c9GQ3mf8ZY/ckQ2rrzqwYljDn7xcSC+d2WvoFcGD\nzpV77/mLcEg77jq2b+/j7m7anYvuhggpizi0rsshpCwipC6HkLKIkLocQsoiQupyCAlwgJAA\nBwgJcICQAAcICXCAkAAHCAlwgJAABwgJcICQAAcICXCAkAAHCAlwgJAABwgJcICQAAcICXCA\nkAAHCAlwgJAABwgJcICQAAcICXCAkAAHCAlwgJAABwgJcOB/AMDoNovZbrkYAAAAAElFTkSu\nQmCC",
      "text/plain": [
       "Plot with title “Histogram of depths”"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(phyloseq)\n",
    "library(ggplot2)\n",
    "library(dplyr)\n",
    "#Make sure the data has the sample names on rows if not just transform the data with t()\n",
    "abund_table<-read.csv(\"Unknown_Cutoff.csv\",row.names=1,check.names=FALSE)\n",
    "meta_table<-read.csv(\"Mapping_file_w_og.csv\",row.names=1,check.names=FALSE)\n",
    "#Transpose the data to have sample names on rows\n",
    "abund_table<-t(abund_table)\n",
    "meta_table<-data.frame(meta_table)\n",
    "#Now load the taxonomy (this is fake)\n",
    "OTU_taxonomy<-read.csv(\"taxonomy.csv\",row.names=1,check.names=FALSE)\n",
    "#Convert to phyloseq\n",
    "library(phyloseq)\n",
    "#Convert the data to phyloseq format\n",
    "OTU = otu_table(as.matrix(abund_table), taxa_are_rows = FALSE)\n",
    "TAX = tax_table(as.matrix(OTU_taxonomy))\n",
    "SAM = sample_data(meta_table)\n",
    "physeq<-merge_phyloseq(phyloseq(OTU, TAX),SAM)\n",
    "# Start to explore the data a bit \n",
    "#number of samples\n",
    "print ('Number of Samples in our Biom Table')\n",
    "nsamples(physeq)\n",
    "# number of sequence variants\n",
    "print ('Number of Sequence variants we have.')\n",
    "ntaxa(physeq)\n",
    "#summary statistics of sampling depth\n",
    "print ('Sequencing depth.')\n",
    "depths <- sample_sums(physeq)\n",
    "summary(depths)\n",
    "hist(depths)\n",
    "head(sort(sample_sums(physeq)),25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-P</li>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Co-L</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Placebo\n",
       "\\item Co-P\n",
       "\\item Letrozole\n",
       "\\item Co-L\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Placebo\n",
       "2. Co-P\n",
       "3. Letrozole\n",
       "4. Co-L\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Placebo   Co-P      Letrozole Co-L     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'2'\n",
       "</details>"
      ],
      "text/latex": [
       "2\n",
       "\\emph{Levels}: '2'"
      ],
      "text/markdown": [
       "2\n",
       "**Levels**: '2'"
      ],
      "text/plain": [
       "[1] 2\n",
       "Levels: 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '0' & Week != '1'& Week != '3'& Week != '4'& Week != '5')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# It is always important to set a seed when you subsample so your result is replicable . Here we will be \n",
    "# subsampling libraries with replacement \n",
    "set.seed(123)\n",
    "        \n",
    " \n",
    "    \n",
    "meta.rarefy <- rarefy_even_depth(physeq.pt, sample.size = 4530000, verbose = FALSE, replace = TRUE)#need to change sample size based off min sequencing depth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df  Sum Sq   Mean Sq      F N.Perm Pr(>F)  \n",
       "Groups     3 0.03739 0.0124632 3.0864    999  0.043 *\n",
       "Residuals 28 0.11307 0.0040381                       \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df  Sum Sq   Mean Sq      F N.Perm Pr(>F)  \n",
       "Groups     3 0.03739 0.0124632 3.0864    999  0.047 *\n",
       "Residuals 28 0.11307 0.0040381                       \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'BarcodeSequence'</li>\n",
       "\t<li>'LinkerPrimerSequence'</li>\n",
       "\t<li>'filename'</li>\n",
       "\t<li>'ID_OG_MAP'</li>\n",
       "\t<li>'Barcode'</li>\n",
       "\t<li>'adaptor'</li>\n",
       "\t<li>'TSRI_Seq_plate'</li>\n",
       "\t<li>'Well_Position'</li>\n",
       "\t<li>'Sequenc_.Name'</li>\n",
       "\t<li>'Sequence'</li>\n",
       "\t<li>'PCR_Date'</li>\n",
       "\t<li>'Mouse'</li>\n",
       "\t<li>'Cage'</li>\n",
       "\t<li>'Week'</li>\n",
       "\t<li>'mouse.time'</li>\n",
       "\t<li>'Cecum'</li>\n",
       "\t<li>'Category_type_time'</li>\n",
       "\t<li>'weight'</li>\n",
       "\t<li>'Category'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'Lh'</li>\n",
       "\t<li>'Insulin'</li>\n",
       "\t<li>'Testosterone'</li>\n",
       "\t<li>'Weight_g'</li>\n",
       "\t<li>'observed_SVs1250'</li>\n",
       "\t<li>'pielou_e1250'</li>\n",
       "\t<li>'faith_pd1250'</li>\n",
       "\t<li>'shannon1250'</li>\n",
       "\t<li>'FBG'</li>\n",
       "\t<li>'Description'</li>\n",
       "\t<li>'ATTRIBUTE_GNPSDefaultGroup'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BarcodeSequence'\n",
       "\\item 'LinkerPrimerSequence'\n",
       "\\item 'filename'\n",
       "\\item 'ID\\_OG\\_MAP'\n",
       "\\item 'Barcode'\n",
       "\\item 'adaptor'\n",
       "\\item 'TSRI\\_Seq\\_plate'\n",
       "\\item 'Well\\_Position'\n",
       "\\item 'Sequenc\\_.Name'\n",
       "\\item 'Sequence'\n",
       "\\item 'PCR\\_Date'\n",
       "\\item 'Mouse'\n",
       "\\item 'Cage'\n",
       "\\item 'Week'\n",
       "\\item 'mouse.time'\n",
       "\\item 'Cecum'\n",
       "\\item 'Category\\_type\\_time'\n",
       "\\item 'weight'\n",
       "\\item 'Category'\n",
       "\\item 'type'\n",
       "\\item 'Lh'\n",
       "\\item 'Insulin'\n",
       "\\item 'Testosterone'\n",
       "\\item 'Weight\\_g'\n",
       "\\item 'observed\\_SVs1250'\n",
       "\\item 'pielou\\_e1250'\n",
       "\\item 'faith\\_pd1250'\n",
       "\\item 'shannon1250'\n",
       "\\item 'FBG'\n",
       "\\item 'Description'\n",
       "\\item 'ATTRIBUTE\\_GNPSDefaultGroup'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BarcodeSequence'\n",
       "2. 'LinkerPrimerSequence'\n",
       "3. 'filename'\n",
       "4. 'ID_OG_MAP'\n",
       "5. 'Barcode'\n",
       "6. 'adaptor'\n",
       "7. 'TSRI_Seq_plate'\n",
       "8. 'Well_Position'\n",
       "9. 'Sequenc_.Name'\n",
       "10. 'Sequence'\n",
       "11. 'PCR_Date'\n",
       "12. 'Mouse'\n",
       "13. 'Cage'\n",
       "14. 'Week'\n",
       "15. 'mouse.time'\n",
       "16. 'Cecum'\n",
       "17. 'Category_type_time'\n",
       "18. 'weight'\n",
       "19. 'Category'\n",
       "20. 'type'\n",
       "21. 'Lh'\n",
       "22. 'Insulin'\n",
       "23. 'Testosterone'\n",
       "24. 'Weight_g'\n",
       "25. 'observed_SVs1250'\n",
       "26. 'pielou_e1250'\n",
       "27. 'faith_pd1250'\n",
       "28. 'shannon1250'\n",
       "29. 'FBG'\n",
       "30. 'Description'\n",
       "31. 'ATTRIBUTE_GNPSDefaultGroup'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"BarcodeSequence\"            \"LinkerPrimerSequence\"      \n",
       " [3] \"filename\"                   \"ID_OG_MAP\"                 \n",
       " [5] \"Barcode\"                    \"adaptor\"                   \n",
       " [7] \"TSRI_Seq_plate\"             \"Well_Position\"             \n",
       " [9] \"Sequenc_.Name\"              \"Sequence\"                  \n",
       "[11] \"PCR_Date\"                   \"Mouse\"                     \n",
       "[13] \"Cage\"                       \"Week\"                      \n",
       "[15] \"mouse.time\"                 \"Cecum\"                     \n",
       "[17] \"Category_type_time\"         \"weight\"                    \n",
       "[19] \"Category\"                   \"type\"                      \n",
       "[21] \"Lh\"                         \"Insulin\"                   \n",
       "[23] \"Testosterone\"               \"Weight_g\"                  \n",
       "[25] \"observed_SVs1250\"           \"pielou_e1250\"              \n",
       "[27] \"faith_pd1250\"               \"shannon1250\"               \n",
       "[29] \"FBG\"                        \"Description\"               \n",
       "[31] \"ATTRIBUTE_GNPSDefaultGroup\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Df</th><th scope=col>SumsOfSqs</th><th scope=col>MeanSqs</th><th scope=col>F.Model</th><th scope=col>R2</th><th scope=col>Pr(&gt;F)</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Category</th><td> 3         </td><td>2.227863   </td><td>0.7426212  </td><td>6.095338   </td><td>0.3950656  </td><td>0.000999001</td></tr>\n",
       "\t<tr><th scope=row>Residuals</th><td>28         </td><td>3.411360   </td><td>0.1218343  </td><td>      NA   </td><td>0.6049344  </td><td>         NA</td></tr>\n",
       "\t<tr><th scope=row>Total</th><td>31         </td><td>5.639224   </td><td>       NA  </td><td>      NA   </td><td>1.0000000  </td><td>         NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       "  & Df & SumsOfSqs & MeanSqs & F.Model & R2 & Pr(>F)\\\\\n",
       "\\hline\n",
       "\tCategory &  3          & 2.227863    & 0.7426212   & 6.095338    & 0.3950656   & 0.000999001\\\\\n",
       "\tResiduals & 28          & 3.411360    & 0.1218343   &       NA    & 0.6049344   &          NA\\\\\n",
       "\tTotal & 31          & 5.639224    &        NA   &       NA    & 1.0000000   &          NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Df | SumsOfSqs | MeanSqs | F.Model | R2 | Pr(>F) |\n",
       "|---|---|---|---|---|---|---|\n",
       "| Category |  3          | 2.227863    | 0.7426212   | 6.095338    | 0.3950656   | 0.000999001 |\n",
       "| Residuals | 28          | 3.411360    | 0.1218343   |       NA    | 0.6049344   |          NA |\n",
       "| Total | 31          | 5.639224    |        NA   |       NA    | 1.0000000   |          NA |\n",
       "\n"
      ],
      "text/plain": [
       "          Df SumsOfSqs MeanSqs   F.Model  R2        Pr(>F)     \n",
       "Category   3 2.227863  0.7426212 6.095338 0.3950656 0.000999001\n",
       "Residuals 28 3.411360  0.1218343       NA 0.6049344          NA\n",
       "Total     31 5.639224         NA       NA 1.0000000          NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library('vegan')\n",
    "###############################################\n",
    "#####  PERMANOVA  ############-----------------\n",
    "###############################################\n",
    "\n",
    "#make a data frame from the sample_data (unweighted unifrac)\n",
    "meta.rarefy=subset_samples(meta.rarefy, type == 'fecal')\n",
    "df = as(sample_data(meta.rarefy), \"data.frame\")\n",
    "d = phyloseq::distance(meta.rarefy, \"bray\")\n",
    "\n",
    "#test for homogeneity of data\n",
    "sampledf <- data.frame(sample_data(meta.rarefy))\n",
    "beta <- betadisper(d, sampledf$Category)\n",
    "permutest(beta)\n",
    "\n",
    "CategorybetaDNA <- betadisper(d, sampledf$Category)\n",
    "permutest(CategorybetaDNA)\n",
    "\n",
    "pdf('Metabolite_Cutoff_T2_ellipse.pdf', width=20, height=15)\n",
    "plot(CategorybetaDNA, ellipse = TRUE, hull=F)  \n",
    "dev.off()\n",
    "\n",
    "#now run permanova\n",
    "set.seed(2)\n",
    "GPNMDS = ordinate(meta.rarefy, \"PCoA\", d)\n",
    "sample_variables(meta.rarefy)\n",
    "a=adonis(d ~ Category, as(sample_data(meta.rarefy), \"data.frame\"), permutations=1000, method='bray', weighted=FALSE)\n",
    "a$aov.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "\tHomogeneity of multivariate dispersions\n",
       "\n",
       "Call: betadisper(d = d, group = sampledf$Category)\n",
       "\n",
       "No. of Positive Eigenvalues: 31\n",
       "No. of Negative Eigenvalues: 0\n",
       "\n",
       "Average distance to median:\n",
       "     Co-L      Co-P Letrozole   Placebo \n",
       "   0.2784    0.2947    0.3514    0.3564 \n",
       "\n",
       "Eigenvalues for PCoA axes:\n",
       "(Showing 8 of 31 eigenvalues)\n",
       " PCoA1  PCoA2  PCoA3  PCoA4  PCoA5  PCoA6  PCoA7  PCoA8 \n",
       "2.1266 0.4432 0.3690 0.3086 0.2562 0.2395 0.2038 0.1778 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "CategorybetaDNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 2088 taxa and 32 samples ]\n",
       "sample_data() Sample Data:       [ 32 samples by 31 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 2088 taxa by 7 taxonomic ranks ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "meta.rarefy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Run 0 stress 0.1024477 \n",
      "Run 1 stress 0.1100675 \n",
      "Run 2 stress 0.1072785 \n",
      "Run 3 stress 0.1024477 \n",
      "... Procrustes: rmse 8.144385e-05  max resid 0.0003651294 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.1078921 \n",
      "Run 5 stress 0.1119651 \n",
      "Run 6 stress 0.1307716 \n",
      "Run 7 stress 0.1050306 \n",
      "Run 8 stress 0.1072783 \n",
      "Run 9 stress 0.1222766 \n",
      "Run 10 stress 0.1220036 \n",
      "Run 11 stress 0.1237396 \n",
      "Run 12 stress 0.1167801 \n",
      "Run 13 stress 0.1077955 \n",
      "Run 14 stress 0.1077954 \n",
      "Run 15 stress 0.1077955 \n",
      "Run 16 stress 0.1024477 \n",
      "... Procrustes: rmse 3.59949e-05  max resid 0.0001558139 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1119652 \n",
      "Run 18 stress 0.1318448 \n",
      "Run 19 stress 0.1050303 \n",
      "Run 20 stress 0.1077974 \n",
      "*** Solution reached\n"
     ]
    }
   ],
   "source": [
    "# Calculate distances\n",
    "meta.rarefy=subset_samples(meta.rarefy, type == 'fecal' )\n",
    "set.seed(123)\n",
    "DistBC = distance(meta.rarefy, method = \"bray\")\n",
    "#ordination\n",
    "ordBC = ordinate(meta.rarefy, method = \"NMDS\", distance = DistBC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "                 \n",
    "cap_ord <- ordinate(\n",
    "    physeq = meta.rarefy, \n",
    "    method = \"CAP\",\n",
    "    distance = \"bray\",\n",
    "    formula = ~ Category \n",
    ")\n",
    "pdf('Metabolite_Cutoff_T2_CAP.pdf', width=15, height=15)\n",
    "cap_plot <- plot_ordination(physeq = meta.rarefy,ordination = cap_ord, \n",
    "    color = \"Category\",shape = \"Category\",axes = c(1,2)) + geom_point(size = 10)+\n",
    "    scale_color_manual(values = c(\"blue\", \"purple\", \"darkgreen\", \"red\", \n",
    "    \"#1919ff\", \"darkorchid3\", \"black\")) + ggtitle(\"CAP: Metabolite T2\") + theme_bw() +\n",
    "    theme(text = element_text(size = 50),\n",
    "    panel.grid.major=element_blank(),\n",
    "    panel.grid.minor=element_blank(), \n",
    "    panel.background=element_rect(colour=\"black\",fill=NA,size=2)) + \n",
    "    scale_shape_manual(values=c(15,18,17,19)) + geom_hline(yintercept=0,colour=\"black\",linetype=\"dashed\") + \n",
    "    geom_vline(xintercept=0,colour=\"black\",linetype=\"dashed\") \n",
    "cap_plot\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"BrayCurtis Permanova\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Df</th><th scope=col>SumsOfSqs</th><th scope=col>MeanSqs</th><th scope=col>F.Model</th><th scope=col>R2</th><th scope=col>Pr(&gt;F)</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Category</th><td> 3         </td><td>2.227863   </td><td>0.7426212  </td><td>6.095338   </td><td>0.3950656  </td><td>0.000999001</td></tr>\n",
       "\t<tr><th scope=row>Residuals</th><td>28         </td><td>3.411360   </td><td>0.1218343  </td><td>      NA   </td><td>0.6049344  </td><td>         NA</td></tr>\n",
       "\t<tr><th scope=row>Total</th><td>31         </td><td>5.639224   </td><td>       NA  </td><td>      NA   </td><td>1.0000000  </td><td>         NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       "  & Df & SumsOfSqs & MeanSqs & F.Model & R2 & Pr(>F)\\\\\n",
       "\\hline\n",
       "\tCategory &  3          & 2.227863    & 0.7426212   & 6.095338    & 0.3950656   & 0.000999001\\\\\n",
       "\tResiduals & 28          & 3.411360    & 0.1218343   &       NA    & 0.6049344   &          NA\\\\\n",
       "\tTotal & 31          & 5.639224    &        NA   &       NA    & 1.0000000   &          NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Df | SumsOfSqs | MeanSqs | F.Model | R2 | Pr(>F) |\n",
       "|---|---|---|---|---|---|---|\n",
       "| Category |  3          | 2.227863    | 0.7426212   | 6.095338    | 0.3950656   | 0.000999001 |\n",
       "| Residuals | 28          | 3.411360    | 0.1218343   |       NA    | 0.6049344   |          NA |\n",
       "| Total | 31          | 5.639224    |        NA   |       NA    | 1.0000000   |          NA |\n",
       "\n"
      ],
      "text/plain": [
       "          Df SumsOfSqs MeanSqs   F.Model  R2        Pr(>F)     \n",
       "Category   3 2.227863  0.7426212 6.095338 0.3950656 0.000999001\n",
       "Residuals 28 3.411360  0.1218343       NA 0.6049344          NA\n",
       "Total     31 5.639224         NA       NA 1.0000000          NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print('BrayCurtis Permanova')\n",
    "a=adonis(DistBC ~  Category, as(sample_data(meta.rarefy), \"data.frame\"), permutations=1000, method='bray')\n",
    "a$aov.tab"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CAP T5 BA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-L</li>\n",
       "\t<li>Co-P</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Placebo\n",
       "\\item Co-L\n",
       "\\item Co-P\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Placebo\n",
       "3. Co-L\n",
       "4. Co-P\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Placebo   Co-L      Co-P     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'5'\n",
       "</details>"
      ],
      "text/latex": [
       "5\n",
       "\\emph{Levels}: '5'"
      ],
      "text/markdown": [
       "5\n",
       "**Levels**: '5'"
      ],
      "text/plain": [
       "[1] 5\n",
       "Levels: 5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '0' & Week != '1'& Week != '3'& Week != '4'& Week != '2')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)\n",
       "Groups     3 0.008358 0.0027861 0.9542    999  0.435\n",
       "Residuals 28 0.081751 0.0029197                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)\n",
       "Groups     3 0.008358 0.0027861 0.9542    999  0.425\n",
       "Residuals 28 0.081751 0.0029197                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'BarcodeSequence'</li>\n",
       "\t<li>'LinkerPrimerSequence'</li>\n",
       "\t<li>'filename'</li>\n",
       "\t<li>'ID_OG_MAP'</li>\n",
       "\t<li>'Barcode'</li>\n",
       "\t<li>'adaptor'</li>\n",
       "\t<li>'TSRI_Seq_plate'</li>\n",
       "\t<li>'Well_Position'</li>\n",
       "\t<li>'Sequenc_.Name'</li>\n",
       "\t<li>'Sequence'</li>\n",
       "\t<li>'PCR_Date'</li>\n",
       "\t<li>'Mouse'</li>\n",
       "\t<li>'Cage'</li>\n",
       "\t<li>'Week'</li>\n",
       "\t<li>'mouse.time'</li>\n",
       "\t<li>'Cecum'</li>\n",
       "\t<li>'Category_type_time'</li>\n",
       "\t<li>'weight'</li>\n",
       "\t<li>'Category'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'Lh'</li>\n",
       "\t<li>'Insulin'</li>\n",
       "\t<li>'Testosterone'</li>\n",
       "\t<li>'Weight_g'</li>\n",
       "\t<li>'observed_SVs1250'</li>\n",
       "\t<li>'pielou_e1250'</li>\n",
       "\t<li>'faith_pd1250'</li>\n",
       "\t<li>'shannon1250'</li>\n",
       "\t<li>'FBG'</li>\n",
       "\t<li>'Description'</li>\n",
       "\t<li>'ATTRIBUTE_GNPSDefaultGroup'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BarcodeSequence'\n",
       "\\item 'LinkerPrimerSequence'\n",
       "\\item 'filename'\n",
       "\\item 'ID\\_OG\\_MAP'\n",
       "\\item 'Barcode'\n",
       "\\item 'adaptor'\n",
       "\\item 'TSRI\\_Seq\\_plate'\n",
       "\\item 'Well\\_Position'\n",
       "\\item 'Sequenc\\_.Name'\n",
       "\\item 'Sequence'\n",
       "\\item 'PCR\\_Date'\n",
       "\\item 'Mouse'\n",
       "\\item 'Cage'\n",
       "\\item 'Week'\n",
       "\\item 'mouse.time'\n",
       "\\item 'Cecum'\n",
       "\\item 'Category\\_type\\_time'\n",
       "\\item 'weight'\n",
       "\\item 'Category'\n",
       "\\item 'type'\n",
       "\\item 'Lh'\n",
       "\\item 'Insulin'\n",
       "\\item 'Testosterone'\n",
       "\\item 'Weight\\_g'\n",
       "\\item 'observed\\_SVs1250'\n",
       "\\item 'pielou\\_e1250'\n",
       "\\item 'faith\\_pd1250'\n",
       "\\item 'shannon1250'\n",
       "\\item 'FBG'\n",
       "\\item 'Description'\n",
       "\\item 'ATTRIBUTE\\_GNPSDefaultGroup'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BarcodeSequence'\n",
       "2. 'LinkerPrimerSequence'\n",
       "3. 'filename'\n",
       "4. 'ID_OG_MAP'\n",
       "5. 'Barcode'\n",
       "6. 'adaptor'\n",
       "7. 'TSRI_Seq_plate'\n",
       "8. 'Well_Position'\n",
       "9. 'Sequenc_.Name'\n",
       "10. 'Sequence'\n",
       "11. 'PCR_Date'\n",
       "12. 'Mouse'\n",
       "13. 'Cage'\n",
       "14. 'Week'\n",
       "15. 'mouse.time'\n",
       "16. 'Cecum'\n",
       "17. 'Category_type_time'\n",
       "18. 'weight'\n",
       "19. 'Category'\n",
       "20. 'type'\n",
       "21. 'Lh'\n",
       "22. 'Insulin'\n",
       "23. 'Testosterone'\n",
       "24. 'Weight_g'\n",
       "25. 'observed_SVs1250'\n",
       "26. 'pielou_e1250'\n",
       "27. 'faith_pd1250'\n",
       "28. 'shannon1250'\n",
       "29. 'FBG'\n",
       "30. 'Description'\n",
       "31. 'ATTRIBUTE_GNPSDefaultGroup'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"BarcodeSequence\"            \"LinkerPrimerSequence\"      \n",
       " [3] \"filename\"                   \"ID_OG_MAP\"                 \n",
       " [5] \"Barcode\"                    \"adaptor\"                   \n",
       " [7] \"TSRI_Seq_plate\"             \"Well_Position\"             \n",
       " [9] \"Sequenc_.Name\"              \"Sequence\"                  \n",
       "[11] \"PCR_Date\"                   \"Mouse\"                     \n",
       "[13] \"Cage\"                       \"Week\"                      \n",
       "[15] \"mouse.time\"                 \"Cecum\"                     \n",
       "[17] \"Category_type_time\"         \"weight\"                    \n",
       "[19] \"Category\"                   \"type\"                      \n",
       "[21] \"Lh\"                         \"Insulin\"                   \n",
       "[23] \"Testosterone\"               \"Weight_g\"                  \n",
       "[25] \"observed_SVs1250\"           \"pielou_e1250\"              \n",
       "[27] \"faith_pd1250\"               \"shannon1250\"               \n",
       "[29] \"FBG\"                        \"Description\"               \n",
       "[31] \"ATTRIBUTE_GNPSDefaultGroup\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Df</th><th scope=col>SumsOfSqs</th><th scope=col>MeanSqs</th><th scope=col>F.Model</th><th scope=col>R2</th><th scope=col>Pr(&gt;F)</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Category</th><td> 3        </td><td>0.6046178 </td><td>0.2015393 </td><td>1.379082  </td><td>0.1287368 </td><td>0.04950495</td></tr>\n",
       "\t<tr><th scope=row>Residuals</th><td>28        </td><td>4.0919254 </td><td>0.1461402 </td><td>      NA  </td><td>0.8712632 </td><td>        NA</td></tr>\n",
       "\t<tr><th scope=row>Total</th><td>31        </td><td>4.6965432 </td><td>       NA </td><td>      NA  </td><td>1.0000000 </td><td>        NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       "  & Df & SumsOfSqs & MeanSqs & F.Model & R2 & Pr(>F)\\\\\n",
       "\\hline\n",
       "\tCategory &  3         & 0.6046178  & 0.2015393  & 1.379082   & 0.1287368  & 0.04950495\\\\\n",
       "\tResiduals & 28         & 4.0919254  & 0.1461402  &       NA   & 0.8712632  &         NA\\\\\n",
       "\tTotal & 31         & 4.6965432  &        NA  &       NA   & 1.0000000  &         NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Df | SumsOfSqs | MeanSqs | F.Model | R2 | Pr(>F) |\n",
       "|---|---|---|---|---|---|---|\n",
       "| Category |  3         | 0.6046178  | 0.2015393  | 1.379082   | 0.1287368  | 0.04950495 |\n",
       "| Residuals | 28         | 4.0919254  | 0.1461402  |       NA   | 0.8712632  |         NA |\n",
       "| Total | 31         | 4.6965432  |        NA  |       NA   | 1.0000000  |         NA |\n",
       "\n"
      ],
      "text/plain": [
       "          Df SumsOfSqs MeanSqs   F.Model  R2        Pr(>F)    \n",
       "Category   3 0.6046178 0.2015393 1.379082 0.1287368 0.04950495\n",
       "Residuals 28 4.0919254 0.1461402       NA 0.8712632         NA\n",
       "Total     31 4.6965432        NA       NA 1.0000000         NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# It is always important to set a seed when you subsample so your result is replicable . Here we will be \n",
    "# subsampling libraries with replacement \n",
    "set.seed(123)\n",
    "        \n",
    " \n",
    "    \n",
    "meta.rarefy <- rarefy_even_depth(physeq.pt, sample.size = 4530000, verbose = FALSE, replace = TRUE)#need to change sample size based off min sequencing depth\n",
    "library('vegan')\n",
    "###############################################\n",
    "#####  PERMANOVA  ############-----------------\n",
    "###############################################\n",
    "\n",
    "#make a data frame from the sample_data (unweighted unifrac)\n",
    "meta.rarefy=subset_samples(meta.rarefy, type == 'fecal')\n",
    "df = as(sample_data(meta.rarefy), \"data.frame\")\n",
    "d = phyloseq::distance(meta.rarefy, \"bray\")\n",
    "\n",
    "#test for homogeneity of data\n",
    "sampledf <- data.frame(sample_data(meta.rarefy))\n",
    "beta <- betadisper(d, sampledf$Category)\n",
    "permutest(beta)\n",
    "\n",
    "CategorybetaDNA <- betadisper(d, sampledf$Category)\n",
    "permutest(CategorybetaDNA)\n",
    "\n",
    "pdf('Metabolite_Cutoff_T5_ellipse.pdf', width=20, height=15)\n",
    "plot(CategorybetaDNA, ellipse = TRUE, hull=F)  \n",
    "dev.off()\n",
    "#now run permanova\n",
    "set.seed(2)\n",
    "GPNMDS = ordinate(meta.rarefy, \"PCoA\", d)\n",
    "sample_variables(meta.rarefy)\n",
    "a=adonis(d ~ Category, as(sample_data(meta.rarefy), \"data.frame\"), permutations=100, method='bray', weighted=FALSE)\n",
    "a$aov.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "\tHomogeneity of multivariate dispersions\n",
       "\n",
       "Call: betadisper(d = d, group = sampledf$Category)\n",
       "\n",
       "No. of Positive Eigenvalues: 31\n",
       "No. of Negative Eigenvalues: 0\n",
       "\n",
       "Average distance to median:\n",
       "     Co-L      Co-P Letrozole   Placebo \n",
       "   0.3697    0.3561    0.3636    0.3275 \n",
       "\n",
       "Eigenvalues for PCoA axes:\n",
       "(Showing 8 of 31 eigenvalues)\n",
       " PCoA1  PCoA2  PCoA3  PCoA4  PCoA5  PCoA6  PCoA7  PCoA8 \n",
       "0.6735 0.4684 0.4497 0.3520 0.3395 0.2926 0.2278 0.1941 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "CategorybetaDNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 1961 taxa and 32 samples ]\n",
       "sample_data() Sample Data:       [ 32 samples by 31 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 1961 taxa by 7 taxonomic ranks ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "meta.rarefy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Run 0 stress 0.2335205 \n",
      "Run 1 stress 0.2334365 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03316744  max resid 0.1542197 \n",
      "Run 2 stress 0.2285289 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.13362  max resid 0.6038356 \n",
      "Run 3 stress 0.2358673 \n",
      "Run 4 stress 0.252035 \n",
      "Run 5 stress 0.2379475 \n",
      "Run 6 stress 0.2750234 \n",
      "Run 7 stress 0.2313188 \n",
      "Run 8 stress 0.2349244 \n",
      "Run 9 stress 0.2334186 \n",
      "Run 10 stress 0.2327126 \n",
      "Run 11 stress 0.2492624 \n",
      "Run 12 stress 0.2487494 \n",
      "Run 13 stress 0.2313781 \n",
      "Run 14 stress 0.2453091 \n",
      "Run 15 stress 0.2446711 \n",
      "Run 16 stress 0.2453543 \n",
      "Run 17 stress 0.2373175 \n",
      "Run 18 stress 0.2404158 \n",
      "Run 19 stress 0.2303059 \n",
      "Run 20 stress 0.2530108 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n"
     ]
    }
   ],
   "source": [
    "# Calculate distances\n",
    "meta.rarefy=subset_samples(meta.rarefy, type == 'fecal' )\n",
    "set.seed(123)\n",
    "DistBC = distance(meta.rarefy, method = \"bray\")\n",
    "#ordination\n",
    "ordBC = ordinate(meta.rarefy, method = \"NMDS\", distance = DistBC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cap_ord <- ordinate(\n",
    "    physeq = meta.rarefy, \n",
    "    method = \"CAP\",\n",
    "    distance = \"bray\",\n",
    "    formula = ~ Category \n",
    ")\n",
    "pdf('Metabolite_Cutoff_T5_CAP.pdf', width=15, height=15)\n",
    "cap_plot <- plot_ordination(physeq = meta.rarefy,ordination = cap_ord, \n",
    "    color = \"Category\",shape = \"Category\",axes = c(1,2)) + geom_point(size = 10)+\n",
    "    scale_color_manual(values = c(\"blue\", \"purple\", \"darkgreen\", \"red\", \n",
    "    \"#1919ff\", \"darkorchid3\", \"black\")) + ggtitle(\"CAP: Metabolite T5\") + theme_bw() +\n",
    "    theme(text = element_text(size = 50),\n",
    "    panel.grid.major=element_blank(),\n",
    "    panel.grid.minor=element_blank(), \n",
    "    panel.background=element_rect(colour=\"black\",fill=NA,size=2)) + \n",
    "    scale_shape_manual(values=c(15,18,17,19)) + geom_hline(yintercept=0,colour=\"black\",linetype=\"dashed\") + \n",
    "    geom_vline(xintercept=0,colour=\"black\",linetype=\"dashed\") \n",
    "cap_plot\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"BrayCurtis Permanova\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Df</th><th scope=col>SumsOfSqs</th><th scope=col>MeanSqs</th><th scope=col>F.Model</th><th scope=col>R2</th><th scope=col>Pr(&gt;F)</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Category</th><td> 3         </td><td>0.6046178  </td><td>0.2015393  </td><td>1.379082   </td><td>0.1287368  </td><td>0.007992008</td></tr>\n",
       "\t<tr><th scope=row>Residuals</th><td>28         </td><td>4.0919254  </td><td>0.1461402  </td><td>      NA   </td><td>0.8712632  </td><td>         NA</td></tr>\n",
       "\t<tr><th scope=row>Total</th><td>31         </td><td>4.6965432  </td><td>       NA  </td><td>      NA   </td><td>1.0000000  </td><td>         NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       "  & Df & SumsOfSqs & MeanSqs & F.Model & R2 & Pr(>F)\\\\\n",
       "\\hline\n",
       "\tCategory &  3          & 0.6046178   & 0.2015393   & 1.379082    & 0.1287368   & 0.007992008\\\\\n",
       "\tResiduals & 28          & 4.0919254   & 0.1461402   &       NA    & 0.8712632   &          NA\\\\\n",
       "\tTotal & 31          & 4.6965432   &        NA   &       NA    & 1.0000000   &          NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Df | SumsOfSqs | MeanSqs | F.Model | R2 | Pr(>F) |\n",
       "|---|---|---|---|---|---|---|\n",
       "| Category |  3          | 0.6046178   | 0.2015393   | 1.379082    | 0.1287368   | 0.007992008 |\n",
       "| Residuals | 28          | 4.0919254   | 0.1461402   |       NA    | 0.8712632   |          NA |\n",
       "| Total | 31          | 4.6965432   |        NA   |       NA    | 1.0000000   |          NA |\n",
       "\n"
      ],
      "text/plain": [
       "          Df SumsOfSqs MeanSqs   F.Model  R2        Pr(>F)     \n",
       "Category   3 0.6046178 0.2015393 1.379082 0.1287368 0.007992008\n",
       "Residuals 28 4.0919254 0.1461402       NA 0.8712632          NA\n",
       "Total     31 4.6965432        NA       NA 1.0000000          NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print('BrayCurtis Permanova')\n",
    "a=adonis(DistBC ~  Category, as(sample_data(meta.rarefy), \"data.frame\"), permutations=1000, method='bray')\n",
    "a$aov.tab"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CAP T0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-P</li>\n",
       "\t<li>Co-L</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Placebo\n",
       "\\item Co-P\n",
       "\\item Co-L\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Placebo\n",
       "3. Co-P\n",
       "4. Co-L\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Placebo   Co-P      Co-L     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'0'\n",
       "</details>"
      ],
      "text/latex": [
       "0\n",
       "\\emph{Levels}: '0'"
      ],
      "text/markdown": [
       "0\n",
       "**Levels**: '0'"
      ],
      "text/plain": [
       "[1] 0\n",
       "Levels: 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df  Sum Sq   Mean Sq      F N.Perm Pr(>F)  \n",
       "Groups     3 0.06867 0.0228898 2.6239    999  0.056 .\n",
       "Residuals 28 0.24426 0.0087235                       \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df  Sum Sq   Mean Sq      F N.Perm Pr(>F)  \n",
       "Groups     3 0.06867 0.0228898 2.6239    999  0.044 *\n",
       "Residuals 28 0.24426 0.0087235                       \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'BarcodeSequence'</li>\n",
       "\t<li>'LinkerPrimerSequence'</li>\n",
       "\t<li>'filename'</li>\n",
       "\t<li>'ID_OG_MAP'</li>\n",
       "\t<li>'Barcode'</li>\n",
       "\t<li>'adaptor'</li>\n",
       "\t<li>'TSRI_Seq_plate'</li>\n",
       "\t<li>'Well_Position'</li>\n",
       "\t<li>'Sequenc_.Name'</li>\n",
       "\t<li>'Sequence'</li>\n",
       "\t<li>'PCR_Date'</li>\n",
       "\t<li>'Mouse'</li>\n",
       "\t<li>'Cage'</li>\n",
       "\t<li>'Week'</li>\n",
       "\t<li>'mouse.time'</li>\n",
       "\t<li>'Cecum'</li>\n",
       "\t<li>'Category_type_time'</li>\n",
       "\t<li>'weight'</li>\n",
       "\t<li>'Category'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'Lh'</li>\n",
       "\t<li>'Insulin'</li>\n",
       "\t<li>'Testosterone'</li>\n",
       "\t<li>'Weight_g'</li>\n",
       "\t<li>'observed_SVs1250'</li>\n",
       "\t<li>'pielou_e1250'</li>\n",
       "\t<li>'faith_pd1250'</li>\n",
       "\t<li>'shannon1250'</li>\n",
       "\t<li>'FBG'</li>\n",
       "\t<li>'Description'</li>\n",
       "\t<li>'ATTRIBUTE_GNPSDefaultGroup'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BarcodeSequence'\n",
       "\\item 'LinkerPrimerSequence'\n",
       "\\item 'filename'\n",
       "\\item 'ID\\_OG\\_MAP'\n",
       "\\item 'Barcode'\n",
       "\\item 'adaptor'\n",
       "\\item 'TSRI\\_Seq\\_plate'\n",
       "\\item 'Well\\_Position'\n",
       "\\item 'Sequenc\\_.Name'\n",
       "\\item 'Sequence'\n",
       "\\item 'PCR\\_Date'\n",
       "\\item 'Mouse'\n",
       "\\item 'Cage'\n",
       "\\item 'Week'\n",
       "\\item 'mouse.time'\n",
       "\\item 'Cecum'\n",
       "\\item 'Category\\_type\\_time'\n",
       "\\item 'weight'\n",
       "\\item 'Category'\n",
       "\\item 'type'\n",
       "\\item 'Lh'\n",
       "\\item 'Insulin'\n",
       "\\item 'Testosterone'\n",
       "\\item 'Weight\\_g'\n",
       "\\item 'observed\\_SVs1250'\n",
       "\\item 'pielou\\_e1250'\n",
       "\\item 'faith\\_pd1250'\n",
       "\\item 'shannon1250'\n",
       "\\item 'FBG'\n",
       "\\item 'Description'\n",
       "\\item 'ATTRIBUTE\\_GNPSDefaultGroup'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BarcodeSequence'\n",
       "2. 'LinkerPrimerSequence'\n",
       "3. 'filename'\n",
       "4. 'ID_OG_MAP'\n",
       "5. 'Barcode'\n",
       "6. 'adaptor'\n",
       "7. 'TSRI_Seq_plate'\n",
       "8. 'Well_Position'\n",
       "9. 'Sequenc_.Name'\n",
       "10. 'Sequence'\n",
       "11. 'PCR_Date'\n",
       "12. 'Mouse'\n",
       "13. 'Cage'\n",
       "14. 'Week'\n",
       "15. 'mouse.time'\n",
       "16. 'Cecum'\n",
       "17. 'Category_type_time'\n",
       "18. 'weight'\n",
       "19. 'Category'\n",
       "20. 'type'\n",
       "21. 'Lh'\n",
       "22. 'Insulin'\n",
       "23. 'Testosterone'\n",
       "24. 'Weight_g'\n",
       "25. 'observed_SVs1250'\n",
       "26. 'pielou_e1250'\n",
       "27. 'faith_pd1250'\n",
       "28. 'shannon1250'\n",
       "29. 'FBG'\n",
       "30. 'Description'\n",
       "31. 'ATTRIBUTE_GNPSDefaultGroup'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"BarcodeSequence\"            \"LinkerPrimerSequence\"      \n",
       " [3] \"filename\"                   \"ID_OG_MAP\"                 \n",
       " [5] \"Barcode\"                    \"adaptor\"                   \n",
       " [7] \"TSRI_Seq_plate\"             \"Well_Position\"             \n",
       " [9] \"Sequenc_.Name\"              \"Sequence\"                  \n",
       "[11] \"PCR_Date\"                   \"Mouse\"                     \n",
       "[13] \"Cage\"                       \"Week\"                      \n",
       "[15] \"mouse.time\"                 \"Cecum\"                     \n",
       "[17] \"Category_type_time\"         \"weight\"                    \n",
       "[19] \"Category\"                   \"type\"                      \n",
       "[21] \"Lh\"                         \"Insulin\"                   \n",
       "[23] \"Testosterone\"               \"Weight_g\"                  \n",
       "[25] \"observed_SVs1250\"           \"pielou_e1250\"              \n",
       "[27] \"faith_pd1250\"               \"shannon1250\"               \n",
       "[29] \"FBG\"                        \"Description\"               \n",
       "[31] \"ATTRIBUTE_GNPSDefaultGroup\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Df</th><th scope=col>SumsOfSqs</th><th scope=col>MeanSqs</th><th scope=col>F.Model</th><th scope=col>R2</th><th scope=col>Pr(&gt;F)</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Category</th><td> 3        </td><td>0.4587321 </td><td>0.1529107 </td><td>0.9581512 </td><td>0.09310136</td><td>0.6336634 </td></tr>\n",
       "\t<tr><th scope=row>Residuals</th><td>28        </td><td>4.4685006 </td><td>0.1595893 </td><td>       NA </td><td>0.90689864</td><td>       NA </td></tr>\n",
       "\t<tr><th scope=row>Total</th><td>31        </td><td>4.9272327 </td><td>       NA </td><td>       NA </td><td>1.00000000</td><td>       NA </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       "  & Df & SumsOfSqs & MeanSqs & F.Model & R2 & Pr(>F)\\\\\n",
       "\\hline\n",
       "\tCategory &  3         & 0.4587321  & 0.1529107  & 0.9581512  & 0.09310136 & 0.6336634 \\\\\n",
       "\tResiduals & 28         & 4.4685006  & 0.1595893  &        NA  & 0.90689864 &        NA \\\\\n",
       "\tTotal & 31         & 4.9272327  &        NA  &        NA  & 1.00000000 &        NA \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Df | SumsOfSqs | MeanSqs | F.Model | R2 | Pr(>F) |\n",
       "|---|---|---|---|---|---|---|\n",
       "| Category |  3         | 0.4587321  | 0.1529107  | 0.9581512  | 0.09310136 | 0.6336634  |\n",
       "| Residuals | 28         | 4.4685006  | 0.1595893  |        NA  | 0.90689864 |        NA  |\n",
       "| Total | 31         | 4.9272327  |        NA  |        NA  | 1.00000000 |        NA  |\n",
       "\n"
      ],
      "text/plain": [
       "          Df SumsOfSqs MeanSqs   F.Model   R2         Pr(>F)   \n",
       "Category   3 0.4587321 0.1529107 0.9581512 0.09310136 0.6336634\n",
       "Residuals 28 4.4685006 0.1595893        NA 0.90689864        NA\n",
       "Total     31 4.9272327        NA        NA 1.00000000        NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '5' & Week != '1'& Week != '3'& Week != '4'& Week != '2')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)\n",
    "\n",
    "# It is always important to set a seed when you subsample so your result is replicable . Here we will be \n",
    "# subsampling libraries with replacement \n",
    "set.seed(123)\n",
    "        \n",
    " \n",
    "    \n",
    "meta.rarefy <- rarefy_even_depth(physeq.pt, sample.size = 4530000, verbose = FALSE, replace = TRUE)#need to change sample size based off min sequencing depth\n",
    "library('vegan')\n",
    "###############################################\n",
    "#####  PERMANOVA  ############-----------------\n",
    "###############################################\n",
    "\n",
    "#make a data frame from the sample_data (unweighted unifrac)\n",
    "meta.rarefy=subset_samples(meta.rarefy, type == 'fecal')\n",
    "df = as(sample_data(meta.rarefy), \"data.frame\")\n",
    "d = phyloseq::distance(meta.rarefy, \"bray\")\n",
    "\n",
    "#test for homogeneity of data\n",
    "sampledf <- data.frame(sample_data(meta.rarefy))\n",
    "beta <- betadisper(d, sampledf$Category)\n",
    "permutest(beta)\n",
    "\n",
    "CategorybetaDNA <- betadisper(d, sampledf$Category)\n",
    "permutest(CategorybetaDNA)\n",
    "\n",
    "pdf('Metabolite_Cutoff_T0_ellipse.pdf', width=20, height=15)\n",
    "plot(CategorybetaDNA, ellipse = TRUE, hull=F)  \n",
    "dev.off()\n",
    "#now run permanova\n",
    "set.seed(2)\n",
    "GPNMDS = ordinate(meta.rarefy, \"PCoA\", d)\n",
    "sample_variables(meta.rarefy)\n",
    "a=adonis(d ~ Category, as(sample_data(meta.rarefy), \"data.frame\"), permutations=100, method='bray', weighted=FALSE)\n",
    "a$aov.tab\n",
    "\n",
    "cap_ord <- ordinate(\n",
    "    physeq = meta.rarefy, \n",
    "    method = \"CAP\",\n",
    "    distance = \"bray\",\n",
    "    formula = ~ Category \n",
    ")\n",
    "pdf('Metabolite_Cutoff_T0_CAP.pdf', width=15, height=15)\n",
    "cap_plot <- plot_ordination(physeq = meta.rarefy,ordination = cap_ord, \n",
    "    color = \"Category\",shape = \"Category\",axes = c(1,2)) + geom_point(size = 10)+\n",
    "    scale_color_manual(values = c(\"blue\", \"purple\", \"darkgreen\", \"red\", \n",
    "    \"#1919ff\", \"darkorchid3\", \"black\")) + ggtitle(\"CAP: Metabolite T0\") + theme_bw() +\n",
    "    theme(text = element_text(size = 50),\n",
    "    panel.grid.major=element_blank(),\n",
    "    panel.grid.minor=element_blank(), \n",
    "    panel.background=element_rect(colour=\"black\",fill=NA,size=2)) + \n",
    "    scale_shape_manual(values=c(15,18,17,19)) + geom_hline(yintercept=0,colour=\"black\",linetype=\"dashed\") + \n",
    "    geom_vline(xintercept=0,colour=\"black\",linetype=\"dashed\") \n",
    "cap_plot\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Co-L</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-P</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Co-L\n",
       "\\item Placebo\n",
       "\\item Co-P\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Co-L\n",
       "3. Placebo\n",
       "4. Co-P\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Co-L      Placebo   Co-P     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'1'\n",
       "</details>"
      ],
      "text/latex": [
       "1\n",
       "\\emph{Levels}: '1'"
      ],
      "text/markdown": [
       "1\n",
       "**Levels**: '1'"
      ],
      "text/plain": [
       "[1] 1\n",
       "Levels: 1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)\n",
       "Groups     3 0.010403 0.0034678 0.9119    999  0.455\n",
       "Residuals 28 0.106474 0.0038026                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)\n",
       "Groups     3 0.010403 0.0034678 0.9119    999  0.426\n",
       "Residuals 28 0.106474 0.0038026                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'BarcodeSequence'</li>\n",
       "\t<li>'LinkerPrimerSequence'</li>\n",
       "\t<li>'filename'</li>\n",
       "\t<li>'ID_OG_MAP'</li>\n",
       "\t<li>'Barcode'</li>\n",
       "\t<li>'adaptor'</li>\n",
       "\t<li>'TSRI_Seq_plate'</li>\n",
       "\t<li>'Well_Position'</li>\n",
       "\t<li>'Sequenc_.Name'</li>\n",
       "\t<li>'Sequence'</li>\n",
       "\t<li>'PCR_Date'</li>\n",
       "\t<li>'Mouse'</li>\n",
       "\t<li>'Cage'</li>\n",
       "\t<li>'Week'</li>\n",
       "\t<li>'mouse.time'</li>\n",
       "\t<li>'Cecum'</li>\n",
       "\t<li>'Category_type_time'</li>\n",
       "\t<li>'weight'</li>\n",
       "\t<li>'Category'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'Lh'</li>\n",
       "\t<li>'Insulin'</li>\n",
       "\t<li>'Testosterone'</li>\n",
       "\t<li>'Weight_g'</li>\n",
       "\t<li>'observed_SVs1250'</li>\n",
       "\t<li>'pielou_e1250'</li>\n",
       "\t<li>'faith_pd1250'</li>\n",
       "\t<li>'shannon1250'</li>\n",
       "\t<li>'FBG'</li>\n",
       "\t<li>'Description'</li>\n",
       "\t<li>'ATTRIBUTE_GNPSDefaultGroup'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BarcodeSequence'\n",
       "\\item 'LinkerPrimerSequence'\n",
       "\\item 'filename'\n",
       "\\item 'ID\\_OG\\_MAP'\n",
       "\\item 'Barcode'\n",
       "\\item 'adaptor'\n",
       "\\item 'TSRI\\_Seq\\_plate'\n",
       "\\item 'Well\\_Position'\n",
       "\\item 'Sequenc\\_.Name'\n",
       "\\item 'Sequence'\n",
       "\\item 'PCR\\_Date'\n",
       "\\item 'Mouse'\n",
       "\\item 'Cage'\n",
       "\\item 'Week'\n",
       "\\item 'mouse.time'\n",
       "\\item 'Cecum'\n",
       "\\item 'Category\\_type\\_time'\n",
       "\\item 'weight'\n",
       "\\item 'Category'\n",
       "\\item 'type'\n",
       "\\item 'Lh'\n",
       "\\item 'Insulin'\n",
       "\\item 'Testosterone'\n",
       "\\item 'Weight\\_g'\n",
       "\\item 'observed\\_SVs1250'\n",
       "\\item 'pielou\\_e1250'\n",
       "\\item 'faith\\_pd1250'\n",
       "\\item 'shannon1250'\n",
       "\\item 'FBG'\n",
       "\\item 'Description'\n",
       "\\item 'ATTRIBUTE\\_GNPSDefaultGroup'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BarcodeSequence'\n",
       "2. 'LinkerPrimerSequence'\n",
       "3. 'filename'\n",
       "4. 'ID_OG_MAP'\n",
       "5. 'Barcode'\n",
       "6. 'adaptor'\n",
       "7. 'TSRI_Seq_plate'\n",
       "8. 'Well_Position'\n",
       "9. 'Sequenc_.Name'\n",
       "10. 'Sequence'\n",
       "11. 'PCR_Date'\n",
       "12. 'Mouse'\n",
       "13. 'Cage'\n",
       "14. 'Week'\n",
       "15. 'mouse.time'\n",
       "16. 'Cecum'\n",
       "17. 'Category_type_time'\n",
       "18. 'weight'\n",
       "19. 'Category'\n",
       "20. 'type'\n",
       "21. 'Lh'\n",
       "22. 'Insulin'\n",
       "23. 'Testosterone'\n",
       "24. 'Weight_g'\n",
       "25. 'observed_SVs1250'\n",
       "26. 'pielou_e1250'\n",
       "27. 'faith_pd1250'\n",
       "28. 'shannon1250'\n",
       "29. 'FBG'\n",
       "30. 'Description'\n",
       "31. 'ATTRIBUTE_GNPSDefaultGroup'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"BarcodeSequence\"            \"LinkerPrimerSequence\"      \n",
       " [3] \"filename\"                   \"ID_OG_MAP\"                 \n",
       " [5] \"Barcode\"                    \"adaptor\"                   \n",
       " [7] \"TSRI_Seq_plate\"             \"Well_Position\"             \n",
       " [9] \"Sequenc_.Name\"              \"Sequence\"                  \n",
       "[11] \"PCR_Date\"                   \"Mouse\"                     \n",
       "[13] \"Cage\"                       \"Week\"                      \n",
       "[15] \"mouse.time\"                 \"Cecum\"                     \n",
       "[17] \"Category_type_time\"         \"weight\"                    \n",
       "[19] \"Category\"                   \"type\"                      \n",
       "[21] \"Lh\"                         \"Insulin\"                   \n",
       "[23] \"Testosterone\"               \"Weight_g\"                  \n",
       "[25] \"observed_SVs1250\"           \"pielou_e1250\"              \n",
       "[27] \"faith_pd1250\"               \"shannon1250\"               \n",
       "[29] \"FBG\"                        \"Description\"               \n",
       "[31] \"ATTRIBUTE_GNPSDefaultGroup\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Df</th><th scope=col>SumsOfSqs</th><th scope=col>MeanSqs</th><th scope=col>F.Model</th><th scope=col>R2</th><th scope=col>Pr(&gt;F)</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Category</th><td> 3        </td><td>0.7027586 </td><td>0.2342529 </td><td>1.703505  </td><td>0.1543472 </td><td>0.00990099</td></tr>\n",
       "\t<tr><th scope=row>Residuals</th><td>28        </td><td>3.8503435 </td><td>0.1375123 </td><td>      NA  </td><td>0.8456528 </td><td>        NA</td></tr>\n",
       "\t<tr><th scope=row>Total</th><td>31        </td><td>4.5531021 </td><td>       NA </td><td>      NA  </td><td>1.0000000 </td><td>        NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       "  & Df & SumsOfSqs & MeanSqs & F.Model & R2 & Pr(>F)\\\\\n",
       "\\hline\n",
       "\tCategory &  3         & 0.7027586  & 0.2342529  & 1.703505   & 0.1543472  & 0.00990099\\\\\n",
       "\tResiduals & 28         & 3.8503435  & 0.1375123  &       NA   & 0.8456528  &         NA\\\\\n",
       "\tTotal & 31         & 4.5531021  &        NA  &       NA   & 1.0000000  &         NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Df | SumsOfSqs | MeanSqs | F.Model | R2 | Pr(>F) |\n",
       "|---|---|---|---|---|---|---|\n",
       "| Category |  3         | 0.7027586  | 0.2342529  | 1.703505   | 0.1543472  | 0.00990099 |\n",
       "| Residuals | 28         | 3.8503435  | 0.1375123  |       NA   | 0.8456528  |         NA |\n",
       "| Total | 31         | 4.5531021  |        NA  |       NA   | 1.0000000  |         NA |\n",
       "\n"
      ],
      "text/plain": [
       "          Df SumsOfSqs MeanSqs   F.Model  R2        Pr(>F)    \n",
       "Category   3 0.7027586 0.2342529 1.703505 0.1543472 0.00990099\n",
       "Residuals 28 3.8503435 0.1375123       NA 0.8456528         NA\n",
       "Total     31 4.5531021        NA       NA 1.0000000         NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '5' & Week != '0'& Week != '3'& Week != '4'& Week != '2')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)\n",
    "\n",
    "# It is always important to set a seed when you subsample so your result is replicable . Here we will be \n",
    "# subsampling libraries with replacement \n",
    "set.seed(123)\n",
    "        \n",
    " \n",
    "    \n",
    "meta.rarefy <- rarefy_even_depth(physeq.pt, sample.size = 4530000, verbose = FALSE, replace = TRUE)#need to change sample size based off min sequencing depth\n",
    "library('vegan')\n",
    "###############################################\n",
    "#####  PERMANOVA  ############-----------------\n",
    "###############################################\n",
    "\n",
    "#make a data frame from the sample_data (unweighted unifrac)\n",
    "meta.rarefy=subset_samples(meta.rarefy, type == 'fecal')\n",
    "df = as(sample_data(meta.rarefy), \"data.frame\")\n",
    "d = phyloseq::distance(meta.rarefy, \"bray\")\n",
    "\n",
    "#test for homogeneity of data\n",
    "sampledf <- data.frame(sample_data(meta.rarefy))\n",
    "beta <- betadisper(d, sampledf$Category)\n",
    "permutest(beta)\n",
    "\n",
    "CategorybetaDNA <- betadisper(d, sampledf$Category)\n",
    "permutest(CategorybetaDNA)\n",
    "\n",
    "pdf('Metabolite_Cutoff_T1_ellipse.pdf', width=20, height=15)\n",
    "plot(CategorybetaDNA, ellipse = TRUE, hull=F)  \n",
    "dev.off()\n",
    "#now run permanova\n",
    "set.seed(2)\n",
    "GPNMDS = ordinate(meta.rarefy, \"PCoA\", d)\n",
    "sample_variables(meta.rarefy)\n",
    "a=adonis(d ~ Category, as(sample_data(meta.rarefy), \"data.frame\"), permutations=100, method='bray', weighted=FALSE)\n",
    "a$aov.tab\n",
    "\n",
    "cap_ord <- ordinate(\n",
    "    physeq = meta.rarefy, \n",
    "    method = \"CAP\",\n",
    "    distance = \"bray\",\n",
    "    formula = ~ Category \n",
    ")\n",
    "pdf('Metabolite_Cutoff_T1_CAP.pdf', width=15, height=15)\n",
    "cap_plot <- plot_ordination(physeq = meta.rarefy,ordination = cap_ord, \n",
    "    color = \"Category\",shape = \"Category\",axes = c(1,2)) + geom_point(size = 10)+\n",
    "    scale_color_manual(values = c(\"blue\", \"purple\", \"darkgreen\", \"red\", \n",
    "    \"#1919ff\", \"darkorchid3\", \"black\")) + ggtitle(\"CAP: Metabolite T1\") + theme_bw() +\n",
    "    theme(text = element_text(size = 50),\n",
    "    panel.grid.major=element_blank(),\n",
    "    panel.grid.minor=element_blank(), \n",
    "    panel.background=element_rect(colour=\"black\",fill=NA,size=2)) + \n",
    "    scale_shape_manual(values=c(15,18,17,19)) + geom_hline(yintercept=0,colour=\"black\",linetype=\"dashed\") + \n",
    "    geom_vline(xintercept=0,colour=\"black\",linetype=\"dashed\") \n",
    "cap_plot\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# T3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-L</li>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Co-P</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Placebo\n",
       "\\item Co-L\n",
       "\\item Letrozole\n",
       "\\item Co-P\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Placebo\n",
       "2. Co-L\n",
       "3. Letrozole\n",
       "4. Co-P\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Placebo   Co-L      Letrozole Co-P     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "3\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'3'\n",
       "</details>"
      ],
      "text/latex": [
       "3\n",
       "\\emph{Levels}: '3'"
      ],
      "text/markdown": [
       "3\n",
       "**Levels**: '3'"
      ],
      "text/plain": [
       "[1] 3\n",
       "Levels: 3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)\n",
       "Groups     3 0.006584 0.0021945 0.4507    999  0.705\n",
       "Residuals 28 0.136351 0.0048697                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)\n",
       "Groups     3 0.006584 0.0021945 0.4507    999  0.733\n",
       "Residuals 28 0.136351 0.0048697                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'BarcodeSequence'</li>\n",
       "\t<li>'LinkerPrimerSequence'</li>\n",
       "\t<li>'filename'</li>\n",
       "\t<li>'ID_OG_MAP'</li>\n",
       "\t<li>'Barcode'</li>\n",
       "\t<li>'adaptor'</li>\n",
       "\t<li>'TSRI_Seq_plate'</li>\n",
       "\t<li>'Well_Position'</li>\n",
       "\t<li>'Sequenc_.Name'</li>\n",
       "\t<li>'Sequence'</li>\n",
       "\t<li>'PCR_Date'</li>\n",
       "\t<li>'Mouse'</li>\n",
       "\t<li>'Cage'</li>\n",
       "\t<li>'Week'</li>\n",
       "\t<li>'mouse.time'</li>\n",
       "\t<li>'Cecum'</li>\n",
       "\t<li>'Category_type_time'</li>\n",
       "\t<li>'weight'</li>\n",
       "\t<li>'Category'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'Lh'</li>\n",
       "\t<li>'Insulin'</li>\n",
       "\t<li>'Testosterone'</li>\n",
       "\t<li>'Weight_g'</li>\n",
       "\t<li>'observed_SVs1250'</li>\n",
       "\t<li>'pielou_e1250'</li>\n",
       "\t<li>'faith_pd1250'</li>\n",
       "\t<li>'shannon1250'</li>\n",
       "\t<li>'FBG'</li>\n",
       "\t<li>'Description'</li>\n",
       "\t<li>'ATTRIBUTE_GNPSDefaultGroup'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BarcodeSequence'\n",
       "\\item 'LinkerPrimerSequence'\n",
       "\\item 'filename'\n",
       "\\item 'ID\\_OG\\_MAP'\n",
       "\\item 'Barcode'\n",
       "\\item 'adaptor'\n",
       "\\item 'TSRI\\_Seq\\_plate'\n",
       "\\item 'Well\\_Position'\n",
       "\\item 'Sequenc\\_.Name'\n",
       "\\item 'Sequence'\n",
       "\\item 'PCR\\_Date'\n",
       "\\item 'Mouse'\n",
       "\\item 'Cage'\n",
       "\\item 'Week'\n",
       "\\item 'mouse.time'\n",
       "\\item 'Cecum'\n",
       "\\item 'Category\\_type\\_time'\n",
       "\\item 'weight'\n",
       "\\item 'Category'\n",
       "\\item 'type'\n",
       "\\item 'Lh'\n",
       "\\item 'Insulin'\n",
       "\\item 'Testosterone'\n",
       "\\item 'Weight\\_g'\n",
       "\\item 'observed\\_SVs1250'\n",
       "\\item 'pielou\\_e1250'\n",
       "\\item 'faith\\_pd1250'\n",
       "\\item 'shannon1250'\n",
       "\\item 'FBG'\n",
       "\\item 'Description'\n",
       "\\item 'ATTRIBUTE\\_GNPSDefaultGroup'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BarcodeSequence'\n",
       "2. 'LinkerPrimerSequence'\n",
       "3. 'filename'\n",
       "4. 'ID_OG_MAP'\n",
       "5. 'Barcode'\n",
       "6. 'adaptor'\n",
       "7. 'TSRI_Seq_plate'\n",
       "8. 'Well_Position'\n",
       "9. 'Sequenc_.Name'\n",
       "10. 'Sequence'\n",
       "11. 'PCR_Date'\n",
       "12. 'Mouse'\n",
       "13. 'Cage'\n",
       "14. 'Week'\n",
       "15. 'mouse.time'\n",
       "16. 'Cecum'\n",
       "17. 'Category_type_time'\n",
       "18. 'weight'\n",
       "19. 'Category'\n",
       "20. 'type'\n",
       "21. 'Lh'\n",
       "22. 'Insulin'\n",
       "23. 'Testosterone'\n",
       "24. 'Weight_g'\n",
       "25. 'observed_SVs1250'\n",
       "26. 'pielou_e1250'\n",
       "27. 'faith_pd1250'\n",
       "28. 'shannon1250'\n",
       "29. 'FBG'\n",
       "30. 'Description'\n",
       "31. 'ATTRIBUTE_GNPSDefaultGroup'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"BarcodeSequence\"            \"LinkerPrimerSequence\"      \n",
       " [3] \"filename\"                   \"ID_OG_MAP\"                 \n",
       " [5] \"Barcode\"                    \"adaptor\"                   \n",
       " [7] \"TSRI_Seq_plate\"             \"Well_Position\"             \n",
       " [9] \"Sequenc_.Name\"              \"Sequence\"                  \n",
       "[11] \"PCR_Date\"                   \"Mouse\"                     \n",
       "[13] \"Cage\"                       \"Week\"                      \n",
       "[15] \"mouse.time\"                 \"Cecum\"                     \n",
       "[17] \"Category_type_time\"         \"weight\"                    \n",
       "[19] \"Category\"                   \"type\"                      \n",
       "[21] \"Lh\"                         \"Insulin\"                   \n",
       "[23] \"Testosterone\"               \"Weight_g\"                  \n",
       "[25] \"observed_SVs1250\"           \"pielou_e1250\"              \n",
       "[27] \"faith_pd1250\"               \"shannon1250\"               \n",
       "[29] \"FBG\"                        \"Description\"               \n",
       "[31] \"ATTRIBUTE_GNPSDefaultGroup\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Df</th><th scope=col>SumsOfSqs</th><th scope=col>MeanSqs</th><th scope=col>F.Model</th><th scope=col>R2</th><th scope=col>Pr(&gt;F)</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Category</th><td> 3        </td><td>0.597882  </td><td>0.1992940 </td><td>1.435459  </td><td>0.133298  </td><td>0.03960396</td></tr>\n",
       "\t<tr><th scope=row>Residuals</th><td>28        </td><td>3.887421  </td><td>0.1388365 </td><td>      NA  </td><td>0.866702  </td><td>        NA</td></tr>\n",
       "\t<tr><th scope=row>Total</th><td>31        </td><td>4.485303  </td><td>       NA </td><td>      NA  </td><td>1.000000  </td><td>        NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       "  & Df & SumsOfSqs & MeanSqs & F.Model & R2 & Pr(>F)\\\\\n",
       "\\hline\n",
       "\tCategory &  3         & 0.597882   & 0.1992940  & 1.435459   & 0.133298   & 0.03960396\\\\\n",
       "\tResiduals & 28         & 3.887421   & 0.1388365  &       NA   & 0.866702   &         NA\\\\\n",
       "\tTotal & 31         & 4.485303   &        NA  &       NA   & 1.000000   &         NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Df | SumsOfSqs | MeanSqs | F.Model | R2 | Pr(>F) |\n",
       "|---|---|---|---|---|---|---|\n",
       "| Category |  3         | 0.597882   | 0.1992940  | 1.435459   | 0.133298   | 0.03960396 |\n",
       "| Residuals | 28         | 3.887421   | 0.1388365  |       NA   | 0.866702   |         NA |\n",
       "| Total | 31         | 4.485303   |        NA  |       NA   | 1.000000   |         NA |\n",
       "\n"
      ],
      "text/plain": [
       "          Df SumsOfSqs MeanSqs   F.Model  R2       Pr(>F)    \n",
       "Category   3 0.597882  0.1992940 1.435459 0.133298 0.03960396\n",
       "Residuals 28 3.887421  0.1388365       NA 0.866702         NA\n",
       "Total     31 4.485303         NA       NA 1.000000         NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '5' & Week != '1'& Week != '0'& Week != '4'& Week != '2')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)\n",
    "\n",
    "# It is always important to set a seed when you subsample so your result is replicable . Here we will be \n",
    "# subsampling libraries with replacement \n",
    "set.seed(123)\n",
    "        \n",
    " \n",
    "    \n",
    "meta.rarefy <- rarefy_even_depth(physeq.pt, sample.size = 4530000, verbose = FALSE, replace = TRUE)#need to change sample size based off min sequencing depth\n",
    "library('vegan')\n",
    "###############################################\n",
    "#####  PERMANOVA  ############-----------------\n",
    "###############################################\n",
    "\n",
    "#make a data frame from the sample_data (unweighted unifrac)\n",
    "meta.rarefy=subset_samples(meta.rarefy, type == 'fecal')\n",
    "df = as(sample_data(meta.rarefy), \"data.frame\")\n",
    "d = phyloseq::distance(meta.rarefy, \"bray\")\n",
    "\n",
    "#test for homogeneity of data\n",
    "sampledf <- data.frame(sample_data(meta.rarefy))\n",
    "beta <- betadisper(d, sampledf$Category)\n",
    "permutest(beta)\n",
    "\n",
    "CategorybetaDNA <- betadisper(d, sampledf$Category)\n",
    "permutest(CategorybetaDNA)\n",
    "\n",
    "pdf('Metabolite_Cutoff_T3_ellipse.pdf', width=20, height=15)\n",
    "plot(CategorybetaDNA, ellipse = TRUE, hull=F)  \n",
    "dev.off()\n",
    "#now run permanova\n",
    "set.seed(2)\n",
    "GPNMDS = ordinate(meta.rarefy, \"PCoA\", d)\n",
    "sample_variables(meta.rarefy)\n",
    "a=adonis(d ~ Category, as(sample_data(meta.rarefy), \"data.frame\"), permutations=100, method='bray', weighted=FALSE)\n",
    "a$aov.tab\n",
    "\n",
    "cap_ord <- ordinate(\n",
    "    physeq = meta.rarefy, \n",
    "    method = \"CAP\",\n",
    "    distance = \"bray\",\n",
    "    formula = ~ Category \n",
    ")\n",
    "pdf('Metabolite_Cutoff_T3_CAP.pdf', width=15, height=15)\n",
    "cap_plot <- plot_ordination(physeq = meta.rarefy,ordination = cap_ord, \n",
    "    color = \"Category\",shape = \"Category\",axes = c(1,2)) + geom_point(size = 10)+\n",
    "    scale_color_manual(values = c(\"blue\", \"purple\", \"darkgreen\", \"red\", \n",
    "    \"#1919ff\", \"darkorchid3\", \"black\")) + ggtitle(\"CAP: Metabolite T3\") + theme_bw() +\n",
    "    theme(text = element_text(size = 50),\n",
    "    panel.grid.major=element_blank(),\n",
    "    panel.grid.minor=element_blank(), \n",
    "    panel.background=element_rect(colour=\"black\",fill=NA,size=2)) + \n",
    "    scale_shape_manual(values=c(15,18,17,19)) + geom_hline(yintercept=0,colour=\"black\",linetype=\"dashed\") + \n",
    "    geom_vline(xintercept=0,colour=\"black\",linetype=\"dashed\") \n",
    "cap_plot\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# T4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>Letrozole</li>\n",
       "\t<li>Co-P</li>\n",
       "\t<li>Placebo</li>\n",
       "\t<li>Co-L</li>\n",
       "</ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<ol class=list-inline>\n",
       "\t\t<li>'Co-L'</li>\n",
       "\t\t<li>'Co-P'</li>\n",
       "\t\t<li>'Letrozole'</li>\n",
       "\t\t<li>'Placebo'</li>\n",
       "\t</ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item Letrozole\n",
       "\\item Co-P\n",
       "\\item Placebo\n",
       "\\item Co-L\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Co-L'\n",
       "\\item 'Co-P'\n",
       "\\item 'Letrozole'\n",
       "\\item 'Placebo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. Letrozole\n",
       "2. Co-P\n",
       "3. Placebo\n",
       "4. Co-L\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Co-L'\n",
       "2. 'Co-P'\n",
       "3. 'Letrozole'\n",
       "4. 'Placebo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Letrozole Co-P      Placebo   Co-L     \n",
       "Levels: Co-L Co-P Letrozole Placebo"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "4\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t'4'\n",
       "</details>"
      ],
      "text/latex": [
       "4\n",
       "\\emph{Levels}: '4'"
      ],
      "text/markdown": [
       "4\n",
       "**Levels**: '4'"
      ],
      "text/plain": [
       "[1] 4\n",
       "Levels: 4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)\n",
       "Groups     3 0.007386 0.0024621 0.7452    999  0.539\n",
       "Residuals 28 0.092512 0.0033040                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Permutation test for homogeneity of multivariate dispersions\n",
       "Permutation: free\n",
       "Number of permutations: 999\n",
       "\n",
       "Response: Distances\n",
       "          Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)\n",
       "Groups     3 0.007386 0.0024621 0.7452    999  0.511\n",
       "Residuals 28 0.092512 0.0033040                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'BarcodeSequence'</li>\n",
       "\t<li>'LinkerPrimerSequence'</li>\n",
       "\t<li>'filename'</li>\n",
       "\t<li>'ID_OG_MAP'</li>\n",
       "\t<li>'Barcode'</li>\n",
       "\t<li>'adaptor'</li>\n",
       "\t<li>'TSRI_Seq_plate'</li>\n",
       "\t<li>'Well_Position'</li>\n",
       "\t<li>'Sequenc_.Name'</li>\n",
       "\t<li>'Sequence'</li>\n",
       "\t<li>'PCR_Date'</li>\n",
       "\t<li>'Mouse'</li>\n",
       "\t<li>'Cage'</li>\n",
       "\t<li>'Week'</li>\n",
       "\t<li>'mouse.time'</li>\n",
       "\t<li>'Cecum'</li>\n",
       "\t<li>'Category_type_time'</li>\n",
       "\t<li>'weight'</li>\n",
       "\t<li>'Category'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'Lh'</li>\n",
       "\t<li>'Insulin'</li>\n",
       "\t<li>'Testosterone'</li>\n",
       "\t<li>'Weight_g'</li>\n",
       "\t<li>'observed_SVs1250'</li>\n",
       "\t<li>'pielou_e1250'</li>\n",
       "\t<li>'faith_pd1250'</li>\n",
       "\t<li>'shannon1250'</li>\n",
       "\t<li>'FBG'</li>\n",
       "\t<li>'Description'</li>\n",
       "\t<li>'ATTRIBUTE_GNPSDefaultGroup'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BarcodeSequence'\n",
       "\\item 'LinkerPrimerSequence'\n",
       "\\item 'filename'\n",
       "\\item 'ID\\_OG\\_MAP'\n",
       "\\item 'Barcode'\n",
       "\\item 'adaptor'\n",
       "\\item 'TSRI\\_Seq\\_plate'\n",
       "\\item 'Well\\_Position'\n",
       "\\item 'Sequenc\\_.Name'\n",
       "\\item 'Sequence'\n",
       "\\item 'PCR\\_Date'\n",
       "\\item 'Mouse'\n",
       "\\item 'Cage'\n",
       "\\item 'Week'\n",
       "\\item 'mouse.time'\n",
       "\\item 'Cecum'\n",
       "\\item 'Category\\_type\\_time'\n",
       "\\item 'weight'\n",
       "\\item 'Category'\n",
       "\\item 'type'\n",
       "\\item 'Lh'\n",
       "\\item 'Insulin'\n",
       "\\item 'Testosterone'\n",
       "\\item 'Weight\\_g'\n",
       "\\item 'observed\\_SVs1250'\n",
       "\\item 'pielou\\_e1250'\n",
       "\\item 'faith\\_pd1250'\n",
       "\\item 'shannon1250'\n",
       "\\item 'FBG'\n",
       "\\item 'Description'\n",
       "\\item 'ATTRIBUTE\\_GNPSDefaultGroup'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BarcodeSequence'\n",
       "2. 'LinkerPrimerSequence'\n",
       "3. 'filename'\n",
       "4. 'ID_OG_MAP'\n",
       "5. 'Barcode'\n",
       "6. 'adaptor'\n",
       "7. 'TSRI_Seq_plate'\n",
       "8. 'Well_Position'\n",
       "9. 'Sequenc_.Name'\n",
       "10. 'Sequence'\n",
       "11. 'PCR_Date'\n",
       "12. 'Mouse'\n",
       "13. 'Cage'\n",
       "14. 'Week'\n",
       "15. 'mouse.time'\n",
       "16. 'Cecum'\n",
       "17. 'Category_type_time'\n",
       "18. 'weight'\n",
       "19. 'Category'\n",
       "20. 'type'\n",
       "21. 'Lh'\n",
       "22. 'Insulin'\n",
       "23. 'Testosterone'\n",
       "24. 'Weight_g'\n",
       "25. 'observed_SVs1250'\n",
       "26. 'pielou_e1250'\n",
       "27. 'faith_pd1250'\n",
       "28. 'shannon1250'\n",
       "29. 'FBG'\n",
       "30. 'Description'\n",
       "31. 'ATTRIBUTE_GNPSDefaultGroup'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"BarcodeSequence\"            \"LinkerPrimerSequence\"      \n",
       " [3] \"filename\"                   \"ID_OG_MAP\"                 \n",
       " [5] \"Barcode\"                    \"adaptor\"                   \n",
       " [7] \"TSRI_Seq_plate\"             \"Well_Position\"             \n",
       " [9] \"Sequenc_.Name\"              \"Sequence\"                  \n",
       "[11] \"PCR_Date\"                   \"Mouse\"                     \n",
       "[13] \"Cage\"                       \"Week\"                      \n",
       "[15] \"mouse.time\"                 \"Cecum\"                     \n",
       "[17] \"Category_type_time\"         \"weight\"                    \n",
       "[19] \"Category\"                   \"type\"                      \n",
       "[21] \"Lh\"                         \"Insulin\"                   \n",
       "[23] \"Testosterone\"               \"Weight_g\"                  \n",
       "[25] \"observed_SVs1250\"           \"pielou_e1250\"              \n",
       "[27] \"faith_pd1250\"               \"shannon1250\"               \n",
       "[29] \"FBG\"                        \"Description\"               \n",
       "[31] \"ATTRIBUTE_GNPSDefaultGroup\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Df</th><th scope=col>SumsOfSqs</th><th scope=col>MeanSqs</th><th scope=col>F.Model</th><th scope=col>R2</th><th scope=col>Pr(&gt;F)</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Category</th><td> 3       </td><td>0.5457665</td><td>0.1819222</td><td>1.186729 </td><td>0.1128063</td><td>0.0990099</td></tr>\n",
       "\t<tr><th scope=row>Residuals</th><td>28       </td><td>4.2923208</td><td>0.1532972</td><td>      NA </td><td>0.8871937</td><td>       NA</td></tr>\n",
       "\t<tr><th scope=row>Total</th><td>31       </td><td>4.8380873</td><td>       NA</td><td>      NA </td><td>1.0000000</td><td>       NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       "  & Df & SumsOfSqs & MeanSqs & F.Model & R2 & Pr(>F)\\\\\n",
       "\\hline\n",
       "\tCategory &  3        & 0.5457665 & 0.1819222 & 1.186729  & 0.1128063 & 0.0990099\\\\\n",
       "\tResiduals & 28        & 4.2923208 & 0.1532972 &       NA  & 0.8871937 &        NA\\\\\n",
       "\tTotal & 31        & 4.8380873 &        NA &       NA  & 1.0000000 &        NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Df | SumsOfSqs | MeanSqs | F.Model | R2 | Pr(>F) |\n",
       "|---|---|---|---|---|---|---|\n",
       "| Category |  3        | 0.5457665 | 0.1819222 | 1.186729  | 0.1128063 | 0.0990099 |\n",
       "| Residuals | 28        | 4.2923208 | 0.1532972 |       NA  | 0.8871937 |        NA |\n",
       "| Total | 31        | 4.8380873 |        NA |       NA  | 1.0000000 |        NA |\n",
       "\n"
      ],
      "text/plain": [
       "          Df SumsOfSqs MeanSqs   F.Model  R2        Pr(>F)   \n",
       "Category   3 0.5457665 0.1819222 1.186729 0.1128063 0.0990099\n",
       "Residuals 28 4.2923208 0.1532972       NA 0.8871937        NA\n",
       "Total     31 4.8380873        NA       NA 1.0000000        NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "physeq.pt = subset_samples(physeq, Week != 'cecum.T5' & Week != '5' & Week != '1'& Week != '3'& Week != '0'& Week != '2')\n",
    "unique(sample_data(physeq.pt)$Category)\n",
    "unique(sample_data(physeq.pt)$Week)\n",
    "\n",
    "# It is always important to set a seed when you subsample so your result is replicable . Here we will be \n",
    "# subsampling libraries with replacement \n",
    "set.seed(123)\n",
    "        \n",
    " \n",
    "    \n",
    "meta.rarefy <- rarefy_even_depth(physeq.pt, sample.size = 4530000, verbose = FALSE, replace = TRUE)#need to change sample size based off min sequencing depth\n",
    "library('vegan')\n",
    "###############################################\n",
    "#####  PERMANOVA  ############-----------------\n",
    "###############################################\n",
    "\n",
    "#make a data frame from the sample_data (unweighted unifrac)\n",
    "meta.rarefy=subset_samples(meta.rarefy, type == 'fecal')\n",
    "df = as(sample_data(meta.rarefy), \"data.frame\")\n",
    "d = phyloseq::distance(meta.rarefy, \"bray\")\n",
    "\n",
    "#test for homogeneity of data\n",
    "sampledf <- data.frame(sample_data(meta.rarefy))\n",
    "beta <- betadisper(d, sampledf$Category)\n",
    "permutest(beta)\n",
    "\n",
    "CategorybetaDNA <- betadisper(d, sampledf$Category)\n",
    "permutest(CategorybetaDNA)\n",
    "\n",
    "pdf('Metabolite_Cutoff_T4_ellipse.pdf', width=20, height=15)\n",
    "plot(CategorybetaDNA, ellipse = TRUE, hull=F)  \n",
    "dev.off()\n",
    "#now run permanova\n",
    "set.seed(2)\n",
    "GPNMDS = ordinate(meta.rarefy, \"PCoA\", d)\n",
    "sample_variables(meta.rarefy)\n",
    "a=adonis(d ~ Category, as(sample_data(meta.rarefy), \"data.frame\"), permutations=100, method='bray', weighted=FALSE)\n",
    "a$aov.tab\n",
    "\n",
    "cap_ord <- ordinate(\n",
    "    physeq = meta.rarefy, \n",
    "    method = \"CAP\",\n",
    "    distance = \"bray\",\n",
    "    formula = ~ Category \n",
    ")\n",
    "pdf('Metabolite_Cutoff_T4_CAP.pdf', width=15, height=15)\n",
    "cap_plot <- plot_ordination(physeq = meta.rarefy,ordination = cap_ord, \n",
    "    color = \"Category\",shape = \"Category\",axes = c(1,2)) + geom_point(size = 10)+\n",
    "    scale_color_manual(values = c(\"blue\", \"purple\", \"darkgreen\", \"red\", \n",
    "    \"#1919ff\", \"darkorchid3\", \"black\")) + ggtitle(\"CAP: Metabolite T4\") + theme_bw() +\n",
    "    theme(text = element_text(size = 50),\n",
    "    panel.grid.major=element_blank(),\n",
    "    panel.grid.minor=element_blank(), \n",
    "    panel.background=element_rect(colour=\"black\",fill=NA,size=2)) + \n",
    "    scale_shape_manual(values=c(15,18,17,19)) + geom_hline(yintercept=0,colour=\"black\",linetype=\"dashed\") + \n",
    "    geom_vline(xintercept=0,colour=\"black\",linetype=\"dashed\") \n",
    "cap_plot\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.4.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
